---
title: "GO_Figures"
author: "Kai Trepka"
date: "2024-10-15"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Overview

This is an R Markdown file for plotting all GO Study Figures. Most modeling is conducted in separate scripts, with outputs loaded here for plotting. 

## Setup
```{r}
# Libraries
library(dplyr)
library(tidyr) # for tidy data
library(tidyverse) # for tidy data
library(readxl)
library(ggplot2)
library(ggpubr)
library(nlme)
library(vegan)  
library(data.table) # fread
library(reshape2) # Melt
library(stringr) # extract
library(survival) # survival plotting
library(survminer) # survival plotting
library(ggfortify) # survival dataframe
library(clusterProfiler) # for GO enrichment analysis
library(patchwork) # plot alignment
library(qiime2R) # for qza objects
library(phyloseq) # for phyloseq objects
library(ggpmisc) # stat_poly_eq
library(ggrepel) # geom_text_repel
library(ape) # drop_tip
library(ggtree) # tree plotting
library(ggcorrplot) # correlogram
library(pROC) # ROC curves
library(metafolio) # gg_color_hue
library(FactoMineR) # PCA
library(Cairo) # greek letters save

sessionInfo()

# Prefix different if local vs wynton 
prefix = "/mnt/tank/labmainshare/qb3share/"

# Functions
source(paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/AmpliconFunctions.R")) # filterSVTableData, plotpcoa, clrfunction

#Directories - set draft version here as well
dir = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Version_241016/")

##Palettes
dates_palette <- c("BL" = "#6E4236", "C1D1" = "#9B8C69", "C1D3" = "#69B16E", "C1D7" = "#FFC107", "C2D1" = "#004D40", "C3D1" = "#1E88E5", "Final" = "#D81B60")
cap_palette_mouse <- c("Veh" = "grey", "CAP" = "red")
donor_colors <- c("19" = "darkred", "50" = "darkorange", "21" = "darkblue", "32" = "darkred", "35" = "darkgreen")
strain.colors <- c("KO"="orange", "WT"= "grey", "OE" = "blue")
tx_palette <- c("Baseline" = "grey", "Cycle 1" = "orange", "Early" = "orange", "Cycle 3" = "orange", "Trough" = "#FFD580", "Late" = "#FFD580", "Pre" = "grey", "Post" = "orange")
timepoint_palette <- c("BL" = "grey", "C1D7" = "orange", "C1D1" = "grey", "C1D3" = "orange", "C2D1" = "#FFD580", "C3D1" = "#FFD580", "Final" = "#FFD580")
regulation_palette <- c("Up" = "blue", "Down" = "orange")
enriched_palette <- c("Enriched" = "blue", "Depleted" = "orange", "Not Significant" = "gray")
conv_palette <- c("CONV-R" = "purple", "CONV-D" = "blue", "GF" = "orange")
avnm_palette <- c("Veh" = "blue", "AVNM" = "orange")
conv_surv_palette <- c("Group=CONV-R" = "purple", "Group=CONV-D" = "blue", "Group=GF" = "orange")
avnm_surv_palette <- c("Group=Veh" = "blue", "Group=AVNM" = "orange")
preta_palette <- c("preTA++" = "blue", "preTA-wt" = "grey", "ΔpreTA" = "orange")
preta_surv_palette <- c("Group=OE" = "blue", "Group=WT" = "grey", "Group=KO" = "orange")
MIC_5FU_Colors = c("<100" = "orange", ">100" = "blue")
MIC_CAP_Colors = c("<5" = "orange", ">5" = "blue")
gf_preta_palette <- c("preTA++" = "blue", "Mock + ΔpreTA" = "orange")
gf_preta_palette3 <- c("preTA++" = "blue", "Mock" = "orange", "ΔpreTA" = "orange")
gf_preta_surv_palette <- c("Group=OE" = "blue", "Group=Mock + KO" = "orange")
tox_palette <- c("No" = "blue", "Yes" = "orange")
tox_palette_2 <- c("No toxicity" = "grey", "Toxicity" = "red")
sig_palette <- c("Not significant" = "grey", "Both" = "red", "GO" = "orange", "NE" = "yellow")
hilo_pt_palette <- c("High" = "grey", "Low" = "red")
prog_palette <- c("No progression" = "grey", "Progression" = "red")
hilo_palette <- c("High" = "blue", "Low" = "orange")
fu_palette <- c("Veh" = "grey", "5FU" = "red")
cap_palette <- c("Veh" = "grey", "5FU" = "red", "CAP" = "orange")
cmot_palette <- c("Veh, Day 0" = "lightgrey", "CAP, Day 0" = "#FFCCCB", "Veh, Day 4" = "#666666", "CAP, Day 4" = "red")
```

## Helper functions
```{r}
# Function for geometric mean
geometric_mean <- function(x) {
  exp(mean(log(x)))
}

# Function for CLR transform
clr_transform <- function(matrix) {
  matrix[matrix == 0] <- min(matrix[matrix > 0])/2 # Replace zeros with half-minimum non-zero value
  gmean <- apply(matrix, 2, geometric_mean)
  log_matrix <- log(matrix)
  clr_matrix <- sweep(log_matrix, 2, log(gmean), "-")
  return(clr_matrix)
}
```

## Brief refs
For geom_smooth, level = 0.68 is standard error: https://forum.posit.co/t/geom-smooth-shading-based-on-standard-error/154449/2, https://stat.ethz.ch/pipermail/r-sig-mixed-models/2019q4/028218.html

For GSEA: http://yulab-smu.top/biomedical-knowledge-mining-book/clusterprofiler-kegg.html

Common clinical CAP toxicities: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4114495/

# Figure 1 - 16S sequencing
```{r}
# Make directory
figdir <- paste0(dir, "Figure1/")
dir.create(figdir)

# Load 16S data
fn_meta <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/Metadata170523_with16Salpha.csv")
meta <- read.csv(fn_meta) %>% as.data.frame() %>% mutate(Patient_ID = replace(Patient_ID, Patient_ID == "9b", "9"))
stat_in <- meta %>% select(Sample_ID, Patient_ID, Treatment_Cycle, Reads, NumSVs_NotRarefied, NumSVs_Rarefied, InvSimpson_NotRarefied, InvSimpson_Rarefied, Shannon_NotRarefied, Shannon_Rarefied, Cohort, Metadata_Radiation) %>%
  as.data.frame()

# Load date data
fn = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/GO_metadata_16S_WithDates.xlsx")
dates <- read_excel(fn) %>% as.data.frame() %>% select(c(Sample_ID, Patient_ID, Treatment_Cycle, Batch_16S, Days_Since_Tx))
dates <- dates %>% filter(!(Patient_ID %in% c("9", "9c")))
dates$Patient_ID[dates$Patient_ID == "9b"] <- "9"
```

## GO design
See Biorender. 

## GO Shannon baseline (BL)-normalized
```{r}
# Data
stat <- left_join(stat_in, dates, by = c("Patient_ID", "Treatment_Cycle"))
stat$Days_Since_Tx[stat$Days_Since_Tx == "No stool"] <- "No date"
stat <- stat %>% filter(!(Patient_ID %in% c("24", "39"))) # Remove patients with no timecourse
stat$Days_Since_Tx <- as.numeric(stat$Days_Since_Tx) # Numeric

# Set up treatment cycle and times
stat_time <- stat
stat_time$Days_Since_Tx[stat_time$Days_Since_Tx < 0] <- 0
stat_time <- stat_time %>% filter(Days_Since_Tx < 50)

# Baseline normalize!
bl <- stat_time %>% filter(Days_Since_Tx == 0) %>% 
  group_by(Patient_ID) %>% 
  dplyr::summarise(Shannon_NotRarefied = mean(Shannon_NotRarefied))
bl <- bl %>% rename(Baseline = Shannon_NotRarefied)
stat_bl <- left_join(stat_time, bl, by = "Patient_ID")
stat_bl$Shannon_Diff <- stat_bl$Shannon_NotRarefied - stat_bl$Baseline

# Stats testing
y = stat_bl$Shannon_Diff
x = stat_bl$Days_Since_Tx
fit <- lm(0 + y~0 + x)
p.value <- summary(fit)$coefficients[1,"Pr(>|t|)"]

# lmer testing
mod <- nlme::lme(0+Shannon_Diff ~ 0+Days_Since_Tx, random = ~1|Patient_ID, data = stat_bl)
p.value <- (summary(mod)$tTable)[[1,5]]

# Plot
p <- ggplot(stat_bl, aes(x = Days_Since_Tx, y = Shannon_Diff)) + 
  geom_smooth(method = stats::loess, color = "black", fill = "orange", se = TRUE, span = 0.9, fullrange = TRUE, level = 0.68) + 
  theme_pubr() + 
  xlab("Days since treatment start") + 
  ylab("Shannon vs Day 0") + 
  theme(legend.position = "none") + 
  annotate("text", x=38, y=0.01, label = paste("italic(p) == ", formatC(signif(p.value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE)

# Save
fn <- paste0(figdir, "GO_Shannon_BL.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```



## GO Volcano

```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/ASVsLinearD50.csv")
v <- read.csv(fn)

# Set the FDR cutoff value
fdr_cutoff <- 0.2

# Count number altered
num <- v %>% filter(FDR < fdr_cutoff) %>% select(Taxa) %>% count()
num_up <- v %>% filter(FDR < fdr_cutoff) %>% filter(Value > 0) %>% select(Taxa) %>% count()
x_up <- v %>% filter(Value > 0) %>% select(Value) %>% pull() %>% max()/2
num_down <- v %>% filter(FDR < fdr_cutoff) %>% filter(Value < 0) %>% select(Taxa) %>% count()
x_down <- v %>% filter(Value < 0) %>% select(Value) %>% pull() %>% min()/2
ymin <- v %>% select(FDR) %>% pull() %>% min(); yval <- -log10(ymin/4); ylim <- -log10(ymin/4)

# Create a volcano plot
p <- ggplot(data = v, aes(x = Value, y = -log10(FDR))) +
  geom_point(aes(color = ifelse(FDR <= fdr_cutoff, ifelse(Value > 0, "Enriched", "Depleted"), "Not Significant")), size = 1.5, alpha = 1) +
  theme_pubr() +
  labs(x = expression(paste("ASV treatment effect (", Delta, "CLR/day)")), y = expression("Significance (-log"[10]*"(FDR))"), color = "Significance") +
  scale_color_manual(values = enriched_palette) +
  theme(legend.position = "none") + 
  annotate("text", x=x_up, y=yval, label = paste0(num_up, " ASVs")) + 
  annotate("text", x=x_down, y=yval, label = paste0(num_down, " ASVs")) +
  ylim(0, ylim) + 
  xlim(-0.081, 0.08)

# Save
fn <- paste0(figdir, "GO_Volcano.pdf")
ggsave(filename = fn, plot = p, width = 3.5, height = 3)

```

## Order bargraph
```{r}
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/ASVsLinearD50.csv")
v <- read.csv(fn)
fdr_cutoff <- 0.2
v <- v %>% filter(FDR < fdr_cutoff)
v$Direction <- case_when(v$Value > 0 ~ "Enriched",
                         TRUE ~ "Depleted")

# Shorten Peptostreptococcales−Tissierellales
v$Order[v$Order == "Peptostreptococcales-Tissierellales"] <- "Peptostreptococcales"

# Calculate the counts for each Order within each Direction
order_counts <- v %>%
  group_by(Direction, Order) %>%
  summarise(Count = n(), .groups = 'drop') %>%
  arrange(Direction, desc(Count))

# Use the arrange() results to reorder Order within each Direction
v <- v %>%
  mutate(Order = factor(Order, levels = unique(order_counts$Order)))

# Plot
p <- ggplot(v, aes(x = Direction, fill = Order)) +
  geom_bar(width = 0.7, position = "stack", color = "black") +
  labs(y = "Number of ASVs", x = "") +
  theme_pubr() +
  theme(legend.position = "right") +
  theme(legend.key.size = unit(0.4, "cm")) +
  theme(legend.title = element_text(size = 11), 
               legend.text = element_text(size = 10)) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Save
fn <- paste0(figdir, "OrderBargraph.pdf")
ggsave(filename = fn, plot = p, width = 3.5, height = 2.9)
```

## GO tree

All clades (orders) that are consistently enriched/depleted are labeled
```{r, message = FALSE}
# Read in tree
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/ASV_denovotree.qza")
tree = read_qza(fn)$data

# Read in differential abundance results
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/ASVsLinearD50.csv")
v <- read.csv(fn)
fdr_cutoff <- 0.2
all_asvs <- unique(v$ASV)
asvs_up <- v %>% filter(FDR < fdr_cutoff) %>% filter(Value > 0) %>% select(ASV) %>% pull() %>% unique()
asvs_down <- v %>% filter(FDR < fdr_cutoff) %>% filter(Value < 0) %>% select(ASV) %>% pull() %>% unique()

# Plot
tree <- drop.tip(tree, tree$tip.label[!tree$tip.label %in% c(asvs_up, asvs_down)]) # remove all the features from the tree that did not pass filtering cutoffs
tree$Change <- case_when(tree$tip.label %in% asvs_up ~ enriched_palette["Enriched"],
                         tree$tip.label %in% asvs_down ~ enriched_palette["Depleted"],
                         TRUE ~ enriched_palette["Not Significant"])
tree$Size <- case_when(tree$tip.label %in% c(asvs_up, asvs_down) ~ 2,
                         TRUE ~ NA)

# Circle
ggtree(tree, layout="circular") + # branch.length = "none" for cladogram
  geom_tippoint(color=tree$Change, fill = tree$Change, size = tree$Size, shape = 21) +
  geom_strip('8e175abe6a746b8f33bae9cd7c8192bb', '5fbc483f79cd97d3a7253931f37c33e8', barsize=1, color='orange', 
            label="Oscillospirales", offset = -0.2, offset.text=0.1, angle = 60, hjust = 0.5)

# Rectangular
p <- ggtree(tree, layout="rectangular") + 
  geom_tippoint(color=tree$Change, fill = tree$Change, size = tree$Size, shape = 21) +
  geom_strip('8e175abe6a746b8f33bae9cd7c8192bb', '5fbc483f79cd97d3a7253931f37c33e8', barsize=1, color=enriched_palette["Depleted"], 
            label="Oscillospirales", offset = -0.2, fontsize = 3.5, offset.text=0.05, angle = 0, hjust = 0) + 
  geom_strip('5f8d35126b0a7321413d21487ec72b73', '2e81e4d1a54e1636af43e46fa332ab11', barsize=1, color=enriched_palette["Depleted"], 
            label="Erysipelotrichales", offset = -0.2, fontsize = 3.5, offset.text=0.05, angle = 0, hjust = 0) + 
  geom_strip('581a55014641e1cd55cdc272c0365a28', '9e9c4270966cf507485654eb33a69880', barsize=1, color=enriched_palette["Enriched"], 
            label="Lactobacillales", offset = -0.2, fontsize = 3.5, offset.text=0.05, angle = 0, hjust = 0) + 
  xlim(0, 1.35) + 
   geom_treescale(x = 1, y = 40, width = 0.2)

# Save
fn <- paste0(figdir, "GOTree.png")
ggsave(plot = p, filename = fn, height=2.9, width=2.5, units = "in")
fn <- paste0(figdir, "GOTree.pdf")
ggsave(plot = p, filename = fn, height=2.9, width=2.5, device="pdf", useDingbats=F)
```


## Isolate MIC

```{r}
# Load in Nature Micro paper results
fn <-  paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/NatMicro2022_Supp.xlsx")
d <- read_excel(fn, sheet = 1, skip = 2) %>% as.data.frame()
d <- d %>% dplyr::rename(Species = `Species name`)

# Load in model results
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/ASVsLinearD50.csv")
m <- read.csv(fn) %>% as.data.frame()
m <- m %>% dplyr::rename(Estimate = Value)
m$Species <- paste0(m$Genus, " ", m$Species)
m$Species2 <- paste0(m$Genus, " ", m$Species_ambiguous)
m$Species = case_when(m$Species %in% d$Species ~ m$Species,
                      TRUE ~ m$Species2)

# Merge
df <- inner_join(d, m, by = "Species")
df <- df %>% dplyr::rename(MIC_5FU = `5-FU MIC\r\n(μg/ml)`) %>%
  dplyr::rename(MIC_FUDR = `FUDR MIC\r\n(μg/ml)`) %>%
  dplyr::rename(MIC_CAP = `CAP MIC\r\n(mg/ml)`)

# Fix factors, categorize to help visualize
df$MIC_5FU_Cat <- case_when(df$MIC_5FU %in% c("0.3", "2", "2.5", "3.9", "5", "7.8", "15.6", "31.3", "63") ~ "<100",
                         TRUE ~ ">100")
df$MIC_5FU_Cat <- factor(df$MIC_5FU_Cat, levels = c("<100", ">100"))

df$MIC_FUDR_Cat <- case_when(df$MIC_FUDR %in% c("0.2", "0.6", "0.1", "1.6", "2", "3.9", "5") ~ "<5",
                         TRUE ~ ">5")
df$MIC_FUDR_Cat <- factor(df$MIC_FUDR_Cat, levels = c("<5", ">5"))


df$MIC_CAP_Cat <- case_when(df$MIC_CAP %in% c("1", "3", "5") ~ "<5",
                        TRUE ~ ">5")
df$MIC_CAP_Cat <- factor(df$MIC_CAP_Cat, levels = c("<5", ">5"))

# Numerics
df$MIC_5FU[df$MIC_5FU == ">1000"] <- 1500
df$MIC_FUDR[df$MIC_FUDR == ">500"] <- 750

df$MIC_5FU <- as.numeric(df$MIC_5FU)
df$MIC_FUDR <- as.numeric(df$MIC_FUDR)
df$MIC_CAP <- as.numeric(df$MIC_CAP)

# Boxplots
p1 <- ggplot(df, aes(x = MIC_5FU_Cat, y = Estimate)) + 
  geom_boxplot(aes(fill = MIC_5FU_Cat), alpha = 0.5, outlier.size = -1) + 
  geom_jitter(width = 0.05, aes(color = MIC_5FU_Cat)) + 
  ylab(expression(atop("Treatment effect in patients", paste("(", Delta, "CLR/day)")))) + 
  xlab(expression(atop("5-FU MIC", paste("(", mu, "g/mL)")))) + 
  theme_pubr() + 
  stat_compare_means(label.x = 1.25, label.y = 0.03, method = "wilcox", method.args = list(alternative = "less"), aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) +
  scale_fill_manual(values = MIC_5FU_Colors) + 
  scale_color_manual(values = MIC_5FU_Colors) +
  ylim(-0.04, 0.033)

p2 <- ggplot(df, aes(x = MIC_CAP_Cat, y = Estimate)) + 
  geom_boxplot(aes(fill = MIC_CAP_Cat), alpha = 0.5, outlier.size = -1) + 
  geom_jitter(width = 0.05, aes(color = MIC_CAP_Cat)) + 
  ylab(expression(atop("Treatment effect in patients", paste("(", Delta, "CLR/day)")))) + 
  xlab(expression(atop("CAP MIC", "(mg/mL)"))) + 
  theme_pubr() + 
  stat_compare_means(label.x = 1.25, label.y = 0.03, method = "wilcox", method.args = list(alternative = "less"), aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  scale_fill_manual(values = MIC_CAP_Colors) + 
  scale_color_manual(values = MIC_CAP_Colors) + 
  ylim(-0.04, 0.033)

# Save
p <- ggarrange(p1 + theme(legend.position = "none"),
          p2 + theme(legend.position = "none") + rremove("ylab") + theme(axis.text.y = element_blank()), 
          widths = c(1.25, 0.75))
fn <- paste0(figdir, "IsolateSensivity_Both.pdf")
ggsave(filename = fn, plot = p, width = 3.5, height = 2.75)
```

# Figure S1 - Study details
```{r}
# Make directory
figdir <- paste0(dir, "FigureS1/")
dir.create(figdir)
```

## Consort diagram
See BioRender.

## Dates 
```{r}
# Load date data
fn = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/GO_metadata_16S_WithDates.xlsx")
dates <- read_excel(fn) %>% as.data.frame() %>% select(c(Sample_ID, Patient_ID, Treatment_Cycle, Batch_16S, Days_Since_Tx))
dates <- dates %>% filter(!(Patient_ID %in% c("9", "9c")))
dates$Patient_ID[dates$Patient_ID == "9b"] <- "9"
dates$Days_Since_Tx <- as.numeric(dates$Days_Since_Tx)
dates <- dates %>% filter(!is.na(Days_Since_Tx))
dates$Treatment_Cycle[dates$Treatment_Cycle == "Baseline"] <- "BL"

# Add cohort information
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/Metadata170523_with16Salpha.csv")
meta <- read.csv(fn) %>% as.data.frame()
meta$Sample_ID <- gsub("-", "\\.", meta$Sample_ID)
meta <- meta %>% select(Patient_ID, Cohort)
dates <- inner_join(dates, meta, by = "Patient_ID")

# Remove tox samples
dates <- dates %>% filter(Treatment_Cycle != "Toxicity")

# Remove samlpes where stool sequencing failed
r1 <- dates %>% filter(Patient_ID == 4) %>% filter(Treatment_Cycle == "C1D7") %>% select(Sample_ID) %>% pull()
r2 <- dates %>% filter(Patient_ID == 29) %>% filter(Treatment_Cycle == "C1D7") %>% select(Sample_ID) %>% pull()
r <- c(r1, r2)
dates <- dates %>% filter(!(Sample_ID %in% r))

# Only plot for patients with an exact date
dates$Sample <- dates$Treatment_Cycle
dates$SamplingScheme <- "Stool samples (pre-day 100)"
pt_order <- dates %>% select(Patient_ID) %>% pull() %>% as.numeric() %>% unique() %>% sort() %>% rev()
dates$Patient_ID <- factor(dates$Patient_ID, levels = pt_order)
p <- ggplot(dates, aes(x = Days_Since_Tx,y = Patient_ID)) + 
  geom_line(color = "grey") + 
  geom_point(aes(color = Sample)) +
  scale_color_manual(values = dates_palette) + 
  facet_grid(Cohort~., scales = "free", space = "free_y") + 
  xlab("Days since treatment start") + 
  scale_x_continuous(trans = scales::pseudo_log_trans(), breaks = c(-10, 0, 10, 50, 250)) + 
  theme_pubr() +
  geom_vline(xintercept = 50, linetype="dotted") + 
  ylab("Patient") + 
  theme(legend.position = "right")

# Export
fn <- paste0(figdir, "Dates.pdf")
ggsave(file = fn, plot = p, width = 6, height = 6.5)
```

# Figure S2 - more GO 16S
```{r}
# Make directory
figdir <- paste0(dir, "FigureS2/")
dir.create(figdir)

# Load 16S data
fn_meta <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/Metadata170523_with16Salpha.csv")
meta <- read.csv(fn_meta) %>% as.data.frame() %>% mutate(Patient_ID = replace(Patient_ID, Patient_ID == "9b", "9"))
stat_in <- meta %>% select(Sample_ID, Patient_ID, Treatment_Cycle, Reads, NumSVs_NotRarefied, NumSVs_Rarefied, InvSimpson_NotRarefied, InvSimpson_Rarefied, Shannon_NotRarefied, Shannon_Rarefied, Cohort, Metadata_Radiation) %>%
  as.data.frame()

# Load date data
fn = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/GO_metadata_16S_WithDates.xlsx")
dates <- read_excel(fn) %>% as.data.frame() %>% select(c(Sample_ID, Patient_ID, Treatment_Cycle, Batch_16S, Days_Since_Tx))
dates <- dates %>% filter(!(Patient_ID %in% c("9", "9c")))
dates$Patient_ID[dates$Patient_ID == "9b"] <- "9"
```

## PERMANOVA
```{r}
rerun_permanova = FALSE
filter = TRUE # TRUE -> only do Baseline vs C1D7

# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/TaxaSummaryASVs.csv")
d <- fread(fn) %>% as.data.frame()
rownames(d) <- d$V1; d <- d %>% select(-V1) 

# CLR transform
geometric_means <- apply(d, 1, function(row) exp(mean(log(row[row > 0]))))
pseudo_count = min(d[d > 0])/2
clr_data <- log((d + pseudo_count) / geometric_means)

# Omit NA values
clr_data <- na.omit(clr_data)

# Remove zero-variance columns
clr_data_t <- t(clr_data)
clr_data_t <- clr_data_t[ , which(apply(clr_data_t, 2, var) != 0)]
d_t <- t(d)
d_t <- d_t[ , which(apply(d_t, 2, var) != 0)]

# Concatenate metadata
meta_c <- meta %>% filter(Sample_ID %in% rownames(clr_data_t) & Sample_ID %in% rownames(d_t)) # only keep samples that we have sequencing data for
clr_data_t <- clr_data_t[meta_c$Sample_ID,] # and vice versa for metadata matching
d_t <- d_t[meta_c$Sample_ID,]

# PERMANOVA testing 
meta_perm <- meta_c 
if (filter){
  meta_row <- meta_c[meta_c$Sample_ID == "GO-Pt6-C1D1",]
  meta_row$Treatment_Cycle <- "Baseline"  # Establish Patient 6 baseline (only patient that has C1D1, but no BL)
  meta_perm <- rbind(meta_perm, meta_row)
  meta_perm <- meta_perm %>% filter(Treatment_Cycle %in% c("Baseline", "C1D7"))
}
row_order <- meta_perm %>% select(Sample_ID) %>% pull(); clr_data_ordered <-clr_data_t[row_order,]; d_ordered <- d_t[row_order,] # arrange
if(sum(meta_perm$Sample_ID == rownames(clr_data_ordered)) + sum(meta_perm$Sample_ID == rownames(d_ordered)) != 2*length(meta_perm$Sample_ID)){ # check that they match
    print("PERMANVA order wrong!")
}
if (rerun_permanova){
  dist_clr <- suppressWarnings(vegdist(clr_data_ordered, method="euclidean", na.rm = TRUE, binary = TRUE)) # CLR-Euclidean
  dist_bray <- vegdist(d_ordered, method="bray", na.rm = TRUE) # Bray
  set.seed(42)
  res_bray <- adonis2(dist_bray ~ Patient_ID + TreatmentCyclev1, data = meta_perm, permutations=1000, strata = meta_perm$Patient_ID, by = "margin")
  res_clr <- adonis2(dist_clr ~ Patient_ID + TreatmentCyclev1, data = meta_perm, permutations=1000, strata = meta_perm$Patient_ID, by = "margin")
}

# Perform PCA
method = "euclidean" # "bray" or "euclidean"
if (method == "bray"){
  data = d_ordered
  axis = 6
  if (filter){axis = 6}
  pca_results <- prcomp(data, center = TRUE)
} else if (method == "euclidean"){
  data = clr_data_ordered
  axis = 2
  pca_results <- prcomp(data, center = TRUE)
}
eigs <- pca_results$sdev^2
eigs <- round(eigs/sum(eigs) * 100, digits = 1)

# Filter to only specific samples for plotting
if (filter){
  meta_row <- meta_c[meta_c$Sample_ID == "GO-Pt6-C1D1",]
  meta_row$Treatment_Cycle <- "Baseline"  # Establish Patient 6 baseline (only patient that has C1D1, but no BL)
  meta_c <- rbind(meta_c, meta_row)
  meta_c <- meta_c %>% filter(Treatment_Cycle %in% c("Baseline", "C1D7"))
  pt_keep <- meta_c %>% group_by(Patient_ID) %>% dplyr::summarise(n = n()) %>% filter(n == 2) %>% select(Patient_ID) %>% pull()
  meta_c <- meta_c %>% filter(Patient_ID %in% pt_keep)
} else {
  meta_row <- meta_c[meta_c$Sample_ID == "GO-Pt6-C1D1",]
  meta_row$Treatment_Cycle <- "Baseline"  # Establish Patient 6 baseline (only patient that has C1D1, but no BL)
  meta_c <- rbind(meta_c, meta_row)
  meta_c <- meta_c %>% filter(Treatment_Cycle != "C1D1")
}
pca_data <- as.data.frame(pca_results$x)
pca_data$Sample_ID <- rownames(pca_data)
pca_merged <- inner_join(pca_data, meta_c, by = "Sample_ID")

# Axis subject plot
pt_order <- pca_merged %>% filter(Treatment_Cycle == "Baseline") %>% arrange(.data[[paste0("PC", axis)]]) %>% select(Patient_ID) %>% pull()
data2 <- pca_merged
data2$Treatment_Cycle[data2$Treatment_Cycle == "Baseline"] <- "BL"
data2$Patient_ID <- factor(data2$Patient_ID, levels = pt_order)
p <- ggplot(data2 %>% arrange(desc(Treatment_Cycle)), aes(x = .data[[paste0("PC", axis)]], y = Patient_ID)) +
  geom_line(aes(group = Patient_ID), color = "grey", linewidth = 0.5) + 
  geom_point(aes(color = Treatment_Cycle), size = 1.5) +
  theme_pubr() +
  labs(x = paste0("PCo", axis, " (", eigs[axis], "% Variation)"),
       y = "Patient",
       color = "Timepoint") +
  theme(legend.position = c(0.2, 0.7)) + 
  theme(axis.text.y = element_blank()) +
  scale_color_manual(values = timepoint_palette) +
  theme(legend.title=element_blank()) +  
  annotate("text", x = -60, y = 30.5, label = "Tx:~italic(p) < 0.01 * ',' ~ R^2 == 0.01", size = 3.5, hjust = 0, parse = TRUE) +
  annotate("text", x = -60, y = 28.3, label = "Pt:~italic(p) == 0.01 * ',' ~ R^2 == 0.57", size = 3.5, hjust = 0, parse = TRUE) 

# Save
fn <- paste0(figdir, "GO_PCOA_Axis2.pdf")
ggsave(filename = fn, plot = p, width = 3.5, height = 3)
```

## GO Simpson BL norm
```{r}
# Data
stat <- left_join(stat_in, dates, by = c("Patient_ID", "Treatment_Cycle"))
stat$Days_Since_Tx[stat$Days_Since_Tx == "No stool"] <- "No date"
stat <- stat %>% filter(!(Patient_ID %in% c("24", "39"))) # Remove patients with no timecourse
stat$Days_Since_Tx <- as.numeric(stat$Days_Since_Tx) # Numeric

# Set up treatment cycle and times
stat_time <- stat
stat_time$Days_Since_Tx[stat_time$Days_Since_Tx < 0] <- 0
stat_time <- stat_time %>% filter(Days_Since_Tx < 50)

# Baseline normalize!
bl <- stat_time %>% filter(Days_Since_Tx == 0) %>% 
  group_by(Patient_ID) %>% 
  dplyr::summarise(InvSimpson_NotRarefied = mean(InvSimpson_NotRarefied))
bl <- bl %>% rename(Baseline = InvSimpson_NotRarefied)
stat_bl <- left_join(stat_time, bl, by = "Patient_ID")
stat_bl$Simpson_Diff <- stat_bl$InvSimpson_NotRarefied - stat_bl$Baseline

# Stats testing
y = stat_bl$Simpson_Diff
x = stat_bl$Days_Since_Tx
fit <- lm(0 + y~0 + x)
p.value <- summary(fit)$coefficients[1,"Pr(>|t|)"]

# lmer testing
mod <- nlme::lme(0+Simpson_Diff ~ 0+Days_Since_Tx, random = ~1|Patient_ID, data = stat_bl)
p.value <- (summary(mod)$tTable)[[1,5]]

# Plot
p <- ggplot(stat_bl, aes(x = Days_Since_Tx, y = Simpson_Diff)) + 
  geom_smooth(method = stats::loess, color = "black", fill = "orange", se = TRUE, span = 0.9, fullrange = TRUE, level = 0.68) + 
  theme_pubr() + 
  xlab("Days since treatment start") + 
  ylab("Simpson index vs Day 0") + 
  theme(legend.position = "none") + 
  annotate("text", x=10, y=1.5, label = paste("italic(p) == ", formatC(signif(p.value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE)

# Save
fn <- paste0(figdir, "GO_Simpson_BL.pdf")
ggsave(filename = fn, plot = p, width = 2.5, height = 3)
```

## GO ASV BL norm
```{r}
# Data
stat <- left_join(stat_in, dates, by = c("Patient_ID", "Treatment_Cycle"))
stat$Days_Since_Tx[stat$Days_Since_Tx == "No stool"] <- "No date"
stat <- stat %>% filter(!(Patient_ID %in% c("24", "39"))) # Remove patients with no timecourse
stat$Days_Since_Tx <- as.numeric(stat$Days_Since_Tx) # Numeric

# Set up treatment cycle and times
stat_time <- stat
stat_time$Days_Since_Tx[stat_time$Days_Since_Tx < 0] <- 0
stat_time <- stat_time %>% filter(Days_Since_Tx < 50)

# Baseline normalize!
bl <- stat_time %>% filter(Days_Since_Tx == 0) %>% 
  group_by(Patient_ID) %>% 
  dplyr::summarise(NumSVs_Rarefied = mean(NumSVs_Rarefied))
bl <- bl %>% rename(Baseline = NumSVs_Rarefied)
stat_bl <- left_join(stat_time, bl, by = "Patient_ID")
stat_bl$ASV_Diff <- stat_bl$NumSVs_Rarefied - stat_bl$Baseline

# Stats testing
y = stat_bl$ASV_Diff
x = stat_bl$Days_Since_Tx
fit <- lm(0 + y~0 + x)
p.value <- summary(fit)$coefficients[1,"Pr(>|t|)"]

# lmer testing
mod <- nlme::lme(0+ASV_Diff ~ 0+Days_Since_Tx, random = ~1|Patient_ID, data = stat_bl)
p.value <- (summary(mod)$tTable)[[1,5]]

# Plot
p <- ggplot(stat_bl, aes(x = Days_Since_Tx, y = ASV_Diff)) + 
  geom_smooth(method = stats::loess, color = "black", fill = "orange", se = TRUE, span = 0.9, fullrange = TRUE, level = 0.68) + 
  theme_pubr() + 
  xlab("Days since treatment start") + 
  ylab("# ASVs (rarefied) vs Day 0") + 
  theme(legend.position = "none") + 
  annotate("text", x=10, y=10, label = paste("italic(p) == ", formatC(signif(p.value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE)

# Save
fn <- paste0(figdir, "GO_ASV_BL.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## GO Trough vs Cycle 1
```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/ASVsGroups.csv")
v <- read.csv(fn)

# Set the FDR cutoff value
fdr_cutoff <- 0.2

# Separate and merge
v_1 <- v %>% select(-X) %>% filter(Group == "Early") %>% 
  select(Value1 = Value, pVal1 = p.value, Taxa) 
v_2 <- v %>% select(-X) %>% filter(Group == "Late") %>%
  select(Value2 = Value, pVal2 = p.value, Taxa) 
v_new <- inner_join(v_1, v_2, by = "Taxa")

# Significance (nominal)
pcut = 0.05
v_new$Significance <- case_when(v_new$pVal1 < pcut & v_new$pVal2 < pcut ~ "Both",
                                v_new$pVal1 < pcut ~ "Early only",
                                v_new$pVal2 < pcut ~ "Late only",
                                TRUE ~ "Neither")

# Get p value
spearman_test <- cor.test(v_new$Value1, v_new$Value2, method = "spearman")
correlation <- spearman_test$estimate
p_value <- spearman_test$p.value

# Create correlation plot
p <- ggplot(v_new, aes(x = Value1, y = Value2)) + 
  geom_point(color = "grey") + 
  theme_pubr() + 
   labs(x = expression(paste(Delta, "CLR (Cycle 1 - Baseline)")), y = expression(paste(Delta, "CLR (Trough - Baseline)"))) + 
  geom_smooth(method = "lm", color = "black") + 
  annotate("text", x=0, y=2.7, label = sprintf("Spearman's rho = %.2f", correlation)) +
  annotate("text", x=0, y=2.25, label = paste("italic(p) < 1e-16", flag="#"), parse = TRUE) 

# Save
fn <- paste0(figdir, "GO_EarlyLateCorr.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## ASV vs time all
```{r}
# Fdr cutoff
fdr_cut = 0.2

# Load volcano data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/ASVsLinearD50.csv")
v <- read.csv(fn)

# Create dataframe with ASVs of interest
target_asvs <- v %>% filter(FDR < fdr_cut) %>% select(ASV) %>% pull()
target_asv_names <- v %>% filter(FDR < fdr_cut) 
target_asv_names <- case_when(!is.na(target_asv_names$Species) ~ paste0(target_asv_names$Genus, " ", target_asv_names$Species),
                              !is.na(target_asv_names$Genus) ~ target_asv_names$Genus,
                              !is.na(target_asv_names$Family) ~ target_asv_names$Family,
                              TRUE ~ target_asv_names$Order)
target_asv_direction <- v %>% filter(FDR < fdr_cut) %>% select(Value) %>% pull() %>% sign()
asvs <- data.frame("ASV" = target_asvs, "ASV_name" = target_asv_names, "Direction" = target_asv_direction)
ASV_order <- asvs %>% arrange(Direction) %>% select(ASV) %>% pull()
ASV_name_order <- asvs %>% arrange(Direction) %>% select(ASV_name) %>% pull()

# Load data and normalize (relative abundance)
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/TaxaSummaryASVs.csv")
a <- read.csv(fn) %>% as.data.frame() 
rownames(a) <- a$X; a <- a %>% select(-X) # Wrangling
a <- a / colSums(a)*100 # Normalize to relative abundance (%)
a$Taxa <- rownames(a) # Re-add row names for melting

# Melt
a <- a %>% reshape2::melt()
colnames(a) <- c("ASV", "Sample_ID", "Relative abundance")
a$Sample_ID <- gsub("\\.", "-", a$Sample_ID)

# Merge with metadata
a <- left_join(a, dates, by = "Sample_ID")

# Merge with ASV data
a_plot <- inner_join(a, asvs, by = "ASV")

# Remove patients for which the ASV abundance is 0 for the entire timecourse
a_plot <- a_plot %>%
  group_by(ASV, Patient_ID) %>%
  filter(sum(`Relative abundance`) > 0) %>%
  ungroup()

# Labels
ASV_labels = ASV_name_order
names(ASV_labels) <- ASV_order

# Plot
colorscale = c("Enriched" = "blue", "Depleted" = "orange")
a_plot$ASV <- factor(a_plot$ASV, levels = ASV_order)

a_plot$Direction <- case_when(a_plot$Direction > 0 ~ "Enriched", TRUE ~ "Depleted")
a_plot$Days_Since_Tx <- as.numeric(a_plot$Days_Since_Tx)
a_plot$Days_Since_Tx <- case_when(a_plot$Days_Since_Tx < 0 ~ 0, TRUE ~ a_plot$Days_Since_Tx)
a_plot <- a_plot %>% filter(Days_Since_Tx <= 50)
offset = min(a_plot$`Relative abundance`[a_plot$`Relative abundance` > 0])/10
p <- ggplot(a_plot, aes(x = Days_Since_Tx, y = `Relative abundance` + offset, fill = Direction)) + 
  geom_smooth(method = stats::loess, color = "black", se = TRUE, span = 0.9, fullrange = TRUE, level = 0.68) + 
  scale_fill_manual(values = colorscale) + 
  theme_pubr() + 
  facet_wrap(~ASV, scales = "free_y", ncol = 3, labeller = labeller(ASV = ASV_labels)) + 
  scale_y_log10() + 
  scale_x_continuous(expand = c(0,0), limits=c(0, 50)) + 
  xlab("Days since treatment start") + 
  ylab("ASV relative abundance (%)") + 
  theme(legend.position = "none") + 
  theme(strip.text = element_text(face = "italic")) +  # Italicize genera
  theme(strip.text = element_text(size = 10)) + 
  theme(axis.text.y = element_text(size = 10)) +
  theme(axis.text.x = element_text(size = 10))

# Save
fn <- paste0(figdir, "GO_ASVtime.pdf")
ggsave(filename = fn, plot = p, width = 9, height = 16)
```

## Heatmap Cohort
```{r}
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/ASVsLinearD50.csv")
v_in <- read.csv(fn)
hits <- v_in %>% filter(FDR < 0.2) %>% select(Taxa) %>% pull()
v <- v_in %>% filter(Taxa %in% hits) %>% select(Value, Taxa)
v$Cohort = "Full"

label_table <- v_in %>% filter(Taxa %in% hits) %>% select(Taxa, Family, Genus, Order, Species)
label_table$Label <- case_when(!is.na(label_table$Species) ~ paste0(label_table$Genus, " ", label_table$Species),
                               !is.na(label_table$Genus) ~ label_table$Genus,
                               !is.na(label_table$Family) ~ label_table$Family,
                               TRUE ~ label_table$Order)

fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/ASVsLinearD50_CohortA_CAP.csv")
v1 <- read.csv(fn) %>% filter(Taxa %in% hits) %>% select(Value, Taxa); v1$Cohort = "A"
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/ASVsLinearD50_CohortB_TAS102.csv")
v2 <- read.csv(fn) %>% filter(Taxa %in% hits) %>% select(Value, Taxa); v2$Cohort = "B"
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/ASVsLinearD50_CohortC_CAP_IO.csv")
v3 <- read.csv(fn) %>% filter(Taxa %in% hits) %>% select(Value, Taxa); v3$Cohort = "C"

v <-rbind(v, v1, v2, v3)
v <- left_join(v, label_table, by = "Taxa")
v$Cohort <- factor(v$Cohort, levels = c("Full", "A", "B", "C"))
tax_order <- v %>% filter(Cohort == "Full") %>% arrange(Value) %>% select(Taxa) %>% pull()
v$Taxa <- factor(v$Taxa, levels = unique(tax_order))

p <- ggplot(v, aes(x = Cohort, y = Taxa, fill = Value)) + 
  geom_tile(colour="white") + 
  scale_fill_gradient2(low = "orange", high = "blue") + 
  ylab("ASV") + 
  scale_y_discrete(labels = setNames(v$Label, v$Taxa)) + 
  theme_pubr() + 
  theme(axis.text.y = element_text(face = "italic")) + 
  labs(fill = expression(paste(Delta,"CLR/day")), parse = TRUE)

# Save
fn <- paste0(figdir, "GO_CohortHeatmap.pdf")
ggsave(filename = fn, plot = p, width = 6, height = 9)
```

# Figure S3 - NE 16S
```{r}
figdir <- paste0(dir, "FigureS3/")
dir.create(figdir)
```

## NE Design
See Biorender. NE = Netherlands study. 

## NE ASV (BL)-normalized
```{r}
# Data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/NE_16S/Metadata_Analysis_withAlpha.csv")
m <- read.csv(fn) %>% as.data.frame()
m$Days_Since_Tx <- case_when(m$Treatment_Cycle == "T1" ~ 0,
                             m$Treatment_Cycle == "T2" ~ 49,
                             m$Treatment_Cycle == "T3" ~ 63)

# Baseline normalize!
bl <- m %>% filter(Days_Since_Tx == 0) %>% 
  group_by(Patient_ID) %>% 
  dplyr::summarise(NumSVs_Rarefied = mean(NumSVs_Rarefied))
bl <- bl %>% rename(Baseline = NumSVs_Rarefied)
stat_bl <- left_join(m, bl, by = "Patient_ID")
stat_bl$ASV_Diff <- stat_bl$NumSVs_Rarefied - stat_bl$Baseline

# Stats testing
y = stat_bl$ASV_Diff
x = stat_bl$Days_Since_Tx
fit <- lm(0 + y~0 + x)
p.value <- summary(fit)$coefficients[1,"Pr(>|t|)"]

# lmer testing
mod <- nlme::lme(0+ASV_Diff ~ 0+Days_Since_Tx, random = ~1|Patient_ID, data = stat_bl)
p.value <- (summary(mod)$tTable)[[1,5]]

# Plot
p <- ggplot(stat_bl, aes(x = Days_Since_Tx, y = ASV_Diff)) + 
  geom_smooth(method = stats::loess, color = "black", fill = "orange", se = TRUE, span = 0.9, fullrange = TRUE, level = 0.68) + 
  theme_pubr() + 
  xlab("Days since treatment start") + 
  ylab("# ASVs (rarefied) vs Day 0") + 
  theme(legend.position = "none") + 
  annotate("text", x=32, y=10, label = paste("italic(p) == ", formatC(signif(p.value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE)

# Save
fn <- paste0(figdir, "NE_ASV_BL.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## NE Shannon Baseline (BL)-normalized
```{r}
# Data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/NE_16S/Metadata_Analysis_withAlpha.csv")
m <- read.csv(fn) %>% as.data.frame()
m$Days_Since_Tx <- case_when(m$Treatment_Cycle == "T1" ~ 0,
                             m$Treatment_Cycle == "T2" ~ 49,
                             m$Treatment_Cycle == "T3" ~ 63)

# Baseline normalize!
bl <- m %>% filter(Days_Since_Tx == 0) %>% 
  group_by(Patient_ID) %>% 
  dplyr::summarise(Shannon_NotRarefied = mean(Shannon_NotRarefied))
bl <- bl %>% rename(Baseline = Shannon_NotRarefied)
stat_bl <- left_join(m, bl, by = "Patient_ID")
stat_bl$Shannon_Diff <- stat_bl$Shannon_NotRarefied - stat_bl$Baseline

# Stats testing
y = stat_bl$Shannon_Diff
x = stat_bl$Days_Since_Tx
fit <- lm(0 + y~0 + x)
p.value <- summary(fit)$coefficients[1,"Pr(>|t|)"]

# lmer testing
mod <- nlme::lme(0+Shannon_Diff ~ 0+Days_Since_Tx, random = ~1|Patient_ID, data = stat_bl)
p.value <- (summary(mod)$tTable)[[1,5]]

# Plot
p <- ggplot(stat_bl, aes(x = Days_Since_Tx, y = Shannon_Diff)) + 
  geom_smooth(method = stats::loess, color = "black", fill = "orange", se = TRUE, span = 0.9, fullrange = TRUE, level = 0.68) + 
  theme_pubr() + 
  xlab("Days since treatment start") + 
  ylab("Shannon vs Day 0") + 
  ylab("ASV") + 
  theme(legend.position = "none") + 
  annotate("text", x=32, y=0.07, label = paste("italic(p) == ", formatC(signif(p.value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE)

# Save
fn <- paste0(figdir, "NE_Shannon_BL.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## NE Simpson BL-norm
```{r}
# Data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/NE_16S/Metadata_Analysis_withAlpha.csv")
m <- read.csv(fn) %>% as.data.frame()
m$Days_Since_Tx <- case_when(m$Treatment_Cycle == "T1" ~ 0,
                             m$Treatment_Cycle == "T2" ~ 49,
                             m$Treatment_Cycle == "T3" ~ 63)

# Baseline normalize!
bl <- m %>% filter(Days_Since_Tx == 0) %>% 
  group_by(Patient_ID) %>% 
  dplyr::summarise(InvSimpson_NotRarefied = mean(InvSimpson_NotRarefied))
bl <- bl %>% rename(Baseline = InvSimpson_NotRarefied)
stat_bl <- left_join(m, bl, by = "Patient_ID")
stat_bl$Simpson_Diff <- stat_bl$InvSimpson_NotRarefied - stat_bl$Baseline

# Stats testing
y = stat_bl$Simpson_Diff
x = stat_bl$Days_Since_Tx
fit <- lm(0 + y~0 + x)
p.value <- summary(fit)$coefficients[1,"Pr(>|t|)"]

# lmer testing
mod <- nlme::lme(0+Simpson_Diff ~ 0+Days_Since_Tx, random = ~1|Patient_ID, data = stat_bl)
p.value <- (summary(mod)$tTable)[[1,5]]

# Plot
p <- ggplot(stat_bl, aes(x = Days_Since_Tx, y = Simpson_Diff)) + 
  geom_smooth(method = stats::loess, color = "black", fill = "orange", se = TRUE, span = 0.9, fullrange = TRUE, level = 0.68) + 
  theme_pubr() + 
  xlab("Days since treatment start") + 
  ylab("Simpson index vs Day 0") + 
  theme(legend.position = "none") + 
  annotate("text", x=32, y=1.5, label = paste("italic(p) == ", formatC(signif(p.value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE)

# Save
fn <- paste0(figdir, "NE_Simpson_BL.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## NE volcano

Volcano plot of a mixed effects model for ASVs. 

```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/NE_16S/ASVsLinear.csv")
v <- read.csv(fn)

# Set the FDR cutoff value
fdr_cutoff <- 0.2

# Create a volcano plot
p <- ggplot(data = v, aes(x = Value, y = -log10(p.value))) +
  geom_point(aes(color = ifelse(FDR <= fdr_cutoff, ifelse(Value > 0, "Enriched", "Depleted"), "Not Significant")), size = 1.5, alpha = 1) +
  theme_pubr() +
  labs(x = expression(paste("Treatment effect (", Delta, "CLR/day)")), y = "-log10(p)", color = "Significance") +
  scale_color_manual(values = enriched_palette) +
  theme(legend.position = "none")

# Save
fn <- paste0(figdir, "NE_Volcano.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)

# Count number of ASVs altered
num <- v %>% filter(FDR < fdr_cutoff) %>% select(Taxa) %>% count()
```

## NE Early vs Late
```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/NE_16S/ASVsGroups.csv")
v <- read.csv(fn)

# Set the FDR cutoff value
fdr_cutoff <- 0.2

# Separate and merge
v_1 <- v %>% select(-X) %>% filter(Group == "Early") %>% 
  select(Value1 = Value, pVal1 = p.value, Taxa) 
v_2 <- v %>% select(-X) %>% filter(Group == "Late") %>%
  select(Value2 = Value, pVal2 = p.value, Taxa) 
v_new <- inner_join(v_1, v_2, by = "Taxa")

# Significance (nominal)
pcut = 0.05
v_new$Significance <- case_when(v_new$pVal1 < pcut & v_new$pVal2 < pcut ~ "Both",
                                v_new$pVal1 < pcut ~ "Early only",
                                v_new$pVal2 < pcut ~ "Late only",
                                TRUE ~ "Neither")

# Get p value
spearman_test <- cor.test(v_new$Value1, v_new$Value2, method = "spearman")
correlation <- spearman_test$estimate
p_value <- spearman_test$p.value

# Create correlation plot
p <- ggplot(v_new, aes(x = Value1, y = Value2)) + 
  geom_point(color = "grey") + 
  theme_pubr() + 
   labs(x = expression(paste(Delta, "CLR (Cycle 3 - Baseline)")), y = expression(paste(Delta, "CLR (Trough - Baseline)"))) + 
  geom_smooth(method = "lm", color = "black") + 
  annotate("text", x=0, y=2.65, label = sprintf("Spearman's rho = %.2f", correlation)) +
  annotate("text", x=0, y=2.25, label = paste("italic(p) < 1e-16", flag="#"), parse = TRUE) 

# Save
fn <- paste0(figdir, "NE_EarlyLateCorr.pdf")
ggsave(filename = fn, plot = p, width = 3.5, height = 3)
```

## GO/NE correlation
```{r}
# Set taxonomic level
taxLevel = "Species"

# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/NE_16S/", taxLevel, "Linear.csv")
e <- read.csv(fn) %>% 
  as.data.frame() %>% 
  select(Taxa, Value_ERP = Value, FDR_ERP = FDR, p_ERP = p.value)

fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/", taxLevel, "LinearD50.csv")
g <- read.csv(fn) %>% 
  as.data.frame() %>% 
  select(Taxa, Value_GO = Value, FDR_GO = FDR, p_GO = p.value)

d <- inner_join(e, g, by = "Taxa")

# Get p value
spearman_test <- cor.test(d$Value_ERP, d$Value_GO, method = "spearman")
correlation <- spearman_test$estimate
p_value <- spearman_test$p.value

# Create correlation plot
p <- ggplot(d, aes(x = Value_GO, y = Value_ERP)) + 
  geom_point(color = "grey") + 
  theme_pubr() + 
  labs(x = expression(paste("GO treatment effect (", Delta, "CLR/day)")),
        y = expression(paste("NE treatment effect (", Delta, "CLR/day)"))) + 
  geom_smooth(method = "lm", color = "black") + 
  annotate("text", x=-0.01, y=0.035, label = sprintf("Spearman's rho = %.2f", correlation)) +
  annotate("text", x=-0.01, y=0.03, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE)

# Save
fn <- paste0(figdir, "Correlation_GOvsNE.pdf")
ggsave(filename = fn, plot = p, width = 3.5, height = 3)
```

## ASV timecourse
```{r}
# Define ASV of interest
target_asvs <- "0be7263910d54a834cdbfae947b96ab0"
target_asv_names <- "Anaerovoracaceae"
ASV_genera <- "Anaerovoracaceae"
target_asv_direction = -1
asvs <- data.frame("ASV" = target_asvs, "ASV_name" = target_asv_names, "Direction" = target_asv_direction)

# Load data and normalize (relative abundance)
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/NE_16S/TaxaSummaryASVs.csv")
a <- read.csv(fn) %>% as.data.frame() 
rownames(a) <- a$X; a <- a %>% select(-X) # Wrangling
a <- a / colSums(a)*100 # Normalize to relative abundance (%)
a$Taxa <- rownames(a) # Re-add row names for melting

# Melt
a <- a %>% reshape2::melt()
colnames(a) <- c("ASV", "Sample_ID", "Relative abundance")

# Merge with metadata
fn_meta <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/NE_16S/Metadata_Analysis_withAlpha.csv")
m <- read.csv(fn_meta) %>% as.data.frame()
a <- left_join(a, m, by = "Sample_ID")

# Merge with ASV data
a_plot <- inner_join(a, asvs, by = "ASV")

# Remove patients for which the ASV abundance is 0 for the entire timecourse
a_plot <- a_plot %>%
  group_by(ASV, Patient_ID) %>%
  filter(sum(`Relative abundance`) > 0) %>%
  ungroup()

# Plot
colorscale = c("Enriched" = "blue", "Depleted" = "orange")
a_plot$ASV_name <- factor(a_plot$ASV_name, levels = ASV_genera)
a_plot$Direction <- case_when(a_plot$Direction > 0 ~ "Enriched", TRUE ~ "Depleted")
a_plot$Days_Since_Tx <- case_when(a_plot$Treatment_Cycle == "T1" ~ 0,
                                  a_plot$Treatment_Cycle == "T2" ~ 49,
                                  TRUE ~ 63)
offset = min(a_plot$`Relative abundance`[a_plot$`Relative abundance` > 0])/10
p <- ggplot(a_plot, aes(x = Days_Since_Tx, y = `Relative abundance` + offset, fill = Direction)) + 
  geom_smooth(method = stats::loess, color = "black", se = TRUE, span = 0.9, fullrange = TRUE, level = 0.95) + 
  scale_fill_manual(values = colorscale) + 
  theme_pubr() + 
  facet_wrap(~ASV_name, scales = "free_y", nrow = 2) + 
  scale_y_log10() + 
  xlab("Days since treatment start") + 
  ylab("ASV relative abundance (%)") + 
  theme(legend.position = "none") + 
  theme(strip.text = element_text(face = "italic")) +  # Italicize genera
  theme(strip.text = element_text(size = 10)) + 
  theme(axis.text.y = element_text(size = 10))

# Save
fn <- paste0(figdir, "NE_ASVtime.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## ASV vs time preTA
```{r}
# Genera of interest
ASV_genera <- c("Escherichia-Shigella", "Anaerostipes", "Eubacterium", "Citrobacter") # set in desired order

# Create dataframe with ASVs of interest
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/NE_16S/ASV_d2taxonomy.txt")
v <- read.csv(fn, sep = "\t") %>% as.data.frame()
target_asvs <- v %>% filter(Genus %in% ASV_genera) %>% select(ASV) %>% pull()

# Load data and normalize (relative abundance)
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/NE_16S/TaxaSummaryASVs.csv")
a <- read.csv(fn) %>% as.data.frame() 
rownames(a) <- a$X; a <- a %>% select(-X) # Wrangling
a <- a / colSums(a)*100 # Normalize to relative abundance (%)
a$Taxa <- rownames(a) # Re-add row names for melting

# Melt
a <- a %>% reshape2::melt()
colnames(a) <- c("ASV", "Sample_ID", "Relative abundance")

# Merge with metadata
fn_meta <- paste0(prefix, "ktrepka/GOStudy/ERP143365/Tables/Metadata_Analysis_withAlpha.csv")
m <- read.csv(fn_meta) %>% as.data.frame()
a <- left_join(a, m, by = "Sample_ID")

# Merge with ASV data
a_plot <- a %>% filter(ASV %in% target_asvs)

# Remove patients for which the ASV abundance is 0 for the entire timecourse
a_plot <- a_plot %>%
  group_by(ASV, Patient_ID) %>%
  filter(sum(`Relative abundance`) > 0) %>%
  ungroup()

# Wrangling
colorscale = c("Enriched" = "blue", "Depleted" = "orange")
a_sum <- a_plot %>% group_by(Sample_ID, Patient_ID, Treatment_Cycle) %>% dplyr::summarise(`Relative abundance` = sum(`Relative abundance`)) %>% as.data.frame()
pt_all <- a_sum %>% 
  as.data.frame() %>%
  group_by(Patient_ID) %>%
  dplyr::summarise(n = n()) %>%
  filter(n == 3) %>% 
  select(Patient_ID) %>% 
  pull()
a_sum$Direction = "Enriched"
a_sum$ASV_name = "preTA+ ASVs"
offset = min(a_sum$`Relative abundance`[a_sum$`Relative abundance` > 0], na.rm = TRUE)
a_sum$Treatment_Cycle <- case_when(a_sum$Treatment_Cycle == "T1" ~ "Baseline",
                                   a_sum$Treatment_Cycle == "T2" ~ "Cycle 3",
                                   TRUE ~ "Trough")
a_sum <- a_sum %>% arrange(Patient_ID)

# Plot
a_sum$Strain <- "preTA+ ASVs"
a_sum$PrePost <- case_when(a_sum$Treatment_Cycle == "Baseline" ~ "Pre",
                                   TRUE ~ "Post")
a_sum$PrePost <- factor(a_sum$PrePost, levels = c("Pre", "Post"))
p <- ggplot(a_sum %>% filter(Patient_ID %in% pt_all), aes(x = PrePost, y = `Relative abundance` + offset, fill = Direction)) + 
  geom_boxplot(aes(fill = PrePost), alpha = 0.5, fatten = 2.2, outlier.size = -1) + 
  geom_jitter(width = 0.05, aes(color = PrePost)) + 
  stat_compare_means(label.x = 1.2, label.y = log10(30), method = "wilcox.test", method.args = list(alternative = "greater"),aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  scale_color_manual(values = tx_palette) + 
  scale_fill_manual(values = tx_palette) + 
  theme_pubr() + 
  xlab("") + 
  ylab("preTA+ ASV abundance (%)") + 
  theme(legend.position = "none") + 
  scale_y_log10(limits = c(0.017, 40)) + 
  theme(axis.text.x = element_text(angle = 30, vjust = 1, hjust=1))

# Save
fn <- paste0(figdir, "ASVtimePreTA.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)
```

# Figure 2 - Genes and SIC
```{r}
# Make directory
figdir <- paste0(dir, "Figure2/")
dir.create(figdir)

# Load date data
fn = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/GO_metadata_16S_WithDates.xlsx")
dates <- read_excel(fn) %>% as.data.frame() %>% select(c(Sample_ID, Patient_ID, Treatment_Cycle, Batch_16S, Days_Since_Tx))
dates <- dates %>% filter(!(Patient_ID %in% c("9", "9c")))
dates$Patient_ID[dates$Patient_ID == "9b"] <- "9"
```

## KO Volcano
```{r}
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/KOGroups_nonzero0_WithBaseline.csv")
u <- read.csv(fn) %>% as.data.frame()
u$log2FC <- log2((u$BaselineGroup + u$Value)/u$BaselineGroup)
u$FDR <- p.adjust(u$p.value, method = "BH")

# Set the FDR cutoff value
fdr_cutoff <- 0.2

# Rename
u$Group <- case_when(u$Group == "Early" ~ "Cycle 1",
                     TRUE ~ "Trough")

# Early only
u <- u %>% filter(Group == "Cycle 1")

# Create a volcano plot
p <- ggplot(data = u, aes(x = log2FC, y = -log10(FDR))) +
  geom_point(alpha = 1, shape = 16, aes(color = ifelse(FDR <= fdr_cutoff, ifelse(Value > 0, "Enriched", "Depleted"), "Not Significant")), size = 1.5) +
  theme_pubr() +
  facet_wrap(~Group) + 
  labs(x = expression(paste("Treatment effect (log2FC/baseline)")), y = "-log10(FDR)", color = "Significance") +
  scale_color_manual(values = enriched_palette) +
  theme(legend.position = "none") + 
  xlim(-2, 2.5)

# Save
fn <- paste0(figdir, "Volcano_KO.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)

# Count number of gene families altered
num_early <- u %>% filter(FDR < fdr_cutoff) %>% filter(Group == "Cycle 1") %>% select(GeneFamily) %>% count()
num_late <- u %>% filter(FDR < fdr_cutoff) %>% filter(Group == "Trough") %>% select(GeneFamily) %>% count()
num_total <- u %>% select(GeneFamily) %>% unique() %>% count()
```



## preTA vs time
```{r}
hits <- c("K17723", "K17722")
directions <- rep("Enriched", 2)

fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/genefamilies_unstratified_nonzero0.csv")
d <- fread(fn) %>% filter(KO %in% hits) %>% as.data.frame() %>% select(-V1)

# Melt
a <- d %>% reshape2::melt(by = "GeneFamily")
colnames(a) <- c("GeneFamily", "Sample_ID", "Abundance")
a$Sample_ID <- gsub("_", "-", a$Sample_ID)
a <- a %>% group_by(Sample_ID) %>% 
 dplyr::summarise(Abundance = sum(Abundance)) %>%
  as.data.frame()

# Merge with metadata, direction
a <- left_join(a, dates, by = "Sample_ID") 
a_plot <- a
a_plot$Days_Since_Tx <- as.numeric(a_plot$Days_Since_Tx)
a_plot$Days_Since_Tx <- case_when(a_plot$Days_Since_Tx < 0 ~ 0, TRUE ~ a_plot$Days_Since_Tx)
a_plot <- a_plot %>% filter(Days_Since_Tx <= 8) 
offset = min(a_plot$Abundance[a_plot$Abundance > 0])

# Don't bias towards patients that submitted more samples! Take the average of each patient's samples at each timepoint
a_plot$Group <- case_when(a_plot$Days_Since_Tx <= 1 ~ 0,
                            a_plot$Days_Since_Tx <= 5 ~ 3,
                            TRUE ~ 7)
a_plot$Group <- factor(a_plot$Group)
a_plot <- a_plot %>% group_by(Patient_ID, Group) %>%
  dplyr::summarise(Abundance = mean(Abundance)) %>%
  as.data.frame()
pt_paired <- a_plot %>% group_by(Patient_ID) %>%
  dplyr::summarise(n = n()) %>% 
  filter(n == 3) %>% 
  select(Patient_ID) %>%
  pull()
p <- ggplot(a_plot, aes(x = Group, y = Abundance + offset)) + 
  geom_boxplot(color = "black", alpha = 0.5, aes(group = Group, fill = Group), outlier.size = -1) +
  theme_pubr() + 
  scale_fill_manual(values = c("0" = "lightgrey", "3" = "#4E4EFF", "7" = "#1414FF")) + 
  ylab("preTA abundance (RPKG)") + 
  xlab("CAP duration (days)") + 
  theme(legend.position = "none") +
  scale_y_log10(limits = c(0.005, 6)) + 
  stat_compare_means(method = "wilcox", paired = TRUE, method.args = list(alternative = "less"), comparisons = list(c("0", "3"), c("0", "7")), aes(label = paste0("p = ", after_stat(p.format))), data = dplyr::filter(a_plot, Patient_ID %in% pt_paired)) 

# Save
fn <- paste0(figdir, "KOvstime_preTAOnly.pdf")
ggsave(filename = fn, plot = p, width = 2.1, height = 3)
```

## GO preTA source
```{r}
# Read in data for kegg orthologs (KOI) of interest
kois <- c("K17722", "K17723") # kegg orthologues to plot
nplot = 10 # number of microbes to plot

# Read in data
k <- NULL
for (koi in kois){
  fn_dir <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/")
  fn <- paste0(fn_dir, "KO_stratified_", koi, ".csv")
  k_in <- read.csv(fn) %>% as.data.frame()
  k_in$KO <- koi
  if (is.null(k)){k <- k_in} else {k <- rbind(k, k_in)}
}
k <- tidyr::separate(k, col = GeneFamily, into = c("GeneFamily", "Taxon"), sep = "\\|")

# Combine KOs together
k_orig <- k
k <- k %>% group_by(Sample_ID, Taxon) %>% 
    dplyr::summarise(RPKG = sum(RPKG))

# Get metadata + join
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/Metadata170523.csv")
meta <- read.csv(fn) %>% as.data.frame()
meta$Sample_ID <- gsub("-", "_", meta$Sample_ID)
k <- k %>% left_join(meta, by = "Sample_ID")
k$Taxon[k$Taxon == "unclassified"] <- "Unclassified"

# Fix taxon mapping - these are mis-mapped (there is no Klebsiella, youngae by taxonomy in these samples)
k$Taxon[k$Taxon == "g__Klebsiella.s__Klebsiella_oxytoca"] <- "g__Escherichia.s__Escherichia_coli"
k$Taxon[k$Taxon == "g__Klebsiella.s__Klebsiella_pneumoniae"] <- "g__Escherichia.s__Escherichia_coli"
k$Taxon[k$Taxon == "g__Citrobacter.s__Citrobacter_youngae"] <- "g__Citrobacter.s__Citrobacter_freundii"
k$Taxon[k$Taxon == "g__Vibrio.s__Vibrio_parahaemolyticus"] <- "g__Citrobacter.s__Citrobacter_amalonaticus"

# Wrangling
k_2 <- k 
highest_taxa <- k_2 %>% dplyr::group_by(Taxon) %>% 
  dplyr::summarize(tot = sum(RPKG)) %>% 
  arrange(desc(tot)) %>% head(n = nplot) %>% pull(Taxon)
k_2 <- k_2 %>% mutate(Microbe = ifelse(Taxon %in% highest_taxa, Taxon, "Other"))
k_2$Microbe <- gsub(".*s__", "", k_2$Microbe)
k_2$Microbe <- gsub("_", " ", k_2$Microbe)

# make a metadata file to merge. I need the values to be 0 for every other patient
hm <- k_2 %>% select(Patient_ID, Treatment_Cycle, Taxon, Microbe, RPKG)
m_merge <- meta %>% select(Patient_ID, Treatment_Cycle)
pts <- m_merge$Patient_ID
tx <- m_merge$Treatment_Cycle
n_pt <- length(pts)
n_microbes <- length(unique(hm$Microbe))
microbes <- unique(hm$Microbe)
m_merge <- data.frame("Patient_ID" = rep(pts, n_microbes), "Treatment_Cycle" = rep(tx, n_microbes), "Microbe" = rep(microbes, each = n_pt))
hm <- left_join(m_merge, hm, by = c("Treatment_Cycle", "Patient_ID", "Microbe"))
hm_all <- hm # keep all data
hm$Group <- case_when(hm$Treatment_Cycle %in% c("Baseline", "C1D1") ~ "Baseline",
                      hm$Treatment_Cycle %in% c("C1D3", "C1D7") ~ "Cycle 1",
                      TRUE ~ "Trough")
hm <- hm %>% group_by(Patient_ID, Group, Microbe) %>% dplyr::summarise(RPKG = mean(RPKG)) # take mean of early timepoints
  
# Order
microbe_order <- hm %>% group_by(Microbe) %>% 
  dplyr::summarise(sum = sum(RPKG, na.rm = TRUE)) %>%
  arrange(sum) %>% 
  select(Microbe) %>% 
  pull()
hm <- hm %>% mutate(Microbe = factor(Microbe, levels = microbe_order))
pt_order <- hm %>% group_by(Patient_ID) %>% 
  dplyr::summarise(sum = max(RPKG, na.rm = TRUE)) %>%
  arrange(sum) %>% 
  select(Patient_ID) %>% 
  pull()
hm <- hm %>% mutate(Patient_ID = factor(Patient_ID, levels = rev(pt_order)))

# Plot
p <- ggplot(hm, aes(x = Patient_ID, y = Microbe, fill = RPKG)) + 
  geom_tile(colour="white") + 
  scale_fill_gradient(low = "lightblue", high = "darkblue", na.value = "white") +
  theme_pubr() +
  theme(legend.position = "top") + 
  facet_wrap(~Group, scales = "free_x") + 
  labs(fill = "preTA\n(RPKG)", x = "Patient", y = "") +
  theme(axis.text.x = element_blank()) + 
  theme(legend.title.align=0.5) +
  guides(fill = guide_colourbar(barwidth = 10, barheight = 1)) + 
  theme(axis.text.y = element_text(face = "italic"))  # Italicize

# Save + export
fn <- paste0(figdir, "preTASource.pdf")
ggsave(filename = fn, plot = p, width = 8, height = 3)
```

## preTA+ ASV vs time 
```{r}
# Genera of interest
ASV_genera <- c("Escherichia-Shigella", "Anaerostipes", "Eubacterium", "Citrobacter") # set in desired order

# Create dataframe with ASVs of interest
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/ASVsLinearD50.csv")
v <- read.csv(fn)
target_asvs <- v %>% filter(Genus %in% ASV_genera) %>% select(ASV) %>% pull()
target_asv_names <- v %>% filter(Genus %in% ASV_genera) %>% select(Genus) %>% pull()  
target_asv_direction <- v %>% filter(Genus %in% ASV_genera) %>% select(Value) %>% pull() %>% sign()
asvs <- data.frame("ASV" = target_asvs, "ASV_name" = target_asv_names, "Direction" = target_asv_direction)

# Load data and normalize (relative abundance)
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/TaxaSummaryASVs.csv")
a <- read.csv(fn) %>% as.data.frame() 
rownames(a) <- a$X; a <- a %>% select(-X) # Wrangling
a <- a / colSums(a)*100 # Normalize to relative abundance (%)
a$Taxa <- rownames(a) # Re-add row names for melting

# Melt
a <- a %>% reshape2::melt()
colnames(a) <- c("ASV", "Sample_ID", "Relative abundance")
a$Sample_ID <- gsub("\\.", "-", a$Sample_ID)

# Merge with metadata
a <- left_join(a, dates, by = "Sample_ID")

# Merge with ASV data
a_plot <- inner_join(a, asvs, by = "ASV")

# Remove patients for which the ASV abundance is 0 for the entire timecourse
a_plot <- a_plot %>%
  group_by(ASV, Patient_ID) %>%
  filter(sum(`Relative abundance`) > 0) %>%
  ungroup()

# Wrangling
colorscale = c("Enriched" = "blue", "Depleted" = "orange")
a_plot$ASV_name <- factor(a_plot$ASV_name, levels = ASV_genera)
a_plot$Direction <- case_when(a_plot$Direction > 0 ~ "Enriched", TRUE ~ "Depleted")
a_plot$Days_Since_Tx <- as.numeric(a_plot$Days_Since_Tx)
a_plot$Days_Since_Tx <- case_when(a_plot$Days_Since_Tx < 0 ~ 0, TRUE ~ a_plot$Days_Since_Tx)
a_plot <- a_plot %>% filter(Days_Since_Tx <= 50)

# Sum and plot
a_sum <- a_plot %>% group_by(Sample_ID, Days_Since_Tx) %>% dplyr::summarise(`Relative abundance` = sum(`Relative abundance`)) %>% as.data.frame()
a_sum$Direction = "Enriched"
a_sum$ASV_name = "preTA+ ASVs"
offset = min(a_sum$`Relative abundance`[a_plot$`Relative abundance` > 0], na.rm = TRUE)
p <- ggplot(a_sum, aes(x = Days_Since_Tx, y = `Relative abundance` + offset, fill = Direction)) + 
  geom_smooth(method = "loess", color = "black", se = TRUE) + 
  scale_fill_manual(values = colorscale) + 
  theme_pubr() + 
  facet_wrap(~ASV_name) + 
  xlab("Days since treatment start") + 
  ylab("Relative abundance (%)") + 
  theme(legend.position = "none") +
  scale_y_log10() + 
  xlim(0, 50)

# Save
fn <- paste0(figdir, "ASVtimePreTA.pdf")
ggsave(filename = fn, plot = p, width = 2.5, height = 3)
```

## preTA+ ASV vs ASV gene
```{r}
# Import from above
asvs <- a_plot 
genes <- hm_all

# Wrangling
asvs <- asvs %>% group_by(Sample_ID) %>% dplyr::summarise(Ab = sum(`Relative abundance`)) %>% as.data.frame()
asvs$Sample_ID <- gsub("-", "_", asvs$Sample_ID)
colnames(asvs) <- c("Sample_ID", "Abundance")
genes$Sample_ID <- paste0("GO_Pt", genes$Patient_ID, "_", genes$Treatment_Cycle)
genes <- genes %>% group_by(Sample_ID) %>% dplyr::summarise(RPKG = sum(RPKG, na.rm = TRUE)) %>% as.data.frame()
df <- inner_join(asvs, genes, by = "Sample_ID")

# Get p value
spearman_test <- cor.test(df$Abundance, df$RPKG, method = "spearman")
correlation <- spearman_test$estimate
p_value <- spearman_test$p.value

# Plot
p <- ggplot(df, aes(x = Abundance, y = RPKG)) + 
  geom_point(color = "grey") +
  geom_smooth(method = "lm", color = "black") + 
  theme_pubr() + 
  scale_x_log10(breaks = c(0.5, 5, 50)) + 
  scale_y_log10(breaks = c(0.001, 0.01, 0.1, 1), limits = c(0.001, 12)) +
  xlab("preTA+ ASV abundance (%)") + 
  ylab("preTA abundance (RPKG)") + 
  annotate("text", x=3, y=12, label = sprintf("Spearman's rho = %.2f", correlation)) +
  annotate("text", x=3, y=6, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) 

# Save
fn <- paste0(figdir, "ASVvsPreTA.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## Single PCO
```{r}
rerun_permanova = FALSE
filter = TRUE # TRUE -> only do Baseline vs C1D7

# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/genefamilies_unstratified_nonzero0.csv")
d <- fread(fn) %>% as.data.frame()
rownames(d) <- d$KO; d <- d %>% select(-KO) %>% select(-V1)

# CLR transform
geometric_means <- apply(d, 1, function(row) exp(mean(log(row[row > 0]))))
pseudo_count = min(d[d > 0])/2
clr_data <- log((d + pseudo_count) / geometric_means)

# Omit NA values
clr_data <- na.omit(clr_data)

# Remove zero-variance columns
clr_data_t <- t(clr_data)
clr_data_t <- clr_data_t[ , which(apply(clr_data_t, 2, var) != 0)]
d_t <- t(d)
d_t <- d_t[ , which(apply(d_t, 2, var) != 0)]

# Load metadata
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/Metadata170523.csv")
meta <- read.csv(fn) %>% as.data.frame()
meta$Sample_ID <- gsub("-", "_", meta$Sample_ID)
meta <- meta %>% filter(Sample_ID %in% rownames(clr_data_t) & Sample_ID %in% rownames(d_t)) # only keep samples that we have sequencing data for
clr_data_t <- clr_data_t[meta$Sample_ID,] # and vice versa for metadata matching
d_t <- d_t[meta$Sample_ID,]

# PERMANOVA testing 
meta_perm <- meta 
if (filter){
  meta_row <- meta[meta$Sample_ID == "GO_Pt6_C1D1",]
  meta_row$Treatment_Cycle <- "Baseline"  # Establish Patient 6 baseline (only patient that has C1D1, but no BL)
  meta_perm <- rbind(meta_perm, meta_row)
  meta_perm <- meta_perm %>% filter(Treatment_Cycle %in% c("Baseline", "C1D7"))
}
row_order <- meta_perm %>% select(Sample_ID) %>% pull(); clr_data_ordered <-clr_data_t[row_order,]; d_ordered <- d_t[row_order,] # arrange
if(sum(meta_perm$Sample_ID == rownames(clr_data_ordered)) + sum(meta_perm$Sample_ID == rownames(d_ordered)) != 2*length(meta_perm$Sample_ID)){ # check that they match
    print("PERMANVA order wrong!")
}
if (rerun_permanova){
  dist_clr <- suppressWarnings(vegdist(clr_data_ordered, method="euclidean", na.rm = TRUE, binary = TRUE)) # CLR-Euclidean
  dist_bray <- vegdist(d_ordered, method="bray", na.rm = TRUE) # Bray
  set.seed(42)
  res_bray <- adonis2(dist_bray ~ Patient_ID + TreatmentCyclev1, data = meta_perm, permutations=1000, strata = meta_perm$Patient_ID, by = "margin")
  res_clr <- adonis2(dist_clr ~ Patient_ID + TreatmentCyclev1, data = meta_perm, permutations=1000, strata = meta_perm$Patient_ID, by = "margin")
}

# Perform PCA
method = "euclidean" # "bray" or "euclidean"
if (method == "bray"){
  data = d_ordered
  axis = 8
  if (filter){axis = 9}
  pca_results <- prcomp(data, center = TRUE)
} else if (method == "euclidean"){
  data = clr_data_ordered
  axis = 6
  pca_results <- prcomp(data, center = TRUE)
}
eigs <- pca_results$sdev^2
eigs <- round(eigs/sum(eigs) * 100, digits = 1)

# Filter to only specific samples for plotting
if (filter){
  meta_row <- meta[meta$Sample_ID == "GO_Pt6_C1D1",]
  meta_row$Treatment_Cycle <- "Baseline"  # Establish Patient 6 baseline (only patient that has C1D1, but no BL)
  meta <- rbind(meta, meta_row)
  meta <- meta %>% filter(Treatment_Cycle %in% c("Baseline", "C1D7"))
  pt_keep <- meta %>% group_by(Patient_ID) %>% dplyr::summarise(n = n()) %>% filter(n == 2) %>% select(Patient_ID) %>% pull()
  meta <- meta %>% filter(Patient_ID %in% pt_keep)
} else {
  meta_row <- meta[meta$Sample_ID == "GO_Pt6_C1D1",]
  meta_row$Treatment_Cycle <- "Baseline"  # Establish Patient 6 baseline (only patient that has C1D1, but no BL)
  meta <- rbind(meta, meta_row)
  meta <- meta %>% filter(Treatment_Cycle != "C1D1")
}
pca_data <- as.data.frame(pca_results$x)
pca_data$Sample_ID <- rownames(pca_data)
pca_merged <- inner_join(pca_data, meta, by = "Sample_ID")

# Axis subject plot
pt_order <- pca_merged %>% filter(Treatment_Cycle == "Baseline") %>% arrange(.data[[paste0("PC", axis)]]) %>% select(Patient_ID) %>% pull()
data2 <- pca_merged
data2$Treatment_Cycle[data2$Treatment_Cycle == "Baseline"] <- "BL"
data2$Patient_ID <- factor(data2$Patient_ID, levels = pt_order)
p <- ggplot(data2 %>% arrange(desc(Treatment_Cycle)), aes(x = .data[[paste0("PC", axis)]], y = Patient_ID)) +
  geom_point(aes(color = Treatment_Cycle), size = 1.5) +
  geom_line(aes(group = Patient_ID), color = "grey", linewidth = 0.5) + 
  theme_pubr() +
  labs(x = paste0("PCo", axis, " (", eigs[axis], "% Variation)"),
       y = "Patient",
       color = "Timepoint") +
  theme(legend.position = c(0.2, 0.7)) + 
  theme(axis.text.y = element_blank()) +
  scale_color_manual(values = timepoint_palette) +
  theme(legend.title=element_blank()) +  
  annotate("text", x = -150, y = 31.3, label = "Treatment:~italic(p) == 0.02 * ',' ~ R^2 == 0.01", size = 3.5, hjust = 0, parse = TRUE) +
  annotate("text", x = -150, y = 29.3, label = "Subject:~italic(p) == 0.02 * ',' ~ R^2 == 0.76", size = 3.5, hjust = 0, parse = TRUE)

# Save
fn <- paste0(figdir, "PCAKO_PCO", axis, ".pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## KO GSEA Early

Current approach: Filter to genes with FDR < 0.2 that are enriched early, perform overrepresentation (GSEA) analysis. 

Another approach gives the same result: Filter to genes with nominal p < 0.01 that are enriched early, perform overrepresentation (GSEA) analysis. 

```{r}
fdrcut = 0.2 # FDR cutoff for genes to include in analysis. Default = 0.2
group = "Early" # "Early" or "Late"
trunc = 0
pcut = 0.1 # p value cutoff for plotting
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/KOGroups_nonzero", trunc, "_WithBaseline.csv")
gfdat <- read.csv(fn) %>% as.data.frame() %>% dplyr::rename(KO = GeneFamily)
gfdat$FDR <- p.adjust(gfdat$p.value, method = "BH")

# Only doing for early enriched genes right now!
genes_up <- gfdat %>% filter(FDR < fdrcut) %>% 
  filter(Group == group) %>% 
  filter(Value > 0) %>% 
  select("KO") %>% pull() 
allGenesVector <- gfdat %>% filter(Group == group) %>% select("KO") %>% pull()

genes_up <- genes_up[!is.na(genes_up)]
allGenesVector <- allGenesVector[!is.na(allGenesVector)]

k_up <- enrichKEGG(gene         = genes_up,
                   organism     = 'ko',
                   universe = allGenesVector,
                   pvalueCutoff = fdrcut,
                   minGSSize = 10,
                   qvalueCutoff = 0.2)
k_up <- k_up@result

# Plot 
k_plot <- k_up %>% filter(pvalue < pcut) # Significant hits only
k_plot$GeneRatio <- apply(k_plot %>% select(GeneRatio), c(1,2), function(x) eval(parse(text = x))) 
k_plot$BgRatio <- apply(k_plot %>% select(BgRatio), c(1,2), function(x) eval(parse(text = x))) 
glob <- c("ko01100", "ko01120", "ko01110", "ko01200", "ko01210", "ko01212", "ko01230", "ko01232", "ko01250", "ko01240", "ko01220") # Remove global/overview kegg pathways b/c not meaningful: https://www.genome.jp/kegg/pathway.html#global. 
glob <- c(glob, "ko03010", "ko00195", "ko00710", "ko04626", "ko04112", "ko00643", "ko05134") # remove ribosome, photosynthesis, Caulobacter
k_plot <- k_plot %>% filter(!(ID %in% glob))

# Add variable to plot
k_plot_early <- k_plot

# Order based on GeneRatio
k_plot_early_order <- k_plot_early %>%
  group_by(Description) %>%
  arrange(Count) %>%
  mutate(DescriptionOrdered = reorder(Description, Count)) %>%
  select(DescriptionOrdered) %>% pull()
k_plot_early$Description <- factor(k_plot_early$Description, levels = k_plot_early_order)

# Plot
k_plot_early$Group <- "Cycle 1"
p <- ggplot(k_plot_early, aes(x = Count, y = Description, fill = log10(Count))) + 
  geom_bar(position="stack", stat="identity") + 
  theme_pubr() + 
  ylab("") + 
  xlab("Enriched KOs (#)") + 
  theme(legend.position = "none") + #c(0.6, 0.2)) + 
  facet_wrap(~Group) + 
  scale_y_discrete(labels=c("Pyrimidine metabolism"=expression(bold(Pyrimidine~metabolism)),
                            "beta-Alanine metabolism"=expression(bold(beta-Alanine~metabolism)),
                            "Pantothenate and CoA biosynthesis"=expression(bold(Pantothenate~and~CoA~biosynthesis)),
                            parse=TRUE)) +
  scale_x_continuous(breaks = c(0, 5, 10)) 

# Save
fn <- paste0(figdir, "KOGSEA_Early_Up.pdf")
ggsave(filename = fn, plot = p, width = 5, height = 3)
```

## SIC preTA source
```{r}
# Read in data for kegg orthologs (KOI) of interest
kois <- c("K17722", "K17723") # kegg orthologues to plot
nplot = 4 # number of microbes to plot

# Read in data
k <- NULL
for (koi in kois){
  fn_dir <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SIC_Data/")
  fn <- paste0(fn_dir, "KO_stratified_", koi, ".csv")
  k_in <- read.csv(fn) %>% as.data.frame()
  k_in$KO <- koi
  if (is.null(k)){k <- k_in} else {k <- rbind(k, k_in)}
}
k <- tidyr::separate(k, col = GeneFamily, into = c("GeneFamily", "Taxon"), sep = "\\|")
k$Sample_ID <- gsub("_S.*", "", k$Sample_ID)
  
# Combine KOs together
k_orig <- k
k <- k %>% group_by(Sample_ID, Taxon) %>% 
    dplyr::summarise(RPKG = sum(RPKG))

# Get metadata + join
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SIC_Data/PreTAvsMetabolism.csv")
meta <- read.csv(fn) %>% as.data.frame() %>% select(-RPKG)
meta$Sample_ID <- meta$MetaphlanID
meta$Patient_ID <- meta$Patient
k <- k %>% left_join(meta, by = "Sample_ID")
k$Taxon[k$Taxon == "unclassified"] <- "Unclassified"

# Fix taxon mapping - these are mis-mapped (there is no Klebsiella, C youngae by taxonomy in these samples)
k$Taxon[k$Taxon == "g__Klebsiella.s__Klebsiella_oxytoca"] <- "g__Escherichia.s__Escherichia_coli"
k$Taxon[k$Taxon == "g__Klebsiella.s__Klebsiella_pneumoniae"] <- "g__Escherichia.s__Escherichia_coli"
k$Taxon[k$Taxon == "g__Citrobacter.s__Citrobacter_youngae"] <- "g__Citrobacter.s__Citrobacter_freundii"
k$Taxon[k$Taxon == "g__Vibrio.s__Vibrio_parahaemolyticus"] <- "g__Citrobacter.s__Citrobacter_amalonaticus"

# Wrangling
k_2 <- k 
k_2 <- k_2 %>% filter(grepl("Pa1", Sample_ID)) %>% 
  filter(grepl("Tx5FU", Sample_ID)) %>% 
  filter(!(grepl("Pt31", Sample_ID)))
k_2 <- k_2 %>% filter(RPKG > 0)
highest_taxa <- k_2 %>% dplyr::group_by(Taxon) %>% 
  dplyr::summarize(tot = sum(RPKG)) %>% 
  arrange(desc(tot)) %>% head(n = nplot) %>% pull(Taxon) 
#k_2 <- k_2 %>% mutate(Microbe = ifelse(Taxon %in% highest_taxa, Taxon, "Other"))
k_2 <- k_2 %>% mutate(Microbe = Taxon)
k_2$Microbe <- gsub(".*s__", "", k_2$Microbe)
k_2$Microbe <- gsub("_", " ", k_2$Microbe)

# make a metadata file to merge. I need the values to be 0 for every other patient
hm <- k_2 %>% select(Sample_ID, Taxon, Microbe, RPKG)
m_merge <- meta %>% select(Sample_ID)
pts <- m_merge$Sample_ID
n_pt <- length(pts)
n_microbes <- length(unique(hm$Microbe))
microbes <- unique(hm$Microbe)
m_merge <- data.frame("Sample_ID" = rep(pts, n_microbes), "Microbe" = rep(microbes, each = n_pt))
hm <- left_join(m_merge, hm, by = c("Sample_ID", "Microbe"))
hm_all <- hm # keep all data
  
# Manualy rename
hm$Microbe <- case_when(hm$Microbe == "Escherichia coli" ~ "E. coli",
                        hm$Microbe == "Escherichia fergusonii" ~ "E. fergusonii",
                        hm$Microbe == "Citrobacter amalonaticus" ~ "C. amalonaticus",
                        hm$Microbe == "Unclassified" ~ "Unclassified",
                        TRUE ~ NA)

# Order
microbe_order <- hm %>% group_by(Microbe) %>% 
  dplyr::summarise(sum = sum(RPKG, na.rm = TRUE)) %>%
  arrange(sum) %>% 
  select(Microbe) %>% 
  pull() %>% rev()
hm <- hm %>% mutate(Microbe = factor(Microbe, levels = rev(microbe_order)))
pt_order <- hm %>% group_by(Sample_ID) %>% 
  dplyr::summarise(sum = sum(RPKG, na.rm = TRUE)) %>%
  arrange(sum) %>% 
  select(Sample_ID) %>% 
  pull() %>% rev()
hm <- hm %>% mutate(Sample_ID = factor(Sample_ID, levels = pt_order))

# Plot
hm$Donor <- case_when(grepl("Pt19", hm$Sample_ID) ~ "Donor\n19",
                      grepl("Pt21", hm$Sample_ID) ~ "Donor\n21",
                      grepl("Pt50", hm$Sample_ID) ~ "Donor\n50",
                      TRUE ~ NA)
p <- ggplot(hm, aes(x = Sample_ID, y = Microbe, fill = RPKG)) + 
  geom_tile(colour="white") + 
  scale_fill_gradient(low = "lightblue", high = "darkblue", na.value = "white") +
  theme_pubr() +
  labs(fill = "preTA\n(RPKG)", y = "") +
  xlab(expression(paste(italic("ex vivo"), " community"))) + 
  theme(axis.text.x = element_blank(), legend.position = "top") + 
 theme(legend.title.align=0.5) +
  facet_wrap(~Donor, scales = "free_x") + 
  guides(fill = guide_colourbar(barwidth = 5, barheight = 1)) + 
  theme(strip.text.x = element_text(size = 8)) + 
  theme(axis.text.y = element_text(face = "italic"))  # Italicize

# Save
fn <- paste0(figdir, "SIC_preTASource.pdf")
ggsave(filename = fn, plot = p, width = 2.9, height = 2.5)
```

## SIC 5FU vs preTA
```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SIC_Data/PreTAvsMetabolism.csv")
preta <- read.csv(fn)
preta$Patient <- factor(preta$Patient)

# Stats
spearman_test <- cor.test(preta$RPKG, preta$CalculatedConcentration/1000, method = "spearman")
correlation <- spearman_test$estimate
p_value <- spearman_test$p.value

# Plot
p <- ggplot(preta, aes(x = RPKG, y = CalculatedConcentration/1000)) + 
  theme_pubr() +
  xlab(expression(paste(italic("ex vivo"), " preTA (RPKG)"))) + 
  ylab("5-FU remaining (ug/mL)") + 
  geom_smooth(method = "lm", color = "black")  + 
    geom_point(aes(color = Patient)) +
  annotate("text", x=2.5, y=70, label = sprintf("Spearman's rho = %.2f", correlation)) +
  annotate("text", x=2.5, y=60, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) +
  scale_color_manual(values = donor_colors) + 
  labs(color = "Donor") + 
  theme(legend.position = "top")
  
# Save
fn <- paste0(figdir, "SIC_preTAvs5FU.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

# Figure S4 - MGS Supplement
```{r}
# Make directory
figdir <- paste0(dir, "FigureS4/")
dir.create(figdir)

# Load date data
fn = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/GO_metadata_16S_WithDates.xlsx")
dates <- read_excel(fn) %>% as.data.frame() %>% select(c(Sample_ID, Patient_ID, Treatment_Cycle, Batch_16S, Days_Since_Tx))
dates <- dates %>% filter(!(Patient_ID %in% c("9", "9c")))
dates$Patient_ID[dates$Patient_ID == "9b"] <- "9"
```

## Genes C1D1 dist
```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/genefamilies_unstratified_nonzero0.csv")
d <- fread(fn) %>% as.data.frame()
rownames(d) <- d$KO; d <- d %>% select(-KO) %>% select(-V1)

# CLR transform
geometric_means <- apply(d, 1, function(row) exp(mean(log(row[row > 0]))))
pseudo_count = min(d[d > 0])/2
clr_data <- log((d + pseudo_count) / geometric_means)

# Load metadata
fn <- paste0(prefix, "ktrepka/GOStudy/GOPaper/Metadata/Metadata170523.csv")
meta <- read.csv(fn) %>% as.data.frame()
meta$Sample_ID <- gsub("-", "_", meta$Sample_ID)

# Wrangle
id_keep <- intersect(meta$Sample_ID, colnames(clr_data))
clr_data <- clr_data[,id_keep]
meta <- meta %>% filter(Sample_ID %in% id_keep)

# Distance matrix
distance_matrix <- dist(t(clr_data), method = "euclidean")
distance_matrix_full <- as.matrix(distance_matrix)

# Melt
rownames(distance_matrix_full) <- colnames(distance_matrix_full) <- rownames(t(clr_data))
melted_matrix <- reshape2::melt(distance_matrix_full)
colnames(melted_matrix) <- c("Row", "Column", "Distance") 
mat <- melted_matrix %>% as.data.frame()

# Wrangling
mat$Pt1 <- str_extract(mat$Row, "Pt\\d+")
mat$Time1 <- case_when(grepl("Baseline", mat$Row) ~ "Baseline",
                       grepl("C1D1", mat$Row) ~ "C1D1",
                       grepl("C1D3", mat$Row) ~ "C1D3",
                       grepl("C1D7", mat$Row) ~ "C1D7",
                       grepl("C2D1", mat$Row) ~ "C2D1",
                       grepl("C3D1", mat$Row) ~ "C3D1",
                       grepl("Final", mat$Row) ~ "Final",
                       TRUE ~ NA)
mat$Pt2 <- str_extract(mat$Column, "Pt\\d+")
mat$Time2 <- case_when(grepl("Baseline", mat$Column) ~ "Baseline",
                       grepl("C1D1", mat$Column) ~ "C1D1",
                       grepl("C1D3", mat$Column) ~ "C1D3",
                       grepl("C1D7", mat$Column) ~ "C1D7",
                       grepl("C2D1", mat$Column) ~ "C2D1",
                       grepl("C3D1", mat$Column) ~ "C3D1",
                       grepl("Final", mat$Column) ~ "Final",
                       TRUE ~ NA)

# Filter
matched <- mat %>% filter(Pt1 == Pt2)
bl <- matched %>% filter(Time1 == "C1D1") %>% filter(Time2 != "Baseline") %>% filter(Time2 != "C1D1")

## Wrangling
b <- bl %>% dplyr::rename(value = Distance)
b$Patient_ID <- b$Pt1 # Pt1 == Pt2, so this is identical
b$Patient_ID <- gsub("Pt", "", b$Patient_ID) # Fix naming convention
b$Treatment_Cycle <- b$Time2 # Set Treatment Cycle
b$Patient_ID[b$Patient_ID == "9b"] <- "9"
b <- b %>% select(Patient_ID, Treatment_Cycle, value)

# Load metadata
dates_b <- dates %>% select(Treatment_Cycle, Patient_ID, Days_Since_Tx) 

# Find the date of the baseline day relative to treatment start 
baseline_day <- dates_b %>% group_by(Patient_ID) %>% dplyr::summarise(BaseDay = min(Days_Since_Tx)) %>% as.data.frame()
dates_b <- left_join(dates_b, baseline_day, by = "Patient_ID")

# Data wrangling
dates_b$Days_Since_Tx[dates_b$Days_Since_Tx == "No stool"] <- "No date"
dates_b <- dates_b %>% filter(!(Patient_ID %in% c("24", "39"))) # Remove patients with no timecourse
dates_b$Days_Since_Tx <- as.numeric(dates_b$Days_Since_Tx) # Numeric
dates_b$BaseDay <- as.numeric(dates_b$BaseDay) # Numeric

# Days elapsed
dates_b$Days_Elapsed <- dates_b$Days_Since_Tx - dates_b$BaseDay

# Join dataframes
b <- left_join(b, dates_b, by = c("Patient_ID", "Treatment_Cycle")) %>% select(-BaseDay)

# Add C1D1 0 distance
c1d1 <- b
c1d1$Treatment_Cycle = "C1D1"
c1d1$value = 0
c1d1$Days_Since_Tx = 0
c1d1$Days_Elapsed = 0
c1d1 <- c1d1 %>% select(Patient_ID, Treatment_Cycle, value, Days_Since_Tx, Days_Elapsed) %>% distinct() # remove duplicate rows
b <- rbind(b, c1d1)

# Plot
p_beta <- ggplot(b %>% filter(Days_Since_Tx <= 50) , aes(x = Days_Since_Tx, y = value)) + 
  geom_smooth(method = "loess", span = 0.6, color = "black", fill = "orange", level = 0.68) + 
  theme_pubr() + 
  theme(legend.position = "none") +
  xlab("Days since treatment start") + 
  ylab("Distance from C1D1 \n (CLR-Euclidean)") + 
  xlim(2, 50) + 
  annotate("text", x=30, y= 210, label = "p < 0.0001")  

# Save
fn <- paste0(figdir, "BetaDistanceKO.pdf")
ggsave(filename = fn, plot = p_beta, width = 3, height = 3)
```

## KO Hits vs time
```{r}
hits <- c("K17723", "K17722", "K01464", "K00756", "K00761", "K01493")
directions <- rep("Enriched", 6)

hit_df <- data.frame(GeneFamily = hits, Direction = directions)

fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/genefamilies_unstratified_nonzero0.csv")
d <- fread(fn) %>% filter(KO %in% hits) %>% as.data.frame() %>% select(-V1)

# Melt
a <- d %>% reshape2::melt(by = "GeneFamily")
colnames(a) <- c("GeneFamily", "Sample_ID", "Abundance")
a$Sample_ID <- gsub("_", "-", a$Sample_ID)

# Merge with metadata, direction
a <- left_join(a, dates, by = "Sample_ID") %>% left_join(hit_df)

# Remove patients for which the ASV abundance is 0 for the entire timecourse
a_plot <- a %>%
  group_by(GeneFamily, Patient_ID) %>%
  filter(sum(Abundance) > 0) %>%
  ungroup()

# Wrangling
colorscale = c("Enriched" = "blue", "Depleted" = "orange")
a_plot$Days_Since_Tx <- as.numeric(a_plot$Days_Since_Tx)
a_plot$Days_Since_Tx <- case_when(a_plot$Days_Since_Tx < 0 ~ 0, TRUE ~ a_plot$Days_Since_Tx)
a_plot <- a_plot %>% filter(Days_Since_Tx <= 50) 
a_plot$GeneFamily <- gsub("UniRef90_", "", a_plot$GeneFamily)
a_plot$GeneFamily <- factor(a_plot$GeneFamily, levels = gsub("UniRef90_", "", hits))
offset = min(a_plot$Abundance[a_plot$Abundance > 0])

# Manually curate annotations
a_plot$GeneAnnotations <- case_when(a_plot$GeneFamily == "K17723" ~ "preA (K17723)",
                               a_plot$GeneFamily == "K17722" ~ "preT (K17722)",
                               a_plot$GeneFamily == "K00756" ~ "pdp (K00756)",
                               a_plot$GeneFamily == "K00761" ~ "upp (K00761)",
                               a_plot$GeneFamily == "K01464" ~ "hydA (K01464)",
                               a_plot$GeneFamily == "K01493" ~ "comEB (K01493)",
                               TRUE ~ NA) 
a_plot$GeneAnnotations <- factor(a_plot$GeneAnnotations, levels = a_plot %>% arrange(GeneFamily) %>% select(GeneAnnotations) %>% unique() %>% pull())

# Plot
p <- ggplot(a_plot, aes(x = Days_Since_Tx, y = Abundance + offset, fill = Direction)) + 
  #geom_point(color = "black", alpha = 0.2) +
  geom_smooth(method = "loess", color = "black", se = TRUE, level = 0.68) + 
  scale_fill_manual(values = colorscale) + 
  theme_pubr() + 
  facet_wrap(~GeneAnnotations, scales = "free_y", nrow = 2) + 
  scale_y_log10() + 
  ylab("Abundance (RPKG)") + 
  xlab("Days since treatment start") + 
  theme(legend.position = "none")

# Save
fn <- paste0(figdir, "KOvstime.pdf")
ggsave(filename = fn, plot = p, width = 6, height = 3)
```

## EC Volcano 
```{r}
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/EC50Groups_nonzero50_WithBaseline.csv")
u <- read.csv(fn) %>% as.data.frame()
u$log2FC <- log2((u$BaselineGroup + u$Value)/u$BaselineGroup)

# Set the FDR cutoff value
fdr_cutoff <- 0.2

# Create a volcano plot
u_filt <- u %>% filter(Group == "Early"); u_filt$Group <- "Cycle 1" # only plot cycle 1, rename facet here
p <- ggplot(data = u_filt, aes(x = log2FC, y = -log10(FDR))) +
  geom_point(aes(color = ifelse(FDR <= fdr_cutoff, ifelse(Value > 0, "Enriched", "Depleted"), "Not Significant"))) +
  theme_pubr() +
  facet_wrap(~Group) + 
  labs(x = expression(paste("Treatment effect (log2FC/baseline)")), y = "-log10(FDR)", color = "Significance") +
  scale_color_manual(values = enriched_palette) +
  theme(legend.position = "none") + 
  xlim(-2, 2.5)

# Save
fn <- paste0(figdir, "ECVolcano.pdf")
ggsave(filename = fn, plot = p, width = 3.5, height = 3)
```

## UR90 Volcano
```{r}
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/UniRef90Groups_nonzero50_WithBaseline.csv")
u <- read.csv(fn) %>% as.data.frame()
u$log2FC <- log2((u$BaselineGroup + u$Value)/u$BaselineGroup)

# Set the FDR cutoff value
fdr_cutoff <- 0.2

# Only Early
u_filt <- u %>% filter(Group == "Early")
u_filt$Group = "Cycle 1"
p <- ggplot(data = u_filt, aes(x = log2FC, y = -log10(p.value))) +
  geom_point(alpha = 0.5, shape = 16, aes(color = ifelse(FDR <= fdr_cutoff, ifelse(Value > 0, "Enriched", "Depleted"), "Not Significant")), size = 1) +
  theme_pubr() +
  facet_wrap(~Group) + 
  labs(x = expression(paste("Treatment effect (log2FC/baseline)")), y = "-log10(p)", color = "Significance") +
  scale_color_manual(values = enriched_palette) +
  theme(legend.position = "none") + 
  xlim(-2, 2.5)

# Save
fn <- paste0(figdir, "Volcano_UniRef_Cycle1.pdf")
ggsave(filename = fn, plot = p, width = 3.5, height = 3)
fn <- paste0(figdir, "Volcano_UniRef_Cycle1.jpg")
ggsave(filename = fn, plot = p, width = 3.5, height = 3)

# Count number of gene families altered
num_early <- u %>% filter(FDR < fdr_cutoff) %>% filter(Group == "Early") %>% select(GeneFamily) %>% count()
num_late <- u %>% filter(FDR < fdr_cutoff) %>% filter(Group == "Late") %>% select(GeneFamily) %>% count()
num_total <- u %>% select(GeneFamily) %>% unique() %>% count()
```

## UR90 GSEA Early
```{r}
trunc = 50 # trunc = 50 is the current optimum
generatiocut = 0 # arbitrary. Set to 0 for early, 0.022 for late to get ~20 hits
fdrcut = 0.2 # FDR cutoff for gene enrichment analysis
fccut = 0 # fold-change cutoff for gene enrichment analysis
pcut = 0.2 # p value cutoff for plotting
group = "Early" # "Early" or "Late"
prefix = "/mnt/tank/labmainshare/qb3share/"
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/UniRef90Groups_nonzero", trunc, "_WithBaseline.csv")
gfdat <- read.csv(fn) %>% as.data.frame()

# Read in mapping file
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/map_genefamilies_unstratified_to_ko.csv")
map <- fread(fn, colClasses=rep('character', 3))
map <- map[2:length(map$V1), ] %>% select(c("V2", "V3"))
colnames(map) <- c("KO", "GeneFamily")

# Add column for KO + FC
gfdat <- left_join(gfdat, map, by = "GeneFamily") %>% filter(!is.na(KO))
gfdat$FC <- gfdat$Value/gfdat$BaselineGroup

genes_up <- gfdat %>% filter(FDR < fdrcut) %>% 
  filter(Group == group) %>% 
  filter(FC > fccut) %>% 
  select("KO") %>% pull() 
genes_down <- gfdat %>% filter(FDR < fdrcut) %>% 
  filter(Group == group) %>% 
  filter(FC < -fccut) %>% 
  select("KO") %>% pull() 
allGenesVector <- gfdat %>% filter(Group == group) %>% select("KO") %>% pull()

genes_up <- genes_up[!is.na(genes_up)]
genes_down <- genes_down[!is.na(genes_down)]

allGenesVector <- allGenesVector[!is.na(allGenesVector)]

k_up <- enrichKEGG(gene         = genes_up,
                   organism     = 'ko',
                   universe = allGenesVector,
                   pvalueCutoff = fdrcut,
                   minGSSize = 10,
                   qvalueCutoff = 1)
k_up <- k_up@result

k_down <- enrichKEGG(gene         = genes_down,
                     organism     = 'ko',
                     universe = allGenesVector,
                     pvalueCutoff = fdrcut,
                     minGSSize = 10,
                     qvalueCutoff = 1)
k_down <- k_down@result

# Plot 
k_up$Regulation = "Up"
k_down$Regulation = "Down"
k <- rbind(k_up, k_down)
k_plot <- k %>% 
  filter(pvalue < pcut) # Significant hits only
k_plot_orig <- k_plot # Save original for investigation
k_plot$GeneRatio <- apply(k_plot %>% select(GeneRatio), c(1,2), function(x) eval(parse(text = x)))
k_plot$BgRatio <- apply(k_plot %>% select(BgRatio), c(1,2), function(x) eval(parse(text = x))) 
glob <- c("ko01100", "ko01120", "ko01110", "ko01200", "ko01210", "ko01212", "ko01230", "ko01232", "ko01250", "ko01240", "ko01220") # Remove global/overview kegg pathways b/c not meaningful: https://www.genome.jp/kegg/pathway.html#global. 
glob <- c(glob, "ko03010", "ko00195", "ko00710", "ko04626", "ko04112", "ko00643", "ko05134") # remove ribosome, photosynthesis, Caulobacter
k_plot <- k_plot %>% filter(!(ID %in% glob))
k_plot <- k_plot %>% filter(GeneRatio > generatiocut) # Manageable number for plotting 

# Early
k_plot_early <- k_plot %>% filter(Regulation == "Up"); k_plot_early$Group <- "Early"

# Order descriptions
k_plot_early_order <- k_plot_early %>%
  group_by(Description) %>%
  dplyr::summarize(TotalGeneRatio = sum(GeneRatio)) %>%
  arrange(TotalGeneRatio) %>%
  mutate(DescriptionOrdered = reorder(Description, TotalGeneRatio)) %>%
  select(DescriptionOrdered) %>% pull()

k_plot_early$Description <- factor(k_plot_early$Description, levels = k_plot_early_order)

# Plot
p <- ggplot(k_plot_early, aes(x = Count, y = Description, fill = log10(Count))) + 
  geom_bar(position="stack", stat="identity") + 
  theme_pubr() + 
  ylab("") + 
  xlab("Enriched UniRef90 Genes (#)") + 
  theme(legend.position = "none") +
  scale_y_discrete(labels=c("Pyrimidine metabolism"=expression(bold(Pyrimidine~metabolism)),
                            "beta-Alanine metabolism"=expression(bold(beta-Alanine~metabolism)),
                            "One carbon pool by folate"=expression(bold(One~carbon~pool~by~folate)),
                            "Pantothenate and CoA biosynthesis"=expression(bold(Pantothenate~and~CoA~biosynthesis)),
                            parse=TRUE)) +
  scale_x_continuous(breaks = c(0, 10, 20)) 

# Plot, save
fn <- paste0(figdir, "UniRefGSEA_Early_Up.pdf")
ggsave(filename = fn, plot = p, width = 6, height = 3)
```

## Enzyme class timecourse
Enzyme classes of interest: 
preTA: 1.3.1.1
upp: 2.4.2.9
pyrimidine phosphorylase: 2.4.2.2
dihydropyrimidase: 3.5.2.2
comEB: 3.5.4.1

```{r}
# Hit df
hits <- c("1.3.1.1", "1.3.1.2", "3.5.2.2", "2.4.2.2", "2.4.2.9", "3.5.4.12")
directions <- c("Enriched", "Enriched", "Enriched", "Enriched", "Enriched", "Enriched")
hit_df <- data.frame(GeneFamily = hits, Direction = directions)

# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/genefamilies_unstratified_nonzero20_TreatmentCyclev0EC.csv")
t <- fread(fn) %>% as.data.frame()

# Select enzyme classes of interest
d <- t %>% filter(EC %in% hits) 
d$GeneFamily <- d$EC; d <- d %>% select(-EC)

# Melt
a <- d %>% reshape2::melt(by = "GeneFamily")
colnames(a) <- c("GeneFamily", "Sample_ID", "Abundance")
a$Sample_ID <- gsub("_", "-", a$Sample_ID)

# Merge with metadata, direction
a <- left_join(a, dates, by = "Sample_ID") %>% left_join(hit_df)

# Remove patients for which the ASV abundance is 0 for the entire timecourse
a_plot <- a %>%
  group_by(GeneFamily, Patient_ID) %>%
  filter(sum(Abundance) > 0) %>%
  ungroup()

# Wrangling
colorscale = c("Enriched" = "blue", "Depleted" = "orange")
a_plot$Days_Since_Tx <- as.numeric(a_plot$Days_Since_Tx)
a_plot$Days_Since_Tx <- case_when(a_plot$Days_Since_Tx < 0 ~ 0, TRUE ~ a_plot$Days_Since_Tx)
a_plot <- a_plot %>% filter(Days_Since_Tx < 50) 
a_plot$GeneFamily <- gsub("UniRef90_", "", a_plot$GeneFamily)
a_plot$GeneFamily <- factor(a_plot$GeneFamily, levels = gsub("UniRef90_", "", hits))
offset = min(a_plot$Abundance[a_plot$Abundance > 0])

# Manually curate annotations
a_plot$GeneAnnotations <- case_when(a_plot$GeneFamily == "1.3.1.1" ~ "preTA (EC 1.3.1.1)",
                                    a_plot$GeneFamily == "1.3.1.2" ~ "preTA (EC 1.3.1.2)",
                               a_plot$GeneFamily == "2.4.2.2" ~ "pdp (EC 2.4.2.2)",
                               a_plot$GeneFamily == "2.4.2.9" ~ "upp (EC 2.4.2.9)",
                               a_plot$GeneFamily == "3.5.2.2" ~ "hydA (EC 3.5.2.2)",
                               a_plot$GeneFamily == "3.5.4.12" ~ "comEB (EC 3.5.4.12)",
                               TRUE ~ NA) 
a_plot$GeneAnnotations <- factor(a_plot$GeneAnnotations, levels = a_plot %>% arrange(GeneFamily) %>% select(GeneAnnotations) %>% unique() %>% pull())

# Plot
p <- ggplot(a_plot, aes(x = Days_Since_Tx, y = Abundance + offset, fill = Direction)) + 
  geom_smooth(method = "loess", color = "black", se = TRUE, level = 0.68) + 
  scale_fill_manual(values = colorscale) + 
  theme_pubr() + 
  facet_wrap(~GeneAnnotations, scales = "free_y", nrow = 2) + 
  scale_y_log10() + 
  ylab("Abundance (RPKG)") + 
  xlab("Days since treatment start") + 
  theme(legend.position = "none")

# Save
fn <- paste0(figdir, "ECvstime.pdf")
ggsave(filename = fn, plot = p, width = 6, height = 3)
```

# Figure S5 - Fluoropyrimidine metabolism pathway
Biorender.

# Figure S6 - CAP regional profiling experiment
```{r}
# Make directory
figdir <- paste0(dir, "FigureS6/")
dir.create(figdir)

tabledir_load = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Regional_Profiling/")
datadir <- tabledir_load
savedir <- figdir

# Set global variable
taxa_levels <- c("Phylum", "Class", "Order", "Family", "Genus", "Species", "ASVs")

# Read in metadata
fn <- paste0(datadir, "metadata.xlsx")
metadata <- read_excel(fn, col_names = T) %>%
  as.data.frame()

# Add number of reads
read_counts <- read_csv(paste0(datadir, "Nreads_raw.txt")) %>%
  select("Sample_ID", "Nreads_illumina") %>%
  dplyr::rename(Reads=Nreads_illumina) %>% 
  dplyr::rename(Sample = Sample_ID)
metadata <- metadata %>% left_join(read_counts, by='Sample')

# Add "TissueType2"
metadata$TissueType <- case_when(metadata$TissueType == "ProxiColon" ~ "Proximal colon",
                                 metadata$TissueType == "DistalColon" ~ "Distal colon",
                                 TRUE ~ metadata$TissueType)
metadata$TissueType2 <- case_when(metadata$Type == "Pellets" ~ "Stool",
                                  TRUE ~ metadata$TissueType)
metadata$TissueType2 <- factor(metadata$TissueType2, levels = c("Esophagus", "Stomach", "Duodenum", "Jejunum", "Ileum", "Cecum", "Proximal colon", "Distal colon", "Stool"))

# Add "Group2", which is group, but setting -7 and 0 as "Pre" treatment for beta diversity coloring
metadata$Group2 <- case_when(metadata$Day %in% c(-7,0) ~ "Pre",
                             TRUE ~ metadata$Group)

#Assign colors to each time course and groups
group.colours.3 <- c("Veh"="#008080", 
                   "Cap"= "red",
                   "CAP" = "red",
                   "Pre" = "grey")

mousepalette <- gg_color_hue(length(unique(metadata$MouseID)))
mouse.colours <- mousepalette
names(mouse.colours) <- unique(metadata$MouseID)

daypalette <- gg_color_hue(length(unique(metadata$Day)))
day.colours <- daypalette
names(day.colours) <- unique(metadata$Day)

sex.colours <- c("Male" = "#ADD8E6",
                 "Female" = "#FFCCCB")

# Fix sample names (match with taxa tables)
metadata$Sample <- gsub("-", "\\.", metadata$Sample)
```

## PCOA by tissue
```{r}
# Pick axes
axis1 = 1; axis2 = 2

# Pick plotting axis
metadata_plot <- metadata %>% filter(Group %in% c("Cap", "Veh")) %>%
  filter(Day %in% c(0, 14)) %>% 
  filter(!(TissueType2 %in% c("Esophagus", "Stomach", "Duodenum")))

# Read in data - we'll look at genera
fn <- paste0(tabledir_load, "TaxaSummaryGenus.csv")
t <- read.csv(fn)

# CLR transform
rownames(t) <- t$X; t <- t %>% select(-X)
t <- clr_transform(t)

# PCA
df <- t(t) %>% as.data.frame()
df$Sample <- rownames(df)
pca_res <- PCA(df[, -which(names(df) %in% c("Sample"))], scale.unit = FALSE, graph = FALSE)

pca_scores <- as.data.frame(pca_res$ind$coord)
pca_scores$Sample <- df$Sample
pca_scores$Group <- df$Group
percent_var <- pca_res$eig[axis1:axis2, "percentage of variance"] # percent variance explained

pca_plot <- inner_join(pca_scores, metadata_plot, by = "Sample")

pca_plot$Group2 <- case_when(pca_plot$Group2 == "Cap" ~ "CAP",
                             TRUE ~ pca_plot$Group2)
pca_plot$Group2 <- factor(pca_plot$Group2, levels = c("Pre", "Veh", "CAP"))

p <- ggplot(pca_plot, aes(x = .data[[paste0("Dim.", axis1)]], y = .data[[paste0("Dim.", axis2)]])) +
  geom_point(aes(color = TissueType2), size = 2) +
  theme_pubr() +
  labs(x = paste("PCo", axis1, " (", sprintf("%.2f", percent_var[1]), "%)", sep = ""),
       y = paste("PCo", axis2, " (", sprintf("%.2f", percent_var[2]), "%)", sep = ""),
       color = "Site") +     
  theme(legend.position = "right") 

# Save
fn <- paste0(savedir, "PCA_Site.pdf")
ggsave(filename = fn, plot = p, width = 4, height = 3)
```

## PCOA by treatment
```{r}
# Pick axes
axis1 = 2; axis2 = 3

# Pick plotting axis
metadata_plot <- metadata %>% filter(Group %in% c("Cap", "Veh")) %>%
  filter(Day %in% c(0, 14)) %>% 
  filter(!(TissueType2 %in% c("Esophagus", "Stomach", "Duodenum"))) 

# Read in data - we'll look at genera
fn <- paste0(tabledir_load, "TaxaSummaryGenus.csv")
t <- read.csv(fn)

# CLR transform
rownames(t) <- t$X; t <- t %>% select(-X)
t <- clr_transform(t)

# PCA
df <- t(t) %>% as.data.frame()
df$Sample <- rownames(df)
pca_res <- PCA(df[, -which(names(df) %in% c("Sample"))], scale.unit = FALSE, graph = FALSE)

pca_scores <- as.data.frame(pca_res$ind$coord)
pca_scores$Sample <- df$Sample
pca_scores$Group <- df$Group
percent_var <- pca_res$eig[axis1:axis2, "percentage of variance"] # percent variance explained

pca_plot <- inner_join(pca_scores, metadata_plot, by = "Sample")

pca_plot$Group2 <- case_when(pca_plot$Group2 == "Cap" ~ "CAP",
                             TRUE ~ pca_plot$Group2)
pca_plot$Group2 <- factor(pca_plot$Group2, levels = c("Pre", "Veh", "CAP"))

pca_plot$TissueType3 <- case_when(pca_plot$TissueType2 %in% c("Jejunum", "Ileum") ~ "Small intestine",
                                  pca_plot$TissueType2 %in% c("Cecum", "Proximal colon", "Distal colon") ~ "Large intestine",
                                  pca_plot$TissueType2 %in% c("Stool") ~ "Stool", 
                                  TRUE ~ NA)
pca_plot$TissueType3 = factor(pca_plot$TissueType3, levels = c("Small intestine", "Large intestine", "Stool"))


p <- ggplot(pca_plot, aes(x = .data[[paste0("Dim.", axis1)]], y = .data[[paste0("Dim.", axis2)]])) +
  geom_point(aes(color = Group2), size = 2) +
  theme_pubr() +
  labs(x = paste("PCo", axis1, " (", sprintf("%.2f", percent_var[1]), "%)", sep = ""),
       y = paste("PCo", axis2, " (", sprintf("%.2f", percent_var[2]), "%)", sep = ""),
       color = "Treatment") +     
  theme(legend.position = "right") +
  scale_color_manual(values = group.colours.3) +
  facet_wrap(~TissueType3) 

# Save
fn <- paste0(savedir, "PCA_Tx.pdf")
ggsave(filename = fn, plot = p, width = 6, height = 3)
```

## PERMANOVA
```{r}
rerun_permanova = TRUE

metadata_plot <- metadata %>% filter(Group %in% c("Cap", "Veh")) %>%
  filter(Day %in% c(0, 14)) %>% 
  filter(!(TissueType2 %in% c("Esophagus", "Stomach", "Duodenum")))

# Read in data
fn <- paste0(tabledir_load, "TaxaSummaryGenus.csv")
d <- fread(fn) %>% as.data.frame()
rownames(d) <- d$V1; d <- d %>% select(-V1) 
colnames(d) <- gsub("-", "\\.", colnames(d))

# CLR transform
geometric_means <- apply(d, 1, function(row) exp(mean(log(row[row > 0]))))
pseudo_count = min(d[d > 0])/2
clr_data <- log((d + pseudo_count) / geometric_means)

# Omit NA values
clr_data <- na.omit(clr_data)

# Remove zero-variance columns
clr_data_t <- t(clr_data)
clr_data_t <- clr_data_t[ , which(apply(clr_data_t, 2, var) != 0)]
d_t <- t(d)
d_t <- d_t[ , which(apply(d_t, 2, var) != 0)]

# Concatenate metadata
meta_c <- metadata_plot %>% filter(Sample %in% rownames(clr_data_t) & Sample %in% rownames(d_t)) # only keep samples that we have sequencing data for
clr_data_t <- clr_data_t[meta_c$Sample,] # and vice versa for metadata matching
d_t <- d_t[meta_c$Sample,]

# PERMANOVA testing 
meta_perm <- meta_c 
row_order <- meta_perm %>% select(Sample) %>% pull(); clr_data_ordered <-clr_data_t[row_order,]; d_ordered <- d_t[row_order,] # arrange
if(sum(meta_perm$Sample == rownames(clr_data_ordered)) + sum(meta_perm$Sample == rownames(d_ordered)) != 2*length(meta_perm$Sample)){ # check that they match
    print("PERMANVA order wrong!")
}
if (rerun_permanova){
  dist_clr <- suppressWarnings(vegdist(clr_data_ordered, method="euclidean", na.rm = TRUE, binary = TRUE)) # CLR-Euclidean
  set.seed(42)
  res_clr <- adonis2(dist_clr ~ TissueType2 + Group2, data = meta_perm, permutations=1000, strata = meta_perm$MouseID, by = "margin")
}
```

## Correlations
```{r}
site1_name = "Stool"
site2_name = "Ileum"

fn <- paste0(tabledir_load, "TaxaSummaryGenusCLR.csv")
t <- read.csv(fn)

t <- t %>% filter(Group %in% c("Cap", "Veh")) %>%
  filter(Day == 14)

# Zero replacement
minval = min(t$Abundance_CLR[is.finite(t$Abundance_CLR)])
t$Abundance_CLR[!is.finite(t$Abundance_CLR)] <- minval

# Type options: "Stool" (1 hit), "Duodenum" (no hits), "Jejunum" (no hits), "Ileum" (no hits), "Cecum" (2 hit), "ProxiColon" (no hits), "DistalColon" (no hits)
fit_model <- function(t, tissuetype = "Stool"){
  if (tissuetype == "Proximal colon"){tissuetype <- "ProxiColon"}
  if (tissuetype == "Distal colon"){tissuetype <- "DistalColon"}
  t_sub <- t %>% filter(TissueType2 %in% c(tissuetype))
  res <- NULL
  taxas = unique(t_sub$Taxa)
  for (taxa in taxas){
    test <- t_sub %>% filter(Taxa == taxa)
    mod <- lm(Abundance_CLR ~ Group2, data = test)
    new_row <- as.data.frame(coef(summary(mod)))
    colnames(new_row) <- c("Estimate", "Error", "t.value", "p.value")
    new_row$Group <- rownames(new_row)
    new_row$Taxa <- taxa
    if(length(res) == 0){
      res <- new_row
    } else {
      res <- rbind(res, new_row)
    }
  }
  
  return(res)
}

# Compare sites
res_1 <- fit_model(t, site1_name)
res_2 <- fit_model(t, site2_name)

# round results so we can remove artifact!
res_1$Estimate <- round(res_1$Estimate*1000)/1000
res_2$Estimate <- round(res_2$Estimate*1000)/1000

site1 <- table(res_1$Estimate) %>% as.data.frame()
site1_bad <- site1 %>% filter(Freq > 5) %>% select(Var1) %>% pull()  # if we get the same value many times, it's likely an artifact
res_1 <- res_1 %>% filter(!(Estimate %in% site1_bad)) %>% 
  filter(Group != "(Intercept)") # Need to fix this!!!

site2 <- table(res_2$Estimate) %>% as.data.frame()
site2_bad <- site2 %>% filter(Freq > 5) %>% select(Var1) %>% pull() # if we get the same value many times, it's likely an artifact
res_2 <- res_2 %>% filter(!(Estimate %in% site2_bad)) %>% 
  filter(Group != "(Intercept)")

# Correlate results
res_1$Estimate_1 <- res_1$Estimate 
res_2$Estimate_2 <- res_2$Estimate 
res_both <- inner_join(res_1, res_2, by = "Taxa")

spearman_test <- cor.test(res_both$Estimate_1, res_both$Estimate_2, method = "pearson")
correlation <- spearman_test$estimate
p_value <- spearman_test$p.value

# Plot
p <- ggplot(res_both, aes(x = log2(exp(-Estimate_1)), y = log2(exp(-Estimate_2)))) + # - signs to switch between CAP/Veh; log2(exp) to convert to log2FC as units
  geom_point() + 
  theme_pubr() +
  geom_smooth(method = "lm", color = "black") + 
  annotate("text", x=-1, y=4, label = sprintf("Pearson's rho = %.2f", correlation)) +
  annotate("text", x=-1, y=3.2, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) + 
  xlab(bquote(.(site1_name) ~ "fold-change (log" [2]~"CAP/Veh)")) + 
  ylab(bquote(.(site2_name) ~ "fold-change (log" [2]~"CAP/Veh)")) +
theme(axis.title.x = element_text(size = 11),    # Adjust size for x-axis label
        axis.title.y = element_text(size = 11))

# Save
fn <- paste0(savedir, "Corr", site2_name, ".pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## Heatmap
```{r}
fn <- paste0(tabledir_load, "TaxaSummaryGenusCLR.csv")
t <- read.csv(fn)

t <- t %>% filter(Group %in% c("Cap", "Veh")) %>%
  filter(Day == 14)

res_all <- NULL
for (site in c("Ileum", "Jejunum", "Cecum", "Proximal colon", "Distal colon", "Stool")){
  res <- fit_model(t, site)
  res <- res %>% arrange(Taxa) %>% filter(Group != "(Intercept)") # exclude intercept group
  res[[site]] = log2(exp(-res$Estimate)) # convert to log2 FC, switch so that positive number -> CAP enriched
  res <- res %>% select(all_of(c(site, "Taxa")))
  if (length(res_all) == 0){
    res_all <- res
  } else {
    res_all <- left_join(res_all, res, by = "Taxa")
  }
}

# Data wrangling
df <- reshape2::melt(res_all, by = "Taxa")
colnames(df) <- c("Taxa", "Site", "log2FC")
taxa_order <- df %>% group_by(Taxa) %>% dplyr::summarise(FC = mean(log2FC)) %>% arrange(FC) %>% select(Taxa) %>% pull()
df$Taxa <- factor(df$Taxa, levels = taxa_order)
df$Site <- factor(df$Site, levels = rev(c("Duodenum", "Jejunum", "Ileum", "Cecum", "Proximal colon", "Distal colon", "Stool")))

# Clustering
df_wide <- df %>% 
  tidyr::pivot_wider(names_from = Site, values_from = log2FC) %>%
  as.data.frame()
rownames(df_wide) <- df_wide$Taxa; df_wide <- df_wide %>% select(-Taxa)
df_wide[is.na(df_wide)] <- 0 # set NA to 0

# Heatmap with ggplot2
df_matrix <- as.matrix(df_wide)

# Perform hierarchical clustering on rows and columns
row_clusters <- hclust(dist(df_matrix, method = "euclidean"), method = "ward.D2")
col_clusters <- hclust(dist(t(df_matrix), method = "euclidean"), method = "ward.D2")

# Order the matrix according to the clustering
ordered_matrix <- df_matrix[rev(row_clusters$order), rev(col_clusters$order)]

# Melt the ordered matrix for ggplot2
df_long <- reshape2::melt(ordered_matrix)
colnames(df_long) <- c("Taxa", "Site", "log2FC")

# Reorder 
df_long$Taxa <- factor(df_long$Taxa, levels = rev(rownames(ordered_matrix)))

# Plot heatmap again
df_long$Site <- factor(df_long$Site, levels = rev(c("Average", "Duodenum", "Jejunum", "Ileum", "Cecum", "Proximal colon", "Distal colon", "Stool")))
p <- ggplot(df_long, aes(y = Site, x = Taxa, fill = log2FC)) + 
  geom_tile(colour="white") + 
  scale_fill_gradient2(low = "orange", high = "blue") + 
  xlab("Genus") + 
  ylab("") + 
  #scale_x_discrete(labels = setNames(df_long$Label, df_long$Taxa)) + 
  theme_pubr() + 
  theme(axis.text.x = element_blank()) + 
  labs(fill = bquote("log" [2]*"FC (CAP/Veh)")) + 
  theme(legend.position = "top")

# Save
fn <- paste0(savedir, "Heatmap.pdf")
ggsave(filename = fn, plot = p, width = 6.5, height = 3)
```

## preTA+ taxa 
```{r}
# Set species of interest (preTA+)
soi <- c("Lachnospiraceae_UCG-001", "[Eubacterium]_siraeum_group", "f__Oscillospiraceae; NA") 

# Read data
fn <- paste0(tabledir_load, "TaxaSummaryGenusCLR.csv")
t <- read.csv(fn)

t <- t %>% filter(Group %in% c("Cap", "Veh")) %>%
  filter(Day == 14) %>% 
  filter(!(TissueType2 %in% c("Esophagus", "Stomach", "Duodenum")))
t$TissueType2[t$TissueType2 == "ProxiColon"] <- "Proximal colon"
t$TissueType2[t$TissueType2 == "DistalColon"] <- "Distal colon"

# Zero replacement
minval = min(t$Abundance_CLR[is.finite(t$Abundance_CLR)])
t$Abundance_CLR[!is.finite(t$Abundance_CLR)] <- minval

# Plot
t_sub <- t %>% filter(grepl(paste0(soi, collapse = "|"), Taxa))
t_sub <- t_sub %>% filter(Taxa != "d__Bacteria;  p__Firmicutes;  c__Clostridia;  o__Oscillospirales;  f__Oscillospiraceae;  g__Oscillibacter;  s__unidentified") %>% filter(Taxa != "d__Bacteria;  p__Firmicutes;  c__Clostridia;  o__Oscillospirales;  f__Oscillospiraceae;  g__Oscillibacter;  NA")
t_sub <- t_sub %>% group_by(TissueType2, Group2, MouseID, Taxa) %>% dplyr::summarise(Abundance_CLR = mean(Abundance_CLR))
t_sub$Group2 <- factor(t_sub$Group2, levels = c("Veh", "Cap")) 
t_sub$TissueType2 <- factor(t_sub$TissueType2, levels = rev(c("Jejunum", "Ileum", "Cecum", "Proximal colon", "Distal colon", "Stool")))
p <- ggplot(t_sub, aes(x = TissueType2, y = Abundance_CLR, color = Group2, fill = Group2)) + 
  geom_boxplot(position = position_dodge(width = 0.75), alpha = 0.5, outlier.size = -1) + 
  theme_pubr() + 
  facet_wrap(~Taxa) + 
  scale_color_manual(values = group.colours.3) + 
  scale_fill_manual(values = group.colours.3) + 
  xlab("") + 
  ylab("Abundance (CLR)") + 
  stat_compare_means(method = "anova", aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "right")

# Calculate p value
t_sub$TissueMouse <- paste0(t_sub$TissueType2, t_sub$MouseID)
model <- nlme::lme(Abundance_CLR ~ Group2, 
             random = ~ 1 | TissueType2, 
             data = t_sub)
p_value <- summary(model)$tTable[2,"p-value"]

# Version 2
t_sub$TissueTypeNum <- 7-as.numeric(factor(t_sub$TissueType2))
t_sub$Group2 <- factor(t_sub$Group2, levels = c("Veh", "Cap", "CAP"))
t_sub$Group2[t_sub$Group2 == "Cap"] <- "CAP"
p <- ggplot(t_sub, aes(x = TissueTypeNum, y = Abundance_CLR, color = Group2, fill = Group2)) + 
  geom_smooth(method = "loess", level = 0.68) + 
  theme_pubr() +
  scale_color_manual(values = group.colours.3) + 
  scale_fill_manual(values = group.colours.3) +
  ylab("Abundance (CLR)") + 
  scale_x_continuous(labels = c("Jejunum", "Ileum", "Cecum", "Proximal colon", "Distal colon", "Stool"), breaks = c(1,2,3,4,5,6)) +
  xlab("") + 
  labs(fill = "", color = "") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  annotate("text", x=5, y=1.25, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) +
  theme(legend.position = "none") + 
  ggtitle(expression(italic(preTA)*"-positive species")) + 
  theme(plot.title = element_text(size = 12))

# Save
fn <- paste0(savedir, "preTA.pdf")
ggsave(filename = fn, plot = p, width = 3.5, height = 3)
```

## top CAP-enriched genera in stool
```{r}
# Set species of interest (preTA+)
soi <- c("Tyzzerella") 

# Read data
fn <- paste0(tabledir_load, "TaxaSummaryGenusCLR.csv")
t <- read.csv(fn)

t <- t %>% filter(Group %in% c("Cap", "Veh")) %>%
  filter(Day == 14) %>% 
  filter(!(TissueType2 %in% c("Esophagus", "Stomach", "Duodenum")))
t$TissueType2[t$TissueType2 == "ProxiColon"] <- "Proximal colon"
t$TissueType2[t$TissueType2 == "DistalColon"] <- "Distal colon"

# Zero replacement
minval = min(t$Abundance_CLR[is.finite(t$Abundance_CLR)])
t$Abundance_CLR[!is.finite(t$Abundance_CLR)] <- minval

# Plot
t_sub <- t %>% filter(grepl(paste0(soi, collapse = "|"), Taxa))
t_sub <- t_sub %>% filter(Taxa != "d__Bacteria;  p__Firmicutes;  c__Clostridia;  o__Oscillospirales;  f__Oscillospiraceae;  g__Oscillibacter;  s__unidentified") %>% filter(Taxa != "d__Bacteria;  p__Firmicutes;  c__Clostridia;  o__Oscillospirales;  f__Oscillospiraceae;  g__Oscillibacter;  NA")
t_sub <- t_sub %>% group_by(TissueType2, Group2, MouseID, Taxa) %>% dplyr::summarise(Abundance_CLR = sum(Abundance_CLR))
t_sub$Group2 <- factor(t_sub$Group2, levels = c("Veh", "Cap")) 
t_sub$TissueType2 <- factor(t_sub$TissueType2, levels = rev(c("Jejunum", "Ileum", "Cecum", "Proximal colon", "Distal colon", "Stool")))
p <- ggplot(t_sub, aes(x = TissueType2, y = Abundance_CLR, color = Group2, fill = Group2)) + 
  geom_boxplot(position = position_dodge(width = 0.75), alpha = 0.5, outlier.size = -1) + 
  theme_pubr() + 
  facet_wrap(~Taxa) + 
  scale_color_manual(values = group.colours.3) + 
  scale_fill_manual(values = group.colours.3) + 
  xlab("") + 
  ylab("Abundance (CLR)") + 
  stat_compare_means(method = "anova", aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "right")

# Calculate p value
t_sub$TissueMouse <- paste0(t_sub$TissueType2, t_sub$MouseID)
model <- nlme::lme(Abundance_CLR ~ Group2, 
             random = ~ 1 | TissueType2, 
             data = t_sub)
p_value <- summary(model)$tTable[2,"p-value"]

# Version 2
t_sub$TissueTypeNum <- 7-as.numeric(factor(t_sub$TissueType2))
t_sub$Group2 <- factor(t_sub$Group2, levels = c("Veh", "Cap", "CAP"))
t_sub$Group2[t_sub$Group2 == "Cap"] <- "CAP"
p <- ggplot(t_sub, aes(x = TissueTypeNum, y = Abundance_CLR, color = Group2, fill = Group2)) + 
  geom_smooth(method = "loess", level = 0.68) + 
  theme_pubr() +
  scale_color_manual(values = group.colours.3) + 
  scale_fill_manual(values = group.colours.3) +
  ylab("Abundance (CLR)") + 
  scale_x_continuous(labels = c("Jejunum", "Ileum", "Cecum", "Proximal colon", "Distal colon", "Stool"), breaks = c(1,2,3,4,5,6)) +
  xlab("") + 
  labs(fill = "", color = "") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  annotate("text", x=5, y=1.25, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) +
  theme(legend.position = "none") + 
  ggtitle(expression(italic(Tyzzerella))) + 
  theme(plot.title = element_text(size = 12))

# Save
fn <- paste0(savedir, "Tyzzerella.pdf")
ggsave(filename = fn, plot = p, width = 3.5, height = 3)
```

## top CAP-depleted genera in stool
```{r}
# Set species of interest 
soi <- c("Coriobacteriaceae_UCG-002") 

# Read data
fn <- paste0(tabledir_load, "TaxaSummaryGenusCLR.csv")
t <- read.csv(fn)

t <- t %>% filter(Group %in% c("Cap", "Veh")) %>%
  filter(Day == 14) %>% 
  filter(!(TissueType2 %in% c("Esophagus", "Stomach", "Duodenum")))
t$TissueType2[t$TissueType2 == "ProxiColon"] <- "Proximal colon"
t$TissueType2[t$TissueType2 == "DistalColon"] <- "Distal colon"

# Zero replacement
minval = min(t$Abundance_CLR[is.finite(t$Abundance_CLR)])
t$Abundance_CLR[!is.finite(t$Abundance_CLR)] <- minval

# Plot
t_sub <- t %>% filter(grepl(paste0(soi, collapse = "|"), Taxa))
t_sub <- t_sub %>% filter(Taxa != "d__Bacteria;  p__Firmicutes;  c__Clostridia;  o__Oscillospirales;  f__Oscillospiraceae;  g__Oscillibacter;  s__unidentified") %>% filter(Taxa != "d__Bacteria;  p__Firmicutes;  c__Clostridia;  o__Oscillospirales;  f__Oscillospiraceae;  g__Oscillibacter;  NA")
t_sub <- t_sub %>% group_by(TissueType2, Group2, MouseID, Taxa) %>% dplyr::summarise(Abundance_CLR = sum(Abundance_CLR))
t_sub$Group2 <- factor(t_sub$Group2, levels = c("Veh", "Cap")) 
t_sub$TissueType2 <- factor(t_sub$TissueType2, levels = rev(c("Jejunum", "Ileum", "Cecum", "Proximal colon", "Distal colon", "Stool")))
p <- ggplot(t_sub, aes(x = TissueType2, y = Abundance_CLR, color = Group2, fill = Group2)) + 
  geom_boxplot(position = position_dodge(width = 0.75), alpha = 0.5, outlier.size = -1) + 
  theme_pubr() + 
  facet_wrap(~Taxa) + 
  scale_color_manual(values = group.colours.3) + 
  scale_fill_manual(values = group.colours.3) + 
  xlab("") + 
  ylab("Abundance (CLR)") + 
  stat_compare_means(method = "anova", aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "right") 

# Calculate p value
t_sub$TissueMouse <- paste0(t_sub$TissueType2, t_sub$MouseID)
model <- nlme::lme(Abundance_CLR ~ Group2, 
             random = ~ 1 | TissueType2, 
             data = t_sub)
p_value <- summary(model)$tTable[2,"p-value"]

# Version 2
t_sub$TissueTypeNum <- 7-as.numeric(factor(t_sub$TissueType2))
t_sub$Group2 <- factor(t_sub$Group2, levels = c("Veh", "Cap", "CAP"))
t_sub$Group2[t_sub$Group2 == "Cap"] <- "CAP"
p <- ggplot(t_sub, aes(x = TissueTypeNum, y = Abundance_CLR, color = Group2, fill = Group2)) + 
  geom_smooth(method = "loess", level = 0.68) + 
  theme_pubr() +
  scale_color_manual(values = group.colours.3) + 
  scale_fill_manual(values = group.colours.3) +
  ylab("Abundance (CLR)") + 
  scale_x_continuous(labels = c("Jejunum", "Ileum", "Cecum", "Proximal colon", "Distal colon", "Stool"), breaks = c(1,2,3,4,5,6)) +
  xlab("") + 
  labs(fill = "", color = "") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  annotate("text", x=5, y=5, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) +
  theme(legend.position = "left") + 
  ggtitle(expression(italic(Coriobacteriaceae)*"_UCG-002")) + 
  theme(plot.title = element_text(size = 12))

# Save
fn <- paste0(savedir, "Coriobacteriaceae_UCG002.pdf")
ggsave(filename = fn, plot = p, width = 4.5, height = 3)
```

# Figure S7 - SIC supplement
```{r}
# Make directory
figdir <- paste0(dir, "FigureS7/")
dir.create(figdir)
```

## SIC growth suppression
```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SIC_Data/GO_SICs_GrowthCurves_fits.csv")
g <- read.csv(fn)

# Filter//wrangle
g <- g %>% filter(Passage == 1) %>%
  dplyr::rename(Donor = Strain) %>% 
  dplyr::rename(Treatment = Condition) %>%
  filter(Donor != "Pt31")
g$Donor <- case_when(g$Donor == "Pt50" ~ "Donor 50",
                     g$Donor == "Pt19" ~ "Donor 19",
                     g$Donor == "Pt21" ~ "Donor 21",
                     TRUE ~ NA)

# Plot carrying capacity
g$Treatment[g$Treatment == "Vehicle"] <- "Veh"
g$Group <- factor(g$Treatment, levels = c("Veh", "CAP", "5FU"))
p <- ggplot(g, aes(x = Group, y = k)) + 
  geom_boxplot(aes(fill = Group), alpha = 0.5, outlier.size = -1) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  ylab("Carrying capacity (OD)") + 
  xlab("") + 
  facet_wrap(~Donor) + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  stat_compare_means(comparisons = list(c("Veh", "CAP"), c("Veh", "5FU"))) +
  scale_color_manual(values = cap_palette) + 
  scale_fill_manual(values = cap_palette) + 
  theme(legend.position = "none") +
  scale_y_continuous(expand = expansion(mult = c(0.05, 0.15)))

# Save
fn <- paste0(figdir, "SIC_CarryingCapacity.pdf")
ggsave(filename = fn, plot = p, height = 3.5, width = 4.5)
```

## SIC Diversity
```{r}
# Metadata
datadir <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SIC_Data/") 

read_counts <- read_csv(paste0(datadir, "Nreads_raw.txt")) %>%
  select("Sample_ID", "Nreads_illumina") %>%
  rename(Reads=Nreads_illumina)

met <- read_counts

raw_read_cutoff <- 5000

stool_samples <- c('Pt50Stool', 'Pt19Stool', 'Pt21Stool', 'Pt31Stool')

met$Passage <- gsub(".*Pa", "", met$Sample_ID)
met$Passage <- case_when(met$Sample_ID == 'Pt50Stool' ~ '0',
                         met$Sample_ID == 'Pt19Stool' ~ '0',
                         met$Sample_ID == 'Pt21Stool' ~ '0',
                         met$Sample_ID == 'Pt31Stool' ~ '0',
                           TRUE ~ gsub("Tx.*", "", met$Passage))
met$Passage <- as.factor(met$Passage)

met$Treatment <- gsub(".*Tx", "", met$Sample_ID)
met$Treatment <- gsub("Pt.*", "", met$Treatment)
met$Treatment[met$Sample_ID %in% stool_samples] <- 'Stool'
met$Treatment <- as.factor(met$Treatment)
met$Treatment <- factor(met$Treatment, levels = c('Veh', '5FU', 'CAP'))

met$Patient_ID <- gsub(".*Pt", "", met$Sample_ID)
met$Patient_ID <- gsub("R.*", "", met$Patient_ID)
met$Patient_ID <- gsub("Stool.*", "", met$Patient_ID)
met$Patient_ID <- paste0("Donor ", met$Patient_ID)
met$Patient_ID <- as.factor(met$Patient_ID)

met$Replicate <-gsub(".*R", "", met$Sample_ID)
met$Replicate <- as.factor(met$Replicate)

met$Type <- case_when(grepl("Stool", met$Sample_ID) ~ "Donor",
                      grepl("Control|Blank", met$Sample_ID) ~ "Control",
                      TRUE ~ "Recipient")
met$Type <- as.factor(met$Type)

met$Experiment <- case_when(met$Replicate %in% c('1', '2', '3') ~ 'TreatStool',
                            met$Replicate %in% c('4', '5', '6') ~ 'SIC',
                            TRUE ~ 'NA')
met$Experiment <- as.factor(met$Experiment)

met$PtRepTx <- paste0(met$Patient_ID, met$Replicate, met$Treatment)

metadata <- met %>% filter(Type %in% c('Donor', 'Recipient'))
metadata <- metadata %>% filter(Reads > raw_read_cutoff)

# Read in qza files
svtable = read_qza(paste0(datadir, "ASV_table.qza")) #contains ASV-ID with counts for each sample
svseqs = read_qza(paste0(datadir, "ASV_sequences.qza")) #contains DNA sequences for each ASV-ID
tree = read_qza(paste0(datadir, "ASV_denovotree.qza"))
taxonomy = read_qza(paste0(datadir, "ASV_q2taxonomy.qza")) #contains taxonomic data for each ASV-ID

# Filter
dat <- filterSVTableData(svtable$data, countcut = 10, samplecut = 3)
dat <- dat[, metadata$Sample_ID]
dat <- filterSVTableData(dat, countcut = 10, samplecut = 3)

# Create taxtable
taxtable = taxonomy$data %>% 
  as_tibble() %>% 
  separate(Taxon, sep=";", c("Kingdom","Phylum","Class","Order","Family","Genus","Species"), remove = FALSE) #contains taxonomic data for each ASV

# Convert to phyloseq object - provide svtable (contains counts for each ASV ID), a phylogenetic tree, tax_table (taxonomy data for each ASV ID), and a metadata
po <- phyloseq(
  otu_table(dat, taxa_are_rows = T), 
  phy_tree(tree$data), 
  tax_table(as.data.frame(taxtable) %>% select(-Confidence) %>% column_to_rownames("Feature.ID") %>% as.matrix()), 
  sample_data(metadata %>% as.data.frame() %>% column_to_rownames("Sample_ID"))
  )

# Rarefy
sample_sums <- sample_sums(po)
rare_depth <- min(sample_sums[sample_sums > 1000])  # Choosing a threshold to avoid too few reads
po_rarefied <- rarefy_even_depth(po, sample.size = rare_depth, rngseed = 1, replace = FALSE, verbose = FALSE)

# Estimate richness
diversity <- estimate_richness(po, measures = c("Shannon", "Simpson", "Observed"))
diversity_rarefied <- estimate_richness(po_rarefied, measures = c("Shannon", "Simpson", "Observed"))

# Merge, plot
colnames(diversity_rarefied) <- paste0(colnames(diversity_rarefied), "_Rarefied")
diversity$Sample_ID <- rownames(diversity); diversity_rarefied$Sample_ID <- rownames(diversity_rarefied)
div <- inner_join(diversity, diversity_rarefied, by = "Sample_ID")

# Filter//wrangle
g <- left_join(div, metadata, by = "Sample_ID")
g <- g %>% filter(Passage == 1) %>%
  dplyr::rename(Donor = Patient_ID) %>% 
  dplyr::filter(Treatment %in% c("Veh", "CAP", "5FU")) %>% 
  filter(Donor != "Donor 31")
g$Group <- factor(g$Treatment, levels = c("Veh", "CAP", "5FU"))

# Plot
p <- ggplot(g, aes(x = Group, y = Observed_Rarefied)) + 
  geom_boxplot(aes(fill = Group), alpha = 0.5, outlier.size = -1) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  ylab("# ASVs (rarefied)") + 
  xlab("") + 
  facet_wrap(~Donor, scales = "free_y") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  stat_compare_means(comparisons = list(c("Veh", "CAP"), c("Veh", "5FU"))) +
  scale_color_manual(values = cap_palette) + 
  scale_fill_manual(values = cap_palette) + 
  theme(legend.position = "none") + 
  scale_y_continuous(expand = expansion(mult = c(0.05, 0.15))) # Prevent p values from being off the top margin

# Save
fn <- paste0(figdir, "SIC_Diversity.pdf")
ggsave(filename = fn, plot = p, height = 3.5, width = 4.5)
```

## SIC PCOA
```{r}
# Metadata
datadir <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SIC_Data/") 

read_counts <- read_csv(paste0(datadir, "Nreads_raw.txt")) %>%
  select("Sample_ID", "Nreads_illumina") %>%
  rename(Reads=Nreads_illumina)

met <- read_counts

raw_read_cutoff <- 5000

stool_samples <- c('Pt50Stool', 'Pt19Stool', 'Pt21Stool', 'Pt31Stool')

met$Passage <- gsub(".*Pa", "", met$Sample_ID)
met$Passage <- case_when(met$Sample_ID == 'Pt50Stool' ~ '0',
                         met$Sample_ID == 'Pt19Stool' ~ '0',
                         met$Sample_ID == 'Pt21Stool' ~ '0',
                         met$Sample_ID == 'Pt31Stool' ~ '0',
                           TRUE ~ gsub("Tx.*", "", met$Passage))
met$Passage <- as.factor(met$Passage)

met$Treatment <- gsub(".*Tx", "", met$Sample_ID)
met$Treatment <- gsub("Pt.*", "", met$Treatment)
met$Treatment[met$Sample_ID %in% stool_samples] <- 'Stool'
met$Treatment <- as.factor(met$Treatment)
met$Treatment <- factor(met$Treatment, levels = c('Veh', '5FU', 'CAP'))

met$Patient_ID <- gsub(".*Pt", "", met$Sample_ID)
met$Patient_ID <- gsub("R.*", "", met$Patient_ID)
met$Patient_ID <- gsub("Stool.*", "", met$Patient_ID)
met$Patient_ID <- paste0("Donor ", met$Patient_ID)
met$Patient_ID <- as.factor(met$Patient_ID)

met$Replicate <-gsub(".*R", "", met$Sample_ID)
met$Replicate <- as.factor(met$Replicate)

met$Type <- case_when(grepl("Stool", met$Sample_ID) ~ "Donor",
                      grepl("Control|Blank", met$Sample_ID) ~ "Control",
                      TRUE ~ "Recipient")
met$Type <- as.factor(met$Type)

met$Experiment <- case_when(met$Replicate %in% c('1', '2', '3') ~ 'TreatStool',
                            met$Replicate %in% c('4', '5', '6') ~ 'SIC',
                            TRUE ~ 'NA')
met$Experiment <- as.factor(met$Experiment)

met$PtRepTx <- paste0(met$Patient_ID, met$Replicate, met$Treatment)

metadata <- met %>% filter(Type %in% c('Donor', 'Recipient'))
metadata <- metadata %>% filter(Reads > raw_read_cutoff)

# Read in qza files
svtable = read_qza(paste0(datadir, "ASV_table.qza")) #contains ASV-ID with counts for each sample
svseqs = read_qza(paste0(datadir, "ASV_sequences.qza")) #contains DNA sequences for each ASV-ID
tree = read_qza(paste0(datadir, "ASV_denovotree.qza"))
taxonomy = read_qza(paste0(datadir, "ASV_q2taxonomy.qza")) #contains taxonomic data for each ASV-ID

# Filter
dat <- filterSVTableData(svtable$data, countcut = 10, samplecut = 3)
dat <- dat[, metadata$Sample_ID]
dat <- filterSVTableData(dat, countcut = 10, samplecut = 3)

# Create taxtable
taxtable = taxonomy$data %>% 
  as_tibble() %>% 
  separate(Taxon, sep=";", c("Kingdom","Phylum","Class","Order","Family","Genus","Species"), remove = FALSE) #contains taxonomic data for each ASV

# Convert to phyloseq object - provide svtable (contains counts for each ASV ID), a phylogenetic tree, tax_table (taxonomy data for each ASV ID), and a metadata
po <- phyloseq(
  otu_table(dat, taxa_are_rows = T), 
  phy_tree(tree$data), 
  tax_table(as.data.frame(taxtable) %>% select(-Confidence) %>% column_to_rownames("Feature.ID") %>% as.matrix()), 
  sample_data(metadata %>% as.data.frame() %>% column_to_rownames("Sample_ID"))
  )

po_clr <- transform_sample_counts(po, ClrFunction )

# Ordinate
po_clr_beta <- subset_samples(po_clr, Type == "Recipient")
po_test <- subset_samples(po_clr_beta, Passage == 1)

po_pt <- subset_samples(po_test, Patient_ID == "Donor 19")
pcoa = phyloseq::ordinate(po_pt, method="RDA", distance="euclidean")
p19 <- plot_pcoa(po_test, pcoa, axes = 1:2, fill_var = 'Treatment', colorscale = cap_palette, type = "NoType", savedir = NA, diffshape = TRUE, sictxshape = TRUE, suffix = "_TreatStool", returnplot = TRUE, facetPt = TRUE, title = FALSE, rmlegend = FALSE)

po_pt <- subset_samples(po_test, Patient_ID == "Donor 21")
pcoa = phyloseq::ordinate(po_pt, method="RDA", distance="euclidean")
p21 <- plot_pcoa(po_test, pcoa, axes = 1:2, fill_var = 'Treatment', colorscale = cap_palette, type = "Bray-Curtis", savedir = NA, diffshape = TRUE, sictxshape = TRUE, suffix = "_TreatStool", returnplot = TRUE, facetPt = TRUE, title = FALSE, rmlegend = FALSE)

po_pt <- subset_samples(po_test, Patient_ID == "Donor 50")
pcoa = phyloseq::ordinate(po_pt, method="RDA", distance="euclidean")
p50 <- plot_pcoa(po_test, pcoa, axes = 1:2, fill_var = 'Treatment', colorscale = cap_palette, type = "Bray-Curtis", savedir = NA, diffshape = TRUE, sictxshape = TRUE, suffix = "_TreatStool", returnplot = TRUE, facetPt = TRUE, title = FALSE, rmlegend = FALSE)

# Save
p <- ggarrange(p19 + guides(shape="none"), p21 + guides(shape="none"), p50 + guides(shape="none"), nrow = 1, common.legend = TRUE) 
fn <- paste0(figdir, "SIC_PCOA.pdf")
ggsave(filename = fn, plot = p, height = 2.75, width = 6)
```

## SIC CAP/5FU FC Correlation
```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SIC_Data/TaxaSummaryASVs.csv")
t <- read.csv(fn) %>% as.data.frame()
rownames(t) <- t$X; t <- t %>% select(-X); t_mat <- t
t <- sweep(t,2,colSums(t),`/`) # normalize to 1

# Wrangling
t_filt <- t[,grepl("Pa1", colnames(t))]
t_filt <- t_filt[,grepl("R1|R2|R3|R4|R5|R6", colnames(t_filt))]
t_filt <- t_filt[,!grepl("Pt31", colnames(t_filt))]

# Function
aggregate_pt <- function(pt_interest = "Pt50", t_filt_in){
  t_filt <- t_filt_in[,grepl(pt_interest, colnames(t_filt_in))]
  
  # Only keep ASVs with at least 3 nonzero entries
  asv_keep <- names(rowSums(t_filt > 0) >= 3)
  t_filt <- t_filt[asv_keep, ]
  
  # Add pseudocount
  t_filt <- t_filt + min(t_filt[t_filt > 0])
  
  # Split into new dataframes
  t_veh <- t_filt[,grepl("TxVeh", colnames(t_filt))]
  t_cap <- t_filt[,grepl("TxCAP", colnames(t_filt))]
  t_5fu <- t_filt[,grepl("Tx5FU", colnames(t_filt))]
  
  # fold change
  fc_cap <- log2(rowMeans(t_cap)/rowMeans(t_veh))
  fc_5fu <- log2(rowMeans(t_5fu)/rowMeans(t_veh))
  df <- data.frame(CAP = fc_cap, FU = fc_5fu, ASV = names(fc_cap), Donor = pt_interest)
}

# Combine all patients
pt50 <- aggregate_pt(pt_interest = "Pt50", t_filt_in = t_filt)
pt19 <- aggregate_pt(pt_interest = "Pt19", t_filt_in = t_filt)
pt21 <- aggregate_pt(pt_interest = "Pt21", t_filt_in = t_filt)

df <- rbind(pt50, pt19, pt21)

# Stats
spearman_test <- cor.test(df$CAP, df$FU, method = "spearman")
correlation <- spearman_test$estimate
p_value <- spearman_test$p.value

# Plot
p <- ggplot(df, aes(x = CAP, y = FU)) + 
  geom_point(alpha = 0.33) + 
  theme_pubr() +
  xlab("Fold-change (log2 CAP/Veh)") + 
  ylab("Fold-change (log2 5FU/Veh)") + 
  annotate("text", x=0, y=13, label = sprintf("Spearman's rho = %.2f", correlation)) +
  annotate("text", x=0, y=11, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) +
  theme(legend.position = "none") + 
  ylim(-13, 13)

# Save
fn <- paste0(figdir, "SIC_FCCorrelation.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)
```

# Figure 3 - Mechanisms of toxicity
```{r}
# Make directory
figdir <- paste0(dir, "Figure3/")
dir.create(figdir)
```

## Mechanism Weights
```{r}
# Read in data sheet from excel file
fn_xl <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/CAP_Tox_Mechanism.xlsx")
w <- read_excel(fn_xl, sheet = "TractableWeights") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(Mouse = variable, Weight = value)
w$Group <- case_when(grepl("9|10|11|12|13|14|15|16|17|18|19|20", w$Mouse) ~ "CAP",
                     TRUE ~ "Veh")
w$Group <- factor(w$Group, levels = c("Veh", "CAP"))

# Mixed-effects model p value - group*time. The interaction term is significant (p < 0.0001)
w_model <- w %>% filter(!is.na(Weight))
m1 <- lme(Weight ~ Group*Day, random = ~ 1 | Mouse, data=w_model)
sums <- summary(m1)$tTable
p_value <- sums[4, 5]

# Two-way anova testing
res.aov2 <- aov(Weight ~ Group*Day, data = w_model)
p_int <- summary(res.aov2)[[1]][["Pr(>F)"]][[3]] # Interaction p value
p_value <- summary(res.aov2)[[1]][["Pr(>F)"]][[1]] # Group (i.e. colonization) p value

# Plot
p <- ggplot(w, aes(x = Day, y = Weight, group = Group, color = Group, shape = Group, fill = Group)) + 
  stat_summary(fun.y="mean", geom="line") + 
  stat_summary(fun.y="mean", geom="point") + 
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.2) + 
  xlab("CAP duration (days)") + 
  ylab("Weight vs baseline (%)") + 
  geom_jitter(width = 0.1, aes(alpha = Group)) + 
  scale_alpha_manual(values = c("CAP" = 0.2, "Veh" = 0.6)) + 
  theme_pubr() + 
  scale_color_manual(values = cap_palette_mouse) + 
  scale_fill_manual(values = cap_palette_mouse) + 
  annotate("text", x=2, y=87, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) +
  theme(legend.title = element_blank()) + 
  scale_x_continuous(breaks = c(0, 4, 8, 12))

# Save
fn <- paste0(figdir, "Mech_Weights.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)

```

## Midpoint/endpoint weights
```{r}
fn_xl <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/CAP_Tox_Mechanism.xlsx")
o <- read_excel(fn_xl, sheet = "OrganSize") %>% as.data.frame()

# Wrangle data for plot
o_mid <- o %>% select(Mouse, Group, WeightMidpointPerc); colnames(o_mid) <- c("Mouse", "Group", "Weight"); o_mid$time = "Day 4"
o_end <- o %>% filter(SacGroup == "Endpoint") %>% select(Mouse, Group, WeightEndpointPerc); colnames(o_end) <- c("Mouse", "Group", "Weight"); o_end$time <- "Day 12"
o_merged <- rbind(o_mid, o_end)

# Plot
o_merged$time <- factor(o_merged$time, levels = c("Day 4", "Day 12"))
o_merged$Group <- factor(o_merged$Group, levels = c("Veh", "CAP"))
p <- ggplot(o_merged, aes(x = Group, y = Weight)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  ylab("Weight vs baseline (%)") + 
  scale_color_manual(values = cap_palette_mouse) + 
  scale_fill_manual(values = cap_palette_mouse) + 
  facet_wrap(~time) + 
  stat_compare_means(method = "wilcox", label.x = 1.2, label.y = 106.4, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "none") + 
  ylim(c(84.5, 107)) + 
  xlab("")

# Save
fn <- paste0(figdir, "Mech_WeightsDays.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## Lipocalin 2 (Lcn-2) standard curve
```{r}
# Read in data
fn_xl <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/ELISA_240221.xlsx")
s <- read_excel(fn_xl, sheet = "StandardCurve") %>% as.data.frame()

# Plot
p <- ggplot(s, aes(x = ConcentrationPGML, y = OD450_Normalized)) + 
  theme_pubr() +
  xlab("Lipocalin-2 standard (pg/mL)") + 
  ylab("OD450 (background-subtracted)") + 
  geom_point() +  # Scatter plot
  geom_smooth(method = "lm", se = FALSE, color = "black") +  # Linear regression line
  stat_poly_eq(formula = y ~ x - 1, 
               aes(label = after_stat(adj.rr.label)),
               parse = TRUE) 

# Save
fn <- paste0(figdir, "Lcn2_StandardCurve.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## Midpoint/endpoint Lcn-2
```{r}
# Read in data
fn_xl <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/ELISA_240221.xlsx")
o <- read_excel(fn_xl, sheet = "MouseData") %>% as.data.frame()

# Plot LI values
o$Group <- factor(o$Group, levels = c("Veh", "CAP"))
o$Timepoint <- factor(o$SacGroup, levels = c("Day 4", "Day 12"))
p <- ggplot(o, aes(x = Group, y = LipocalinPGperG)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  scale_color_manual(values = cap_palette_mouse) + 
  scale_fill_manual(values = cap_palette_mouse) + 
  facet_wrap(~Timepoint) + 
  ylab("Lipocalin-2 (pg/g colon contents)") + 
  theme_pubr() + 
  stat_compare_means(method = "wilcox", label.x = 1, label.y = 4.9, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "none") + 
  xlab("") + 
  scale_y_log10(limits = c(8*10^2,10^5)) + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

# Save
fn <- paste0(figdir, "Mech_LILcn2.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## Lcn-2 vs other metrics
```{r}
# Read in data
fn_xl <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/ELISA_240221.xlsx")
lcn <- read_excel(fn_xl, sheet = "MouseData") %>% as.data.frame()
lcn$Mouse <- lcn$MouseID
lcn <- lcn %>% select(-Group) %>% select(-SacGroup)

fn_xl <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/CAP_Tox_Mechanism.xlsx")
o <- read_excel(fn_xl, sheet = "OrganSize") %>% as.data.frame()
ps <- read_excel(fn_xl, sheet = "HotPlateLatency") %>% as.data.frame()
o$Group <- factor(o$Group, levels = c("Veh", "CAP")); ps$Group <- factor(ps$Group, levels = c("Veh", "CAP"))
o$Constant <- 1
ps_bl <- ps %>% filter(Timepoint == "Baseline") %>% select(Mouse, TimeS, Group) %>% dplyr::rename(BL_S = TimeS)
ps_mid <- ps %>% filter(Timepoint == "Midpoint") %>% select(Mouse, TimeS) %>% dplyr::rename(M_S = TimeS)
ps_end <- ps %>% filter(Timepoint == "Endpoint") %>% select(Mouse, TimeS) %>% dplyr::rename(E_S = TimeS)
ps_merge <- left_join(ps_bl, ps_mid, by = "Mouse")
ps_merge <- left_join(ps_merge, ps_end, by = "Mouse")
c <- left_join(ps_merge, o, by = c("Mouse", "Group"))

merged <- left_join(lcn, c, by = c("Mouse"))

c1 <- cor.test(merged$LipocalinPGperG, merged$WeightSacPerc, method = c("spearman"))
c2 <- cor.test(merged$LipocalinPGperG, merged$LILengthCM, method = c("spearman"))
c3 <- cor.test(merged$LipocalinPGperG, merged$M_S, method = c("spearman"))
c4 <- cor.test(merged$LipocalinPGperG, merged$E_S, method = c("spearman"))
merged_end <- merged %>% filter(SacGroup == "Endpoint")
c1_end <- cor.test(merged_end$LipocalinPGperG, merged_end$WeightSacPerc, method = c("spearman"))
c2_end <- cor.test(merged_end$LipocalinPGperG, merged_end$LILengthCM, method = c("spearman"))
```


## Stool DNA
How much DNA is present in stool? This is a proxy for bacterial load. Extracted then quantified with PicoGreen; details in "RA + CAP Mechanisms of Toxicity MGS" lab notebook in Benchling
```{r}
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/StoolDNA.csv")
dat <- read.csv(fn) %>% as.data.frame()
dat$Group <- factor(dat$Group, levels = c("Veh", "CAP"))
dat$Timepoint <- factor(dat$Time, levels = c("Day 0", "Day 4", "Day 12"))

# Plot
p <- ggplot(dat, aes(x = Group, y = DNA.in.stool..ng.mg.)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  scale_color_manual(values = cap_palette_mouse) + 
  scale_fill_manual(values = cap_palette_mouse) + 
  facet_wrap(~Timepoint) + 
  ylab("Stool DNA (ng/mg)") + 
  theme_pubr() + 
  stat_compare_means(method = "wilcox", label.x = 1, label.y = 700, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "none") + 
  ylim(c(0, 750)) + 
  xlab("") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

# Save
fn <- paste0(figdir, "Mech_StoolDNA.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)

# Plot
p1 <- ggplot(dat, aes(x = Timepoint, y = DNA.in.stool..ng.mg.)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  scale_color_manual(values = cap_palette_mouse) + 
  scale_fill_manual(values = cap_palette_mouse) + 
  facet_wrap(~Group) + 
  ylab("Stool DNA (ng/mg)") + 
  theme_pubr() + 
  stat_compare_means(method = "wilcox", label.x = 1, label.y = 700, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "none") + 
  ylim(c(0, 750)) + 
  xlab("") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

# Save
fn <- paste0(figdir, "Mech_StoolDNA_Flipped.pdf")
ggsave(filename = fn, plot = p1, width = 3, height = 3)
```

## Paw withdrawal latency (thermal hyperalgesia)
```{r}
# Read in data sheet from excel file
fn_xl <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/CAP_Tox_Mechanism.xlsx")
dat <- read_excel(fn_xl, sheet = "HotPlateLatency") %>% as.data.frame()
dat$Group <- factor(dat$Group, levels = c("Veh", "CAP"))
dat$Timepoint <- case_when(dat$Timepoint == "Baseline" ~ "Day 0",
                           dat$Timepoint == "Midpoint" ~ "Day 4",
                           TRUE ~ "Day 12")
dat$Timepoint <- factor(dat$Timepoint, levels = c("Day 0", "Day 4", "Day 12"))

# Plot
p <- ggplot(dat, aes(x = Group, y = TimeS)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  scale_color_manual(values = cap_palette_mouse) + 
  scale_fill_manual(values = cap_palette_mouse) + 
  facet_wrap(~Timepoint) + 
  ylab("Paw withdrawal latency (s)") + 
  theme_pubr() + 
  stat_compare_means(method = "wilcox", label.x = 1, label.y = 26, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "none") + 
  ylim(c(4, 27)) + 
  xlab("") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
fn <- paste0(figdir, "Mech_HFS.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)
```

## Colon length
```{r}
fn_xl <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/CAP_Tox_Mechanism.xlsx")
o <- read_excel(fn_xl, sheet = "OrganSize") %>% as.data.frame()

# Plot
o$Timepoint <- case_when(o$SacGroup == "Midpoint" ~ "Day 4",
                         TRUE ~ "Day 12")
o$Timepoint <- factor(o$Timepoint, levels = c("Day 4", "Day 12"))
o$Group <- factor(o$Group, levels = c("Veh", "CAP"))
p <- ggplot(o, aes(x = Group, y = LILengthCM)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, height = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  ylab("Colon length (cm)") + 
  scale_color_manual(values = cap_palette_mouse) + 
  scale_fill_manual(values = cap_palette_mouse) + 
  facet_wrap(~Timepoint) + 
  stat_compare_means(method = "wilcox", label.x = 1.2, label.y = 8.1, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "none") + 
  xlab("") + 
  ylim(c(5.8, 8.2))
fn <- paste0(figdir, "Mech_ColonLength.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)
```

## Metric correlations
```{r}
# Read in data
fn_xl <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/CAP_Tox_Mechanism.xlsx")
o <- read_excel(fn_xl, sheet = "OrganSize") %>% as.data.frame()
ps <- read_excel(fn_xl, sheet = "HotPlateLatency") %>% as.data.frame()
o$Group <- factor(o$Group, levels = c("Veh", "CAP")); ps$Group <- factor(ps$Group, levels = c("Veh", "CAP"))
o$Constant <- 1
ps_bl <- ps %>% filter(Timepoint == "Baseline") %>% select(Mouse, TimeS, Group) %>% dplyr::rename(BL_S = TimeS)
ps_mid <- ps %>% filter(Timepoint == "Midpoint") %>% select(Mouse, TimeS) %>% dplyr::rename(M_S = TimeS)
ps_end <- ps %>% filter(Timepoint == "Endpoint") %>% select(Mouse, TimeS) %>% dplyr::rename(E_S = TimeS)
ps_merge <- left_join(ps_bl, ps_mid, by = "Mouse")
ps_merge <- left_join(ps_merge, ps_end, by = "Mouse")
c <- left_join(ps_merge, o, by = c("Mouse", "Group"))

# Create correlation matrix
cor_matrix <- c %>% select(BL_S, M_S, E_S, LILengthCM, WeightSacPerc)
colnames(cor_matrix) <- c("Paw (Day 0)", "Paw (Day 4)", "Paw (Day 12)", "Colon length", "Weight")
cor_v <- cor(cor_matrix, use="complete.obs", method = "spearman") # complete.obs removes NA values
p.mat <- cor_pmat(cor_matrix, use="complete.obs", method = "spearman")
p <- ggcorrplot(cor_v, outline.col = "white", p.mat = p.mat, 
                lab = FALSE, hc.order = TRUE, method = "square", legend.title = "rho", digits = 1) + 
  theme_pubr() + 
  xlab("") + 
  ylab("") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + 
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", breaks=c(-1, 0, 1), limit=c(-1, 1)) + 
  labs(fill = "Spearman's\nrho") +
  theme(legend.title.align=0.5) +
  guides(fill = guide_colourbar(barwidth = 3.25, barheight = 1)) 

fn <- paste0(figdir, "Mech_CorrPlot.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)
```

## Metric correlations - new
```{r}
# Read in data
fn_xl <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/ELISA_240221.xlsx")
lcn <- read_excel(fn_xl, sheet = "MouseData") %>% as.data.frame()
lcn$Mouse <- lcn$MouseID
lcn <- lcn %>% select(-Group) %>% select(-SacGroup)

fn_xl <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/CAP_Tox_Mechanism.xlsx")
o <- read_excel(fn_xl, sheet = "OrganSize") %>% as.data.frame()
ps <- read_excel(fn_xl, sheet = "HotPlateLatency") %>% as.data.frame()
o$Group <- factor(o$Group, levels = c("Veh", "CAP")); ps$Group <- factor(ps$Group, levels = c("Veh", "CAP"))
o$Constant <- 1
ps_bl <- ps %>% filter(Timepoint == "Baseline") %>% select(Mouse, TimeS, Group) %>% dplyr::rename(BL_S = TimeS)
ps_mid <- ps %>% filter(Timepoint == "Midpoint") %>% select(Mouse, TimeS) %>% dplyr::rename(M_S = TimeS)
ps_end <- ps %>% filter(Timepoint == "Endpoint") %>% select(Mouse, TimeS) %>% dplyr::rename(E_S = TimeS)
ps_merge <- left_join(ps_bl, ps_mid, by = "Mouse")
ps_merge <- left_join(ps_merge, ps_end, by = "Mouse")
c <- left_join(ps_merge, o, by = c("Mouse", "Group"))
c <- left_join(lcn, c, by = c("Mouse"))

# Create correlation matrix
cor_matrix <- c %>% select(BL_S, M_S, E_S, LILengthCM, WeightSacPerc, LipocalinPGperG)
colnames(cor_matrix) <- c("Paw (Day 0)", "Paw (Day 4)", "Paw (Day 12)", "Colon length", "Weight", "Lcn-2")
cor_v <- cor(cor_matrix, use="complete.obs", method = "spearman") # complete.obs removes NA values
p.mat <- cor_pmat(cor_matrix, use="complete.obs", method = "spearman")
p <- ggcorrplot(cor_v, outline.col = "white", p.mat = p.mat, 
                lab = FALSE, hc.order = FALSE, method = "square", legend.title = "rho", digits = 1, 
                sig.level = 0.1, insig = c("blank")) + 
  theme_pubr() + 
  xlab("") + 
  ylab("") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + 
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", breaks=c(-1, 0, 1), limit=c(-1, 1)) +
  labs(fill = "Spearman's \n rho") +
  theme(legend.title.align=0.5) +
  guides(fill = guide_colourbar(barwidth = 3.25, barheight = 1)) +
  theme(legend.position = "none")

fn <- paste0(figdir, "Mech_CorrPlot_withLcn.pdf")
ggsave(filename = fn, plot = p, height = 2.7, width = 2.7)

# Export scalebar only
p <- ggcorrplot(cor_v, outline.col = "white", p.mat = p.mat, 
                lab = FALSE, hc.order = FALSE, method = "square", legend.title = "rho", digits = 1, 
                insig = c("blank"), sig.level = 0.1) + 
  theme_pubr() + 
  xlab("") + 
  ylab("") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + 
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", breaks=c(-1, 0, 1), limit=c(-1, 1)) +
  labs(fill = "Spearman's \n rho") +
  theme(legend.title.align=0.5) +
  guides(fill = guide_colourbar(barwidth = 3.25, barheight = 1))

fn <- paste0(figdir, "Mech_CorrPlot_withLcn_Scalebar.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)
```

## Body composition
```{r}
# Read in echoMRI data
folder_echo <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/EchoMRIData/")
echo = NULL
files <- paste0(folder_echo, list.files(folder_echo))
for (file in files){
  echo_in <- read_excel(file) %>% as.data.frame() %>% select("Label", "Fat", "Lean", "FreeWater", "TotalWater", "Weight")
  if (length(echo) == 0){
    echo <- echo_in
  } else {
    echo <- rbind(echo, echo_in)
  }
}

# Data wrangling
echo$Mouse <- gsub(".*_", "", echo$Label)
echo$Timepoint <- gsub("_.*", "", echo$Label)
echo$FatPercentage <- echo$Fat/echo$Weight*100
echo$LeanPercentage <- echo$Lean/echo$Weight*100
echo$FreeWaterPercentage <- echo$FreeWater/echo$Weight*100
echo$TotalWaterPercentage <- echo$TotalWater/echo$Weight*100
echo$Group <- case_when(as.numeric(echo$Mouse) <= 8 ~ "Veh",
                            TRUE ~ "CAP")
echo$Group <- factor(echo$Group, levels = c("Veh", "CAP"))
echo$Timepoint = case_when(echo$Timepoint == "BL" ~ "Day 0",
                           echo$Timepoint == "M" ~ "Day 4",
                           TRUE ~ "Day 12")

# Plot fat
p <- ggplot(echo, aes(x = Group, y = FatPercentage)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  scale_color_manual(values = cap_palette_mouse) + 
  scale_fill_manual(values = cap_palette_mouse) + 
  facet_wrap(~Timepoint) + 
  ylab("Fat percentage") + 
  theme_pubr() + 
  stat_compare_means(method = "wilcox", label.x = 1, label.y = 16.3, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "none") + 
  ylim(c(8, 17)) + 
  xlab("") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

fn <- paste0(figdir, "Mech_FatPercentage.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)

# Plot lean
p <- ggplot(echo, aes(x = Group, y = LeanPercentage)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  scale_color_manual(values = cap_palette_mouse) + 
  scale_fill_manual(values = cap_palette_mouse) + 
  facet_wrap(~Timepoint) + 
  ylab("Lean percentage") + 
  theme_pubr() + 
  stat_compare_means(method = "wilcox", label.x = 1, label.y = 86, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "none") + 
  ylim(c(75, 87)) + 
  xlab("") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

fn <- paste0(figdir, "Mech_LeanPercentage.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)
```

# Figure S8 - supplemental mechanisms of toxicity
```{r}
# Make directory
figdir <- paste0(dir, "FigureS8/")
dir.create(figdir)
```

## Body composition - echoMRI
```{r}
# Read in echoMRI data
folder_echo <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/EchoMRIData/")
echo = NULL
files <- paste0(folder_echo, list.files(folder_echo))
for (file in files){
  echo_in <- read_excel(file) %>% as.data.frame() %>% select("Label", "Fat", "Lean", "FreeWater", "TotalWater", "Weight")
  if (length(echo) == 0){
    echo <- echo_in
  } else {
    echo <- rbind(echo, echo_in)
  }
}

# Data wrangling
echo$Mouse <- gsub(".*_", "", echo$Label)
echo$Timepoint <- gsub("_.*", "", echo$Label)
echo$FatPercentage <- echo$Fat/echo$Weight*100
echo$LeanPercentage <- echo$Lean/echo$Weight*100
echo$FreeWaterPercentage <- echo$FreeWater/echo$Weight*100
echo$TotalWaterPercentage <- echo$TotalWater/echo$Weight*100
echo$Group <- case_when(as.numeric(echo$Mouse) <= 8 ~ "Veh",
                            TRUE ~ "CAP")
echo$Group <- factor(echo$Group, levels = c("Veh", "CAP"))
echo$Timepoint = case_when(echo$Timepoint == "BL" ~ "Day 0",
                           echo$Timepoint == "M" ~ "Day 4",
                           TRUE ~ "Day 12")
echo$Timepoint <- factor(echo$Timepoint, levels = c("Day 0", "Day 4", "Day 12"))

# Plot fat
p <- ggplot(echo, aes(x = Group, y = FatPercentage)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  scale_color_manual(values = cap_palette_mouse) + 
  scale_fill_manual(values = cap_palette_mouse) + 
  facet_wrap(~Timepoint) + 
  ylab("Fat percentage") + 
  theme_pubr() + 
  stat_compare_means(method = "wilcox", label.x = 1, label.y = 16.3, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "none") + 
  ylim(c(8, 17)) + 
  xlab("") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

fn <- paste0(figdir, "Mech_FatPercentage.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)

# Plot lean
p <- ggplot(echo, aes(x = Group, y = LeanPercentage)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  scale_color_manual(values = cap_palette_mouse) + 
  scale_fill_manual(values = cap_palette_mouse) + 
  facet_wrap(~Timepoint) + 
  ylab("Lean percentage") + 
  theme_pubr() + 
  stat_compare_means(method = "wilcox", label.x = 1, label.y = 86, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "none") + 
  ylim(c(75, 87)) + 
  xlab("") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

fn <- paste0(figdir, "Mech_LeanPercentage.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)
```

## Gonadal fat pads
```{r}
fn_xl <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/CAP_Tox_Mechanism.xlsx")
o <- read_excel(fn_xl, sheet = "OrganSize") %>% as.data.frame()

# Plot
o$Timepoint <- case_when(o$SacGroup == "Midpoint" ~ "Day 4",
                         TRUE ~ "Day 12")
o$Timepoint <- factor(o$Timepoint, levels = c("Day 4", "Day 12"))
o$Group <- factor(o$Group, levels = c("Veh", "CAP"))
o$GonadalFatPercent <- 100*o$GonadalFatMG/(1000*o$WeightSacG)
p <- ggplot(o, aes(x = Group, y = GonadalFatPercent)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, height = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  ylab("gWAT weight/body weight (%)") + 
  scale_color_manual(values = cap_palette_mouse) + 
  scale_fill_manual(values = cap_palette_mouse) + 
  facet_wrap(~Timepoint) + 
  stat_compare_means(method = "wilcox", label.x = 1.2, label.y = 2, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "none") + 
  xlab("") + 
  ylim(c(0.75, 2.1))
fn <- paste0(figdir, "Mech_FatPad.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)
```

## SI length
```{r}
fn_xl <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/CAP_Tox_Mechanism.xlsx")
o <- read_excel(fn_xl, sheet = "OrganSize") %>% as.data.frame()

# Plot
o$Timepoint <- case_when(o$SacGroup == "Midpoint" ~ "Day 4",
                         TRUE ~ "Day 12")
o$Timepoint <- factor(o$Timepoint, levels = c("Day 4", "Day 12"))
o$Group <- factor(o$Group, levels = c("Veh", "CAP"))
p <- ggplot(o, aes(x = Group, y = SILengthCM)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, height = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  ylab("Small intestine length (cm)") + 
  scale_color_manual(values = cap_palette_mouse) + 
  scale_fill_manual(values = cap_palette_mouse) + 
  facet_wrap(~Timepoint) + 
  stat_compare_means(method = "wilcox", label.x = 1.2, label.y = 36, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "none") + 
  xlab("") + 
  ylim(c(29, 37))
fn <- paste0(figdir, "Mech_SI.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)
```

## Spleen weight
```{r}
fn_xl <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/CAP_Tox_Mechanism.xlsx")
o <- read_excel(fn_xl, sheet = "OrganSize") %>% as.data.frame()

# Plot
o$Timepoint <- case_when(o$SacGroup == "Midpoint" ~ "Day 4",
                         TRUE ~ "Day 12")
o$Timepoint <- factor(o$Timepoint, levels = c("Day 4", "Day 12"))
o$Group <- factor(o$Group, levels = c("Veh", "CAP"))
o$SpleenWeightPercent <- 100*o$SpleenMG/(1000*o$WeightSacG)
p <- ggplot(o, aes(x = Group, y = SpleenWeightPercent)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, height = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  ylab("Spleen weight/body weight (%)") + 
  scale_color_manual(values = cap_palette_mouse) + 
  scale_fill_manual(values = cap_palette_mouse) + 
  facet_wrap(~Timepoint) + 
  stat_compare_means(method = "wilcox", label.x = 1.2, label.y = 0.57, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "none") + 
  ylim(c(0.28, 0.59))
  xlab("")
fn <- paste0(figdir, "Mech_Spleen.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)
```

# Figure S9 - CAP alters microbiome mice
```{r}
# Make directory
figdir <- paste0(dir, "FigureS9/")
dir.create(figdir)

# Set directory
tabledir = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/SequencingData/")

# Load metadata
meta_fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Tox_Model/SequencingData/Metadata.csv")
metadata <- read.csv(meta_fn)
metadata$Group <- factor(metadata$Group, levels = c("Veh", "CAP"))
metadata$PrePost <- factor(metadata$PrePost, levels = c("Pre", "Post"))
metadata$Legend <- case_when(metadata$Time == "D4" & metadata$Group == "CAP" ~ "CAP, Day 4",
                             metadata$Time == "D4" ~ "Veh, Day 4",
                             metadata$Group == "Veh" ~ "Veh, Day 0",
                             TRUE ~ "CAP, Day 0")
```

## Alpha diversity
```{r}
# Read in data - we'll look at species
fn <- paste0(tabledir, "TaxaSpecies.csv")
t <- read.csv(fn)

# Calculate diversity
div <- vegan::diversity(t(t %>% select(-Taxa)), index = "shannon")
df <- reshape2::melt(div, by = NULL)
colnames(df) <- "Shannon"; df$Sample <- rownames(df)
df <- left_join(df, metadata, by = "Sample")
df$Days <- as.factor(df$Days)

# Plot - days
p <- ggplot(df, aes(x = Days, y = Shannon)) + 
  geom_boxplot(aes(group = Days, fill = Legend), alpha = 0.5) + 
  geom_point(aes(color = Legend)) +
  theme_pubr() +
  facet_wrap(~Group) + 
  theme(legend.position = "none") + 
  stat_compare_means(method = "wilcox", label.x = 1.2, label.y = 3.9, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  scale_color_manual(values = cmot_palette) + 
  scale_fill_manual(values = cmot_palette) +
  ylim(1.9, 4)

# Save
fn <- paste0(figdir, "AlphaDays.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)
```

## PERMANOVA 
```{r}
rerun_permanova = FALSE

# Read in data
fn <- paste0(tabledir, "TaxaGenus.csv")
d <- fread(fn) %>% as.data.frame()
rownames(d) <- d$Taxa; d <- d %>% select(-Taxa) 

# CLR transform
geometric_means <- apply(d, 1, function(row) exp(mean(log(row[row > 0]))))
pseudo_count = min(d[d > 0])/2
clr_data <- log((d + pseudo_count) / geometric_means)

# Omit NA values
clr_data <- na.omit(clr_data)

# Remove zero-variance columns
clr_data_t <- t(clr_data)
clr_data_t <- clr_data_t[ , which(apply(clr_data_t, 2, var) != 0)]
d_t <- t(d)
d_t <- d_t[ , which(apply(d_t, 2, var) != 0)]

# Concatenate metadata
meta_c <- metadata %>% filter(Sample %in% rownames(clr_data_t) & Sample %in% rownames(d_t)) # only keep samples that we have sequencing data for
clr_data_t <- clr_data_t[meta_c$Sample,] # and vice versa for metadata matching
d_t <- d_t[meta_c$Sample,]

# PERMANOVA testing 
meta_perm <- meta_c 
row_order <- meta_perm %>% select(Sample) %>% pull(); clr_data_ordered <-clr_data_t[row_order,]; d_ordered <- d_t[row_order,] # arrange
if(sum(meta_perm$Sample == rownames(clr_data_ordered)) + sum(meta_perm$Sample == rownames(d_ordered)) != 2*length(meta_perm$Sample)){ # check that they match
    print("PERMANVA order wrong!")
}
if (rerun_permanova){
  dist_clr <- suppressWarnings(vegdist(clr_data_ordered, method="euclidean", na.rm = TRUE, binary = TRUE)) # CLR-Euclidean
  dist_bray <- vegdist(d_ordered, method="bray", na.rm = TRUE) # Bray
  set.seed(42)
  res_bray <- adonis2(dist_bray ~ Legend, data = meta_perm, permutations=1000, strata = meta_perm$Mouse, by = "margin")
  res_clr <- adonis2(dist_clr ~ Legend, data = meta_perm, permutations=1000, strata = meta_perm$Mouse, by = "margin")
}
```

## PCoA
```{r}
# Pick axes
axis1 = 1; axis2 = 2

# Read in data - we'll look at genera
fn <- paste0(tabledir, "TaxaGenus.csv")
t <- read.csv(fn)

# CLR transform
rownames(t) <- t$Taxa; t <- t %>% select(-Taxa)
t <- clr_transform(t)

# PCA
df <- t(t) %>% as.data.frame()
df$Sample <- rownames(df)
pca_res <- PCA(df[, -which(names(df) %in% c("Sample"))], scale.unit = FALSE, graph = FALSE)

pca_scores <- as.data.frame(pca_res$ind$coord)
pca_scores$Sample <- df$Sample
pca_scores$Group <- df$Group
percent_var <- pca_res$eig[axis1:axis2, "percentage of variance"] # percent variance explained

pca_plot <- left_join(pca_scores, metadata, by = "Sample")

p <- ggplot(pca_plot, aes(x = .data[[paste0("Dim.", axis1)]], y = .data[[paste0("Dim.", axis2)]])) +
  geom_point(aes(color = Legend), size = 2) +
  theme_pubr() +
  labs(x = paste("PCo", axis1, " (", sprintf("%.2f", percent_var[1]), "%)", sep = ""),
       y = paste("PCo", axis2, " (", sprintf("%.2f", percent_var[2]), "%)", sep = "")) +
  theme(legend.position = "right") +
  scale_color_manual(values = cmot_palette)  +
  annotate("text", x = -25, y = 25, label = "PERMANOVA:~italic(p) < 0.0001 * ',' ~ R^2 == 0.18", size = 3.5, hjust = 0, parse = TRUE)

# Save
fn <- paste0(figdir, "PCoA.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 5.5)
```

## Volcano - veh/CAP
```{r}
taxLevel = "Genus"
treatment = "Veh" # CAP or Veh

fn <- paste0(tabledir, taxLevel, "PrePost_", treatment, ".csv")
u <- read.csv(fn) %>% as.data.frame()
u <- u %>% filter(Group == "PrePostPost")
u$log2FC <- log2(exp(u$Value))
u$FDR <- p.adjust(u$p.value, method = "BH")
u$Group <- treatment
u_veh <- u

treatment = "CAP" 
fn <- paste0(tabledir, taxLevel, "PrePost_", treatment, ".csv")
u <- read.csv(fn) %>% as.data.frame()
u <- u %>% filter(Group == "PrePostPost")
u$log2FC <- log2(exp(u$Value))
u$FDR <- p.adjust(u$p.value, method = "BH")
u$Group <- treatment
u_cap <- u

u <- rbind(u_veh, u_cap)
u$Group <- factor(u$Group, levels = c("Veh", "CAP"))

# Set the FDR cutoff value
fdr_cutoff <- 0.2

# Count number altered
num <- u %>% filter(FDR < fdr_cutoff) %>% select(Taxa) %>% count()
num_up <- u %>% filter(FDR < fdr_cutoff) %>% filter(Value > 0) %>% select(Taxa) %>% count()
x_up <- u %>% filter(Value > 0) %>% select(Value) %>% pull() %>% max()/2
x_up <- exp(log2(x_up))
num_down <- u %>% filter(FDR < fdr_cutoff) %>% filter(Value < 0) %>% select(Taxa) %>% count()
x_down <- u %>% filter(Value < 0) %>% select(Value) %>% pull() %>% min()/2
x_down <- -exp(log2(-x_down))
ymin <- u %>% select(FDR) %>% pull() %>% min(); yval <- -log10(ymin/4); ylim <- -log10(ymin/4)

# Create a volcano plot
p <- ggplot(data = u, aes(x = log2FC, y = -log10(FDR))) +
  geom_point(alpha = 1, shape = 16, aes(color = ifelse(FDR <= fdr_cutoff, ifelse(Value > 0, "Enriched", "Depleted"), "Not Significant")), size = 1.5) +
  theme_pubr() +
  facet_wrap(~Group) + 
  labs(x = expression(paste("Treatment effect (log2FC/baseline)")), y = "-log10(FDR)", color = "Genus") +
  scale_color_manual(values = enriched_palette) +
  theme(legend.position = "left") +
  annotate("text", x=x_up, y=yval, label = paste0(num_up, "")) + 
  annotate("text", x=x_down, y=yval, label = paste0(num_down, "")) +
  ylim(0, ylim) 

# Save
fn <- paste0(figdir, "Volcano_Taxa.pdf")
ggsave(filename = fn, plot = p, width = 5.5, height = 3)
```

## Bargraph - CAP
```{r}
# Read in
taxLevel = "Genus"
treatment = "CAP" # CAP or Veh
fn <- paste0(tabledir, taxLevel, "PrePost_", treatment, ".csv")
u <- read.csv(fn) %>% as.data.frame()
u <- u %>% filter(Group == "PrePostPost")
u$log2FC <- log2(exp(u$Value))
u$FDR <- p.adjust(u$p.value, method = "BH")
fdr_cutoff <- 0.2
v <- u %>% filter(FDR < fdr_cutoff)
v$Direction <- case_when(v$Value > 0 ~ "Enriched",
                         TRUE ~ "Depleted")

# Extract order
v$Order <- gsub(".*\\|o__", "", v$Taxa)
v$Order <- gsub("\\|f__.*", "", v$Order )

# Remove orders that are unknown/unclassified
v$Order <- case_when(grepl("OFGB|unclassified", v$Order) ~ "Unclassified/OFGB",
                     TRUE ~ v$Order)

# Mutate
# Calculate the counts for each Order within each Direction
order_counts <- v %>%
  group_by(Direction, Order) %>%
  summarise(Count = n(), .groups = 'drop') %>%
  arrange(Direction, desc(Count))

# Use the arrange() results to reorder Order within each Direction
v <- v %>%
  mutate(Order = factor(Order, levels = unique(order_counts$Order)))

# Plot
p <- ggplot(v, aes(x = Direction, fill = Order)) +
  geom_bar(width = 0.7, position = "stack", color = "black") +
  labs(y = "Number of genera", x = "") +
  theme_pubr() +
  theme(legend.position = "right") +
  theme(legend.key.size = unit(0.4, "cm")) +
  theme(legend.title = element_text(size = 11), 
               legend.text = element_text(size = 10)) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Save
fn <- paste0(figdir, "OrderBarplot.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)
```

## Boxplots of all significant taxa - CAP
```{r}
# Set taxa of interest (toi)
taxLevel = "Genus"
treatment = "CAP" # CAP or Veh
fn <- paste0(tabledir, taxLevel, "PrePost_", treatment, ".csv")
u <- read.csv(fn) %>% as.data.frame()
u <- u %>% filter(Group == "PrePostPost")
u$FDR <- p.adjust(u$p.value, method = "BH")
fdr_cutoff <- 0.2
v <- u %>% filter(FDR < fdr_cutoff)
toi <- v$Taxa

# Make direction dataframe
dirs <- v %>% select(Taxa, Value, BaselineGroup)
dirs$Direction = case_when(dirs$Value > 0 ~ "Enriched",
                                   dirs$Value < 0 ~ "Depleted",
                                   TRUE ~ NA)

# Read in data
taxa_fn <- paste0(tabledir, "Taxa", taxLevel, ".csv") # taxa filename
t_in <- read.csv(taxa_fn) %>% 
  as.data.frame()
rownames(t_in) <- t_in$Taxa; t_in <- t_in %>% select(-Taxa)
t_in <- clr_transform(t_in)
t_in$Taxa <- rownames(t_in)
t_in <- t_in %>% filter(Taxa %in% toi)

# Melt
df <- t_in %>% reshape2::melt(id.vars = "Taxa")
colnames(df) <- c("Taxa", "Sample", "Abundance")
df <- inner_join(df, metadata, by = "Sample")
df <- df %>% filter(Group == "CAP")
df$Days <- as.factor(df$Days)

# Merge
df <- left_join(df, dirs, by = "Taxa")

# Plot
df$Taxa <- gsub(".*\\|g__", "", df$Taxa) # shorten names
df$Legend <- factor(df$Legend, levels = c("CAP, Day 4", "CAP, Day 0")) # reorder
df <- df %>% arrange(BaselineGroup)
df$Taxa <- factor(df$Taxa, levels = unique(df$Taxa))
p <- ggplot(df, aes(x = Abundance, y = Taxa, fill = Legend)) + 
  geom_boxplot(alpha = 0.5, outlier.size = -1) + 
  geom_point(aes(color = Legend), position = position_dodge(width = 0.75), size = 1) + 
  theme_pubr() +
  xlab("Abundance (CLR)") + 
  theme(legend.position = "top") +
  ylab("Genus") + 
  facet_wrap(~Direction, scale = "free_y", nrow = 1) + 
  scale_color_manual(values = cmot_palette) +
  scale_fill_manual(values = cmot_palette) 

# Save
fn <- paste0(figdir, "Boxplots.pdf")
ggsave(filename = fn, plot = p, width = 9, height = 8)
```

# Figure 4 - Microbiome depletion ~ tox
```{r}
# Make directory
figdir <- paste0(dir, "Figure4/")
dir.create(figdir)
```

## SPF GF weight
Weight loss - SPF vs GF
```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_vs_GF/KT_CAP3.xlsx")
w <- read_excel(fn, sheet = "TractableWeightsNormalized") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w$Group <- case_when(grepl("9|10|11|12|13|14|15|16", w$MouseID) ~ "CONV-D",
                     grepl("17|18|19|20|21|22|23", w$MouseID) ~ "CONV-R",
                     TRUE ~ "GF")
w$Group <- factor(w$Group, levels = c("GF", "CONV-D", "CONV-R"))

# Mixed-effects model p value - group*time. The interaction term is significant (p < 0.0001)
w_model <- w %>% filter(!is.na(Weight))
m1 <- lme(Weight ~ Group*Day, random = ~ 1 | MouseID, data=w_model)
summary(m1)
p_value <- summary(m1)$tTable["GroupCONV-D:Day", 5]

# Two-way anova testing
res.aov2 <- aov(Weight ~ Group*Day, data = w_model)
p_int <- summary(res.aov2)[[1]][["Pr(>F)"]][[3]] # Interaction p value
p_value <- summary(res.aov2)[[1]][["Pr(>F)"]][[1]] # Group (i.e. colonization) p value

# Plot
w$Group <- factor(w$Group, levels = c("CONV-R", "GF", "CONV-D"))
p <- ggplot(w, aes(x = Day, y = Weight, group = Group, color = Group, shape = Group, fill = Group)) + 
  stat_summary(fun="mean", geom="line") + 
  stat_summary(fun="mean", geom="point") + 
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.2) + 
  xlab("CAP duration (days)") + 
  ylab("Weight vs baseline (%)") + 
  geom_jitter(width = 0.1, alpha = 0.2) + 
  theme_pubr() + 
  scale_color_manual(values = conv_palette) + 
  scale_fill_manual(values = conv_palette) + 
  annotate("text", x=2.5, y=87, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) + 
   theme(legend.title = element_blank()) + 
  scale_x_continuous(breaks = seq(0,9,by=3)) + 
  theme(legend.text=element_text(size=9)) # decrease font size to fit

# Save
fn <- paste0(figdir, "SPFGF_Weights.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## SPF GF survival
Survival - SPF vs GF
```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_vs_GF/KT_CAP3.xlsx")
w <- read_excel(fn, sheet = "Survival") %>% as.data.frame() %>% dplyr::rename(Day = DayPostCAP, Event = Status)
w$Group <- case_when(grepl("9|10|11|12|13|14|15|16", w$MouseID) ~ "CONV-D",
                     grepl("17|18|19|20|21|22|23", w$MouseID) ~ "CONV-R",
                     TRUE ~ "GF")
w$Group <- factor(w$Group, levels = c("CONV-R", "GF", "CONV-D"))

# object generation
surv_obj <- with(w, Surv(Day, Event))
fit <- survfit(Surv(Day, Event) ~ Group, data = w)
surv_df <- fortify(fit)

# Analysis
log_rank_test <- survdiff(Surv(Day, Event) ~ Group, data = w)
p_value <- 1 - pchisq(log_rank_test$chisq, df = 1) # Adjust the degrees of freedom if needed

# Plot the survival curve
ggsurv <- ggsurvplot(fit, data = w, pval = FALSE, conf.int = FALSE,
           risk.table = FALSE, xlim = c(0, 14)) #, linetype = "strata")
p <- ggsurv$plot
p <- p + scale_color_manual(values = conv_surv_palette) +
  xlab("CAP duration (days)") + 
  ylab("Survival probability") + 
  theme_pubr() + 
  annotate("text", x = 3, y = 0.1, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) + 
  theme(legend.position = "none") 

# Save
fn <- paste0(figdir, "SPFGF_Survival.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## SPF GF endpoint weight
```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_vs_GF/KT_CAP3.xlsx")
w <- read_excel(fn, sheet = "TractableWeightsNormalized") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w$Group <- case_when(grepl("9|10|11|12|13|14|15|16", w$MouseID) ~ "CONV-D",
                     grepl("17|18|19|20|21|22|23", w$MouseID) ~ "CONV-R",
                     TRUE ~ "GF")

# Plot
w$Group <- factor(w$Group, levels = c("CONV-R", "GF", "CONV-D"))
w <- w %>% filter(Day == 9)
w$Day <- "Day 9"

p <- ggplot(w, aes(x = Group, y = Weight)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  ylab("Weight vs baseline (%)") + 
  scale_color_manual(values = conv_palette) + 
  scale_fill_manual(values = conv_palette) + 
  facet_wrap(~Day) + 
  stat_compare_means(method = "kruskal", label.x = 1.5, label.y = 102, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "none") + 
  ylim(c(82, 103)) + 
  xlab("")

fn <- paste0(figdir, "SPFGF_EndpointWeight.pdf")
ggsave(filename = fn, plot = p, width = 2.5, height = 3)
```


## AVNM Fat pad mass - FIX
No correlation between gonadal fat % (converted to body fat % using regression) and endpoint weight loss (for mice that made it to day 14)

https://www.degruyter.com/document/doi/10.1515/biol-2015-0010/html?lang=en:
 - Gonadal fat is associated with 
 - Regression to calculate total body fat from gonadal fat pad: Fat percent = Gonadal fat percent * 4.931 - 0.454
```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/AVNM_Model/SPF_CAP_Abx_Nov23.xlsx")
meta <- read_excel(fn, sheet = "Weights", skip = 1) %>% as.data.frame()
ce <- read_excel(fn, sheet = "FatPads") %>% as.data.frame() %>% dplyr::rename(Tube = Mouse)
ce <- left_join(ce, meta, by = "Tube")
ce$Group <- factor(ce$Group, levels = c("Veh", "AVNM"))

# Analysis
cor.test.result <- cor.test(ce$FatPadWeightMG/ce$D0, ce$D14/ce$D0, method = "spearman")
correlation <- cor.test.result$estimate["rho"]
p_value <- cor.test.result$p.value
text_label <- sprintf("Spearman's rho = %.2f \np = %.2f", correlation, p_value)

# Graph. Invert the "WeightLoss" variable, which is confusingly the % weight loss (so it's positive if the mouse lost weight)
p <- ggplot(ce, aes(x = 100*FatPadWeightMG/(1000*D0), y = 100*D14/D0)) + 
  geom_point() + 
  theme_pubr() + 
  xlab("Endpoint gonadal fat/baseline weight (%)") + 
  ylab("Weight vs baseline (%)") + 
  theme(legend.position = "none") + 
  geom_smooth(method = "lm", se = TRUE, color = "blue") + 
  annotate("text", x=1.25, y=104, label = text_label) 

# Save
fn <- paste0(figdir, "AVNM_FatPadMass.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## AVNM weight
p value: two way anova, Group*Day term

```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/AVNM_Model/SPF_CAP_Abx_Nov23.xlsx")
w <- read_excel(fn, sheet = "WeightsNorm") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w$Group <- case_when(as.numeric(w$MouseID) <= 12 ~ "AVNM",
                     TRUE ~ "Veh")

# Mixed-effects model p value - group*time. The interaction term is significant (p < 0.0001)
w_model <- w %>% filter(!is.na(Weight))
m1 <- lme(Weight ~ Group*Day, random = ~ 1 | MouseID, data=w_model)
summary(m1)
p_value <- summary(m1)$tTable["GroupVeh:Day", 5]

# Two-way anova testing
res.aov2 <- aov(Weight ~ Group*Day, data = w_model)
p_int <- summary(res.aov2)[[1]][["Pr(>F)"]][[3]] # Interaction p value
p_value <- summary(res.aov2)[[1]][["Pr(>F)"]][[1]] # Group (i.e. colonization) p value

# Plot
w$Group <- factor(w$Group, levels = c("Veh", "AVNM"))
p <- ggplot(w, aes(x = Day, y = Weight, group = Group, color = Group, shape = Group, fill = Group)) + 
  stat_summary(fun="mean", geom="line") + 
  stat_summary(fun="mean", geom="point") + 
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.2) + 
  xlab("CAP duration (days)") + 
  ylab("Weight vs baseline (%)") + 
  geom_jitter(width = 0.1, alpha = 0.2) + 
  theme_pubr() + 
  scale_color_manual(values = avnm_palette) + 
  scale_fill_manual(values = avnm_palette) + 
  annotate("text", x=1.5, y=85, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) + 
  theme(legend.title = element_blank())

# Save
fn <- paste0(figdir, "AVNM_Weights.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## AVNM endpoint weight
```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/AVNM_Model/SPF_CAP_Abx_Nov23.xlsx")
w <- read_excel(fn, sheet = "WeightsNorm") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w$Group <- case_when(as.numeric(w$MouseID) <= 12 ~ "AVNM",
                     TRUE ~ "Veh")

# Plot
w$Group <- factor(w$Group, levels = c("Veh", "AVNM"))
w <- w %>% filter(Day == 5)
w$Day <- "Day 5"
p <- ggplot(w, aes(x = Group, y = Weight)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5, width = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  ylab("Weight vs baseline (%)") + 
  scale_color_manual(values = avnm_palette) + 
  scale_fill_manual(values = avnm_palette) + 
  facet_wrap(~Day) + 
  stat_compare_means(method = "kruskal", label.x = 1.2, label.y = 103, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "none") + 
  ylim(c(80, 104)) + 
  xlab("")

fn <- paste0(figdir, "AVNM_EndpointWeight.pdf")
ggsave(filename = fn, plot = p, width = 2.5, height = 3)
```

## AVNM survival
```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/AVNM_Model/SPF_CAP_Abx_Nov23.xlsx")
w <- read_excel(fn, sheet = "Survival") %>% as.data.frame() %>% dplyr::rename(MouseID = Mouse, Day = Time)
w$Group <- case_when(as.numeric(w$MouseID) <= 12 ~ "AVNM",
                     TRUE ~ "Veh")

# object generation
surv_obj <- with(w, Surv(Day, Event))
fit <- survfit(Surv(Day, Event) ~ Group, data = w)
surv_df <- fortify(fit)

# Analysis
log_rank_test <- survdiff(Surv(Day, Event) ~ Group, data = w)
p_value <- 1 - pchisq(log_rank_test$chisq, df = 1) # Adjust the degrees of freedom if needed

# Plot the survival curve
ggsurv <- ggsurvplot(fit, data = w, pval = FALSE, conf.int = FALSE,
           risk.table = FALSE, xlim = c(0, 14))
p <- ggsurv$plot
p <- p + scale_color_manual(values = avnm_surv_palette) +
  xlab("CAP duration (days)") + 
  ylab("Survival probability") + 
  theme_pubr() + 
  annotate("text", x = 3, y = 0.1, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) + 
  theme(legend.position = "none")

# Save
fn <- paste0(figdir, "AVNM_Survival.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```


# Figure S10 - validation of microbiota depletion
```{r}
# Make directory
figdir <- paste0(dir, "FigureS10/")
dir.create(figdir)
```

## AVNM CFU
```{r}
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/AVNM_Model/CFU.csv")
dat <- read.csv(fn)
dat$Group <- case_when(as.numeric(dat$MouseID) <= 12 ~ "AVNM",
                       TRUE ~ "Veh")
dat$Group <- factor(dat$Group, levels = c("Veh", "AVNM"))

# Calculate p value
x <- dat %>% filter(Group == "Veh") %>% select(CFU.g) %>% pull()
y <- dat %>% filter(Group == "AVNM") %>% select(CFU.g) %>% pull()
p_value = wilcox.test(x,y)[[3]]

# Graph
p <- ggplot(dat, aes(x = Group, y = CFU.g)) + 
  geom_boxplot(aes(fill = Group), alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  xlab("") + 
  ylab("Colonization (CFU/g)") + 
  scale_y_log10(limits = c(10, 100000000000)) + 
  scale_color_manual(values = avnm_palette) + 
  scale_fill_manual(values = avnm_palette) + 
  theme(legend.position = "none") + 
  annotate("text", x=2, y = 10, label = "n.d.") + 
  annotate("text", x=1.5, y=100000000000, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE)

fn <- paste0(figdir, "AVNM_Colonization.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)
```

## AVNM Cecal mass
```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/AVNM_Model/SPF_CAP_Abx_Nov23.xlsx")
meta <- read_excel(fn, sheet = "Weights", skip = 1) %>% as.data.frame()
ce <- read_excel(fn, sheet = "CecalWeight") %>% as.data.frame()
colnames(ce) <- c("Tube", "Weight")
ce <- left_join(ce, meta, by = "Tube")
ce$Group <- factor(ce$Group, levels = c("Veh", "AVNM"))

# Graph
p <- ggplot(ce, aes(x = Group, y = Weight)) + 
  geom_boxplot(aes(fill = Group), alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  xlab("") + 
  ylab("Cecal contents weight (mg)") + 
  stat_compare_means(method = "wilcox", label.x=1.1, label.y = 1450, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  scale_color_manual(values = avnm_palette) + 
  scale_fill_manual(values = avnm_palette) + 
  theme(legend.position = "none") + 
  scale_y_continuous(breaks = seq(0, 1400, by = 200), limits = c(0, 1500)) 

# Save
fn <- paste0(figdir, "AVNM_CecalMass.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)
```

## SPF Rescue Colonization
```{r}
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/EColiCFU.csv")
t <- read.csv(fn) %>% as.data.frame()
t$Strain <- factor(t$Strain, levels = c("KO", "WT", "OE"))
t <- t %>% filter(CFUperG > 10) # clear technical outliar

p2 <- ggplot(t %>% filter(Experiment == "Tox"), aes(x = Strain, y = log10(CFUperG), color = Strain, fill = Strain)) + 
  geom_point(position = position_dodge(width = 0.75)) +
  geom_boxplot(position = position_dodge(width = 0.75), alpha = 0.5, color = "black") + 
  theme_pubr() + 
  scale_color_manual(values = strain.colors) + 
  scale_fill_manual(values = strain.colors) + 
  xlab("Strain") + 
  ylab(expression("Colonization (log"[10]~"CFU/g)"))+ 
  theme(legend.position = "none") + 
  ylim(5, 10.5) + 
  stat_compare_means(method = "kruskal.test", label.x = 1.5, label.y = 10.2, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE)

fn <- paste0(figdir, "ToxCFU.pdf")
ggsave(filename = fn, plot = p2, width = 2, height = 2.5)
```

# Figure 5 - preTA rescues toxicity
```{r}
# Make directory
figdir <- paste0(dir, "Figure5/")
dir.create(figdir)

# Sequencing outliers to remove for preta hi/lo experiment - mouse #8 "preTA_HiLo_8" had errors when run through the MGS pipeline
mice_remove <- c("8")
```

## PK CFU
```{r}
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/EColiCFU.csv")
t <- read.csv(fn) %>% as.data.frame()
t$Strain <- factor(t$Strain, levels = c("KO", "WT", "OE"))

# Remove outliar
t <- t %>% filter(CFUperG > 10)

# PK CFU, with only the included mice
p3 <- ggplot(t %>% filter(Experiment == "PK") %>% filter(Sample %in% c("PK1", "PK2", "PK3", "PK4", "PK5", "PK6", "PK13", "PK14", "PK15", "PK16", "PK17", "PK18")), 
             aes(x = Strain, y = log10(CFUperG), fill = Strain)) + 
  geom_point(position = position_dodge(width = 0.75), aes(color = Strain)) +
  geom_boxplot(position = position_dodge(width = 0.75), alpha = 0.5) + 
  theme_pubr() + 
  scale_color_manual(values = strain.colors) + 
  scale_fill_manual(values = strain.colors) + 
  xlab("") + 
  ylab(expression("Colonization (log"[10]~"CFU/g)"))+ 
  theme(legend.position = "none") + 
  ylim(8.6, 9.8) + 
  stat_compare_means(method = "kruskal.test", label.x = 1.25, label.y = 9.75, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE)

fn <- paste0(figdir, "PKCFUSub.pdf")
ggsave(filename = fn, plot = p3, width = 1.5, height = 2.5)
```


## PK experiment
```{r}
# Read in data for standard curves
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Stool_PK/data_240725.xlsx")
t_in <- read_excel(fn, sheet = "processed_data") %>% as.data.frame()

# Normalize
t <- t_in
t$`5FU` <- t$`5FU`/t$CAP_IS_4; t$CAP <- t$CAP/t$CAP_IS_4
t$`5FU` <- t$`5FU`/t$StoolWeightMG; t$CAP <- t$CAP/t$StoolWeightMG

# Remove blanks and IS-only samples
t <- t %>% filter(SampleID != "blank") %>% filter(SampleID != "11") %>% filter(SampleID != "10")
t$CAPperStool_True = 1000*t$Added_CAP_mg/t$StoolWeightMG; t$FUperStool_True <- 1000*t$Added_5FU_mg/t$StoolWeightMG
t$CAP = t$CAP/t$StoolWeightMG; t$`5FU` <- t$`5FU`/t$StoolWeightMG
model_CAP <- lm(CAPperStool_True ~ CAP, data = t %>% filter(!(SampleID %in% c("1", "2", "7", "8", "9", "10", "11")))) # remove saturated samples and no signal samples
model_5FU <- lm(FUperStool_True ~ `5FU`, data = t %>% filter(!(SampleID %in% c("1", "2", "6", "7", "8", "9", "10", "11"))))
t$predict_CAP <- predict(model_CAP, newdata = t)
t$predict_5FU <- predict(model_5FU, newdata = t)

# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Stool_PK/240828_kai_PreTA_df10.txt")
t_in <- read.csv(fn, sep = "\t") %>% as.data.frame()
t_in$Area <- as.numeric(t_in$Area) # cast to numeric

# Normalize
t <- t_in %>% dplyr::group_by(Sample.Name) %>% dplyr::summarise(Area = Area / Area[Component.Name == "CAP_IS_4"]) %>% as.data.frame()
t$Component.Name <- t_in$Component.Name
t$Group <- case_when(as.numeric(t$Sample.Name) <= 18 ~ "KO",
                     as.numeric(t$Sample.Name) <= 36 ~ "OE",
                     TRUE ~ "Control")

# Convert into AUC/mg stool
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Stool_PK/StoolWeights.csv")
w <- read.csv(fn) %>% as.data.frame()
w$Sample.Name = as.character(w$Sample) # recast
t <- left_join(t, w, by = "Sample.Name")
t$AreaPerMG = t$Area/t$DryStoolWeightMG

#  plots
t_5FU <- t %>% filter(Component.Name == "5FU") %>% filter(Group != "Control")
t_5FU$`5FU` <- t_5FU$AreaPerMG
t_5FU$MGperG <- predict(model_5FU, newdata = t_5FU)
t_5FU$MGperG[t_5FU$MGperG < 0] <- 0
t_5FU$TimepointHrs <- factor(t_5FU$TimepointHrs)
p <- ggplot(t_5FU, aes(x = Group, y = MGperG)) + 
  geom_point(aes(color = Group)) + 
  geom_boxplot(alpha = 0.5, aes(fill = Group), outlier.size = -1) + 
  theme_pubr() + 
  xlab("") + 
  ylab("5-FU (mg/g stool)") + 
  facet_wrap(~TimepointHrs) + 
  stat_compare_means(method = "wilcox.test", aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE,
                     method.args = list(alternative = "less"), label.y = 0.048) +
  ylim(0, 0.05) + 
  scale_color_manual(values = c("KO" = "orange", "OE" = "blue")) + 
  scale_fill_manual(values = c("KO" = "orange", "OE" = "blue")) + 
  theme(legend.position = "none") 
fn <- paste0(figdir, "Plot5FU.pdf")
ggsave(filename = fn, plot = p, width = 3.5, height = 3)


t_CAP <- t %>% filter(Component.Name == "CAP") %>% filter(Group != "Control")
t_CAP$CAP <- t_CAP$AreaPerMG
t_CAP$MGperG <- predict(model_CAP, newdata = t_CAP)
t_CAP$MGperG[t_CAP$MGperG < 0] <- 0
t_CAP$TimepointHrs <- factor(t_CAP$TimepointHrs)
p <- ggplot(t_CAP, aes(x = Group, y = MGperG)) + 
  geom_point(aes(color = Group)) + 
  geom_boxplot(alpha = 0.5, aes(fill = Group), outlier.size = -1) + 
  theme_pubr() + 
  scale_color_manual(values = c("KO" = "orange", "OE" = "blue")) + 
  scale_fill_manual(values = c("KO" = "orange", "OE" = "blue")) + 
  xlab("") + 
  ylab("CAP (mg/g stool)") + 
  facet_wrap(~TimepointHrs) + 
  stat_compare_means(method = "wilcox.test", aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE,
                     label.y = 12.6) +
  ylim(1.5, 13) + 
  theme(legend.position = "none") 
fn <- paste0(figdir, "PlotCAP.pdf")
ggsave(filename = fn, plot = p, width = 3.5, height = 3)

# Better plots!
res.aov2 <- aov(MGperG ~ Group*TimepointHrs, data = t_CAP)
p_int <- summary(res.aov2)[[1]][["Pr(>F)"]][[3]] # Interaction p value
p_value <- summary(res.aov2)[[1]][["Pr(>F)"]][[1]] # Group (i.e. colonization) p value

p1 <- ggplot(t_CAP, aes(x = TimepointHrs, y = MGperG, group = Group, color = Group, fill = Group)) + 
  stat_summary(fun="mean", geom="line") + 
  stat_summary(fun="mean", geom="point") + 
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.2) + 
  xlab("Time post-gavage (hrs)") + 
  ylab("CAP (mg/g stool)") + 
  geom_jitter(width = 0.1, alpha = 0.35) + 
  theme_pubr() + 
  scale_color_manual(values = c("KO" = "orange", "OE" = "blue"),
                     labels = c(expression(Delta*italic("preTA")), expression(italic("preTA")^"++"))) + 
  scale_fill_manual(values = c("KO" = "orange", "OE" = "blue"),
                    labels = c(expression(Delta*italic("preTA")), expression(italic("preTA")^"++"))) + 
  annotate("text", x=2, y=14, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) +
  theme(legend.title = element_blank()) + 
  theme(legend.position = "right") 

# 5-FU
res.aov2 <- aov(MGperG ~ Group*TimepointHrs, data = t_5FU)
p_int <- summary(res.aov2)[[1]][["Pr(>F)"]][[3]] # Interaction p value
p_value <- summary(res.aov2)[[1]][["Pr(>F)"]][[1]] # Group (i.e. colonization) p value

p2 <- ggplot(t_5FU, aes(x = TimepointHrs, y = MGperG, group = Group, color = Group, fill = Group)) + 
  stat_summary(fun="mean", geom="line") + 
  stat_summary(fun="mean", geom="point") + 
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.2) + 
  xlab("Time post-gavage (hrs)") + 
  ylab("5-FU (mg/g stool)") + 
  geom_jitter(width = 0.1, alpha = 0.35) + 
  theme_pubr() + 
  scale_color_manual(values = c("KO" = "orange", "OE" = "blue"),
                     labels = c(expression(Delta*italic("preTA")), expression(italic("preTA")^"++"))) + 
  scale_fill_manual(values = c("KO" = "orange", "OE" = "blue"),
                    labels = c(expression(Delta*italic("preTA")), expression(italic("preTA")^"++"))) + 
  annotate("text", x=2, y=0.05, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) +
  theme(legend.title = element_blank()) + 
  theme(legend.position = "right") 

# Arrange, common legend, save together
p <- ggarrange(p1, p2, common.legend = TRUE)
fn <- paste0(figdir, "Linegraph_both.pdf")
ggsave(filename = fn, plot = p, width = 6, height = 3)

## Correlate 5-FU levels with colonization
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/EColiCFU.csv")
col <- read.csv(fn) %>% as.data.frame() %>% filter(CFUperG > 10) %>% filter(Experiment == "PK") %>% select(Sample, CFUperG)
t_join <- t_5FU; t_join$Sample <- paste0("PK", t_join$Mouse)
t_merge <- inner_join(t_join, col, by = "Sample")

# Plot - the 3 hr 5-FU levels are correlated with 5-FU in the OE group!!
dat_5fu_3hr <- t_merge %>% filter(TimepointHrs == "3") %>% filter(Group == "OE")
corr <- cor.test(x = dat_5fu_3hr$MGperG, y = dat_5fu_3hr$CFUperG, data = dat_5fu_3hr, method = "spearman")
p_value <- corr$p.value
p <- ggplot(dat_5fu_3hr, aes(x = log10(CFUperG), y = MGperG)) + 
  geom_point(color = "blue", alpha = 0.5) + 
  stat_smooth(method = "glm", method.args = list(family = "binomial"), se = FALSE, color = "blue", linewidth = 0.8) +  # Logistic curve
  theme_pubr() + 
  annotate("text", x=9.5, y=0.015, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) +
  xlab(expression(italic("preTA")^"++"~"colonization (log"[10]~"CFU/g)")) + 
  ylab("5-FU at 3 hr (mg/g stool)") 

fn <- paste0(figdir, "Corr.pdf")
ggsave(filename = fn, plot = p, width = 3.25, height = 3)
```

## PreTA Hi/Lo Weights
```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/PreTA_HiLo.xlsx")
w <- read_excel(fn, sheet = "WeightsNorm") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w$Group <- case_when(as.numeric(w$MouseID) <= 8 ~ "Low",
                     TRUE ~ "High") # Donor grouping
w$Group <- case_when(as.numeric(w$MouseID) %in% c(1, 2, 3, 4, 9, 10, 16) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!

# add data from repeat experiment
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/PreTA_HiLo_v2.xlsx")
w2 <- read_excel(fn, sheet = "WeightsNorm") %>% as.data.frame()
w2 <- w2 %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w2$Group <- case_when(as.numeric(w2$MouseID) %in% c(1, 2, 3, 4, 5, 7) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!
w2$MouseID <- as.factor(as.numeric(w2$MouseID) + 100) # make distinct ID for plotting
w <- rbind(w, w2) # merge

# Mixed-effects model p value - group*time
w_model <- w %>% filter(!is.na(Weight))
m1 <- nlme::lme(Weight ~ Group*Day, random = ~ 1 | MouseID, data=w_model)
summary(m1)
p_value <- summary(m1)$tTable["GroupLow:Day", 5]

# Two-way anova testing
res.aov2 <- aov(Weight ~ Group*Day, data = w_model)
p_int <- summary(res.aov2)[[1]][["Pr(>F)"]][[3]] # Interaction p value
p_value <- summary(res.aov2)[[1]][["Pr(>F)"]][[1]] # Group (i.e. colonization) p value

# Plot
w_plot <- w 
w_plot$Day <- w_plot$Day
p <- ggplot(w_plot, aes(x = Day, y = Weight, group = Group, color = Group, fill = Group)) + 
  stat_summary(fun="mean", geom="line") + 
  stat_summary(fun="mean", geom="point") + 
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.2) + 
  xlab("CAP duration (days)") + 
  ylab("Weight vs baseline (%)") + 
  geom_jitter(width = 0.1, alpha = 0.2) + 
  theme_pubr() + 
  scale_color_manual(values = hilo_palette) + 
  scale_fill_manual(values = hilo_palette) + 
  annotate("text", x=3, y=91, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) +
  theme(legend.title = element_blank()) + 
  xlim(0, 14)

# Save
fn <- paste0(figdir, "PreTAHiLo_Weights.pdf")
ggsave(filename = fn, plot = p, width = 4, height = 3)
```

## PreTA hi/lo colon length
```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/PreTA_HiLo.xlsx")
col <- read_excel(fn, sheet = "Metadata") %>% as.data.frame()
col$Group <- col$PreTA; col$Group[col$Group == "Hi"] <- "High"; col$Group[col$Group == "Lo"] <- "Low"
col$Group <- case_when(as.numeric(col$Tube) %in% c(1, 2, 3, 4, 9, 10, 16) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!
col$Group <- factor(col$Group, levels = c("Low", "High")) 

# add data from repeat experiment
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/PreTA_HiLo_v2.xlsx")
col2 <- read_excel(fn, sheet = "Metadata") %>% as.data.frame()
col2$Group <- case_when(as.numeric(col2$Tube) %in% c(1, 2, 3, 4, 5, 7) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!
col2$Group <- factor(col2$Group, levels = c("Low", "High")) 
col <- rbind(col, col2) # merge

# Plot
p <- ggplot(col, aes(x = Group, y = ColonLengthCM)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  #xlab("Baseline preTA") + 
  xlab("preTA level") + 
  ylab("Colon length (cm)") + 
  scale_color_manual(values = hilo_palette) + 
  scale_fill_manual(values = hilo_palette) + 
  theme(legend.position = "none") +
  stat_compare_means(method = "wilcox", label.x = 1.1, label.y = 8.9, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) +
  ylim(c(5.8, 9.1))
  
# Save
fn <- paste0(figdir, "PreTAHiLo_ColonLength.pdf")
ggsave(filename = fn, plot = p, width = 1.5, height = 3)
```

## preTA hi/lo endpoint weight
```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/PreTA_HiLo.xlsx")
w <- read_excel(fn, sheet = "WeightsNorm") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w$Group <- case_when(as.numeric(w$MouseID) <= 8 ~ "Low",
                     TRUE ~ "High") ## Grouping by donor
w$Group <- case_when(as.numeric(w$MouseID) %in% c(1, 2, 3, 4, 9, 10, 16) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!
col <- w %>% filter(Day == 14)
col$Group <- factor(col$Group, levels = c("Low","High"))

# add data from repeat experiment
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/PreTA_HiLo_v2.xlsx")
w <- read_excel(fn, sheet = "WeightsNorm") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w$Group <- case_when(as.numeric(w$MouseID) %in% c(1, 2, 3, 4, 5, 7) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!
col2 <- w %>% filter(Day == 14)
col <- rbind(col, col2) # merge

# Plot
col$Day <- "Day 14"
p <- ggplot(col, aes(x = Group, y = Weight)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  #xlab("Baseline preTA") + 
  xlab("preTA level") + 
  facet_wrap(~Day) + 
  ylab("Weight vs baseline (%)") + 
  scale_color_manual(values = hilo_palette) + 
  scale_fill_manual(values = hilo_palette) + 
  theme(legend.position = "none") +
  stat_compare_means(method = "wilcox", label.x = 1.0, label.y = 106, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) +
  ylim(c(88, 107)) 
  
# Save
fn <- paste0(figdir, "PreTAHiLo_EndpointWeight.pdf")
ggsave(filename = fn, plot = p, width = 1.6, height = 3)
```

## PreTA hi/lo spleen weights 
```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/PreTA_HiLo.xlsx")
col <- read_excel(fn, sheet = "Metadata") %>% as.data.frame()
col$Group <- case_when(as.numeric(col$Tube) %in% c(1, 2, 3, 4, 9, 10, 16) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!
col$Group <- factor(col$Group, levels = c("Low", "High")) 

# Add repeat experiment data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/PreTA_HiLo.xlsx")
col2 <- read_excel(fn, sheet = "Metadata") %>% as.data.frame()
col2$Group <- case_when(as.numeric(col2$Tube) %in% c(1, 2, 3, 4, 5, 7) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!
col2$Group <- factor(col2$Group, levels = c("Low", "High")) 
col <- rbind(col, col2)

# Plot
p <- ggplot(col, aes(x = Group, y = SpleenWeightMG, group = Group, color = Group)) + 
  geom_boxplot(outlier.size = -1) + 
  geom_jitter(width = 0.1) +
  theme_pubr() + 
  xlab("") + 
  ylab("Spleen weight (mg)") + 
  theme(legend.position = "none") +
  stat_compare_means(method = "wilcox", aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  scale_color_manual(values = hilo_palette) 
  
# Save
fn <- paste0(figdir, "PreTAHiLo_SpleenWeight.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)

# Plot
p <- ggplot(col, aes(x = Group, y = SpleenWeightMG*100/(EndingWeightG*1000), group = Group, color = Group)) + 
  geom_boxplot(outlier.size = -1) + 
  geom_jitter(width = 0.1) +
  theme_pubr() + 
  xlab("") + 
  ylab("Spleen weight/body weight (%)") + 
  theme(legend.position = "none") +
  stat_compare_means(method = "wilcox", aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  scale_color_manual(values = hilo_palette) 
  
# Save
fn <- paste0(figdir, "PreTAHiLo_SpleenWeightPercent.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## PreTAHi/Lo fat pad weight
```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/PreTA_HiLo.xlsx")
col <- read_excel(fn, sheet = "Metadata") %>% as.data.frame()
col$Group <- case_when(as.numeric(col$Tube) %in% c(1, 2, 3, 4, 9, 10, 16) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!

# Add repeat experiment data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/PreTA_HiLo_v2.xlsx")
col2 <- read_excel(fn, sheet = "Metadata") %>% as.data.frame()
col2$Group <- case_when(as.numeric(col2$Tube) %in% c(1, 2, 3, 4, 5, 7) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!

# merge
col <- rbind(col, col2)
col$Group <- factor(col$Group, levels = c("Low", "High")) 

# Plot
p <- ggplot(col, aes(x = Group, y = GonadFatMG, group = Group, color = Group)) + 
  geom_boxplot(outlier.size = -1) + 
  geom_jitter(width = 0.1) +
  theme_pubr() + 
  xlab("") + 
  ylab("Gonadal fat pad weight (mg)") + 
  theme(legend.position = "none") +
  stat_compare_means(method = "wilcox", aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  scale_color_manual(values = hilo_palette) 
  
# Save
fn <- paste0(figdir, "Sup_PreTAHiLo_FatWeight.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)

# Plot
p <- ggplot(col, aes(x = Group, y = GonadFatMG/(EndingWeightG*1000)*100, group = Group, color = Group)) + 
  geom_boxplot(outlier.size = -1) + 
  geom_jitter(width = 0.1) +
  theme_pubr() + 
  xlab("") + 
  ylab("gWAT weight/body weight (%)") + 
  theme(legend.position = "none") +
  stat_compare_means(method = "wilcox", aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  scale_color_manual(values = hilo_palette) 
  
# Save
fn <- paste0(figdir, "PreTAHiLo_GonadalFat.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## preTA vs weight loss 
```{r}
# Read in preTA data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/KO1.csv")
preta <- fread(fn) %>% as.data.frame() %>%
  filter(KO %in% c("K17723", "K17722")) %>%
  select(-V1) %>% select(-KO) %>%
  melt()
colnames(preta) <- c("Sample", "RPKG")
preta <- preta %>% group_by(Sample) %>% 
  dplyr::summarise(RPKG = sum(RPKG)) %>%
  as.data.frame()
preta$Time <- case_when(grepl("_Post", preta$Sample) ~ "Day 15",
                        TRUE ~ "Day 0")
preta$MouseID <- gsub(".*_", "", preta$Sample) %>% as.numeric()
preta$Group <- case_when(preta$MouseID <= 8 ~ "Low",
                         TRUE ~ "High")
preta$Group <- case_when(as.numeric(preta$MouseID) %in% c(1, 2, 3, 4, 9, 10, 16) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!

# Filter
preta <- preta %>% filter(!(MouseID %in% mice_remove))
preta$Experiment = 1
preta1 <- preta

# Read in endpoint weight
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/PreTA_HiLo.xlsx")
w <- read_excel(fn, sheet = "WeightsNorm") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w$MouseID <- as.numeric(w$MouseID)
w$Group <- case_when(w$MouseID <= 8 ~ "Low",
                     TRUE ~ "High")
w$Group <- case_when(as.numeric(w$MouseID) %in% c(1, 2, 3, 4, 9, 10, 16) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!
col <- w %>% filter(Day == 14)
col$Experiment <- 1
col1 <- col

## EXPERIMENT 2
# Read in preTA data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/KO2.csv")
preta <- fread(fn) %>% as.data.frame() %>%
  filter(KO %in% c("K17723", "K17722")) %>%
  select(-V1) %>% select(-KO) %>%
  melt()
colnames(preta) <- c("Sample", "RPKG")
preta <- preta %>% group_by(Sample) %>% 
  dplyr::summarise(RPKG = sum(RPKG)) %>%
  as.data.frame()
preta$Time <- case_when(grepl("_Post", preta$Sample) ~ "Day 15",
                        TRUE ~ "Day 0")
preta$MouseID <- gsub(".*_", "", preta$Sample) %>% as.numeric()
preta$Group <- case_when(as.numeric(preta$MouseID) %in% c(1, 2, 3, 4, 5, 7) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!
preta$Experiment = 2
preta2 <- preta

# Read in endpoint weight
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/PreTA_HiLo_v2.xlsx")
w <- read_excel(fn, sheet = "WeightsNorm") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w$MouseID <- as.numeric(w$MouseID)
w$Group <- case_when(as.numeric(w$MouseID) %in% c(1, 2, 3, 4, 5, 7) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!
col <- w %>% filter(Day == 14)
col$Experiment = 2
col2 <- col

preta <- rbind(preta1, preta2)
col <- rbind(col1, col2)

# Merge
merged <- left_join(preta, col, by = c("MouseID", "Group", "Experiment"))
bl <- merged %>% filter(Time == "Day 0")

# Stats
spearman_test <- cor.test(bl$RPKG, bl$Weight, method = "spearman")
correlation <- spearman_test$estimate
p_value <- spearman_test$p.value

# Plot
p <- ggplot(bl, aes(x = RPKG, y = Weight)) + 
  geom_smooth(method = "lm", color = "black")  + 
  geom_point(aes(color = Group)) + 
  xlab("Baseline preTA level (RPKG)") + 
  ylab("D14 weight vs baseline (%)") + 
  scale_color_manual(values = hilo_palette) + 
  theme_pubr() + 
  theme(legend.position = "none") + 
  annotate("text", x=0.05, y=108, label = sprintf("Spearman's rho = %.2f", correlation)) +
annotate("text", x=0.05, y=106, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) +
  ylim(88, 108) 

# Save
fn <- paste0(figdir, "preTAvsWeight.pdf")
ggsave(plot = p, filename = fn, height = 3, width = 3)
```

## preTA vs colon length 
```{r}
# Read in preTA data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/KO1.csv")
preta <- fread(fn) %>% as.data.frame() %>%
  filter(KO %in% c("K17723", "K17722")) %>%
  select(-V1) %>% select(-KO) %>%
  melt()
colnames(preta) <- c("Sample", "RPKG")
preta <- preta %>% group_by(Sample) %>% 
  dplyr::summarise(RPKG = sum(RPKG)) %>%
  as.data.frame()
preta$Time <- case_when(grepl("_Post", preta$Sample) ~ "Day 15",
                        TRUE ~ "Day 0")
preta$MouseID <- gsub(".*_", "", preta$Sample) %>% as.numeric()
preta$Group <- case_when(preta$MouseID <= 8 ~ "Low",
                         TRUE ~ "High")
preta$Group <- case_when(as.numeric(preta$MouseID) %in% c(1, 2, 3, 4, 9, 10, 16) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!

# Filter
preta <- preta %>% filter(!(MouseID %in% mice_remove))
preta1 <- preta
preta1$Experiment = 1

# Read in endpoint colon length
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/PreTA_HiLo.xlsx")
col <- read_excel(fn, sheet = "Metadata") %>% as.data.frame()
col$Group <- col$PreTA; col$Group[col$Group == "Hi"] <- "High"; col$Group[col$Group == "Lo"] <- "Low"
col$Group <- factor(col$Group, levels = c("Low", "High"))
col$Group <- case_when(as.numeric(col$Tube) %in% c(1, 2, 3, 4, 9, 10, 16) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!
col$MouseID <- as.numeric(col$Tube)
col1 <- col
col1$Experiment = 1

## Experiment 2
# Read in preTA data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/KO2.csv")
preta <- fread(fn) %>% as.data.frame() %>%
  filter(KO %in% c("K17723", "K17722")) %>%
  select(-V1) %>% select(-KO) %>%
  melt()
colnames(preta) <- c("Sample", "RPKG")
preta <- preta %>% group_by(Sample) %>% 
  dplyr::summarise(RPKG = sum(RPKG)) %>%
  as.data.frame()
preta$Time <- case_when(grepl("_Post", preta$Sample) ~ "Day 15",
                        TRUE ~ "Day 0")
preta$MouseID <- gsub(".*_", "", preta$Sample) %>% as.numeric()
preta$Group <- case_when(as.numeric(preta$MouseID) %in% c(1, 2, 3, 4, 5, 7) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!

# Filter
preta2 <- preta
preta2$Experiment <- 2

# Read in endpoint colon length
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/PreTA_HiLo.xlsx")
col <- read_excel(fn, sheet = "Metadata") %>% as.data.frame()
col$Group <- case_when(as.numeric(col$Tube) %in% c(1, 2, 3, 4, 5, 7) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!
col$MouseID <- as.numeric(col$Tube)
col2 <- col
col2$Experiment <- 2

# Merge
col <- rbind(col1, col2)
preta <- rbind(preta1, preta2)
merged <- left_join(preta, col, by = c("MouseID", "Group", "Experiment"))
bl <- merged %>% filter(Time == "Day 0")

# Stats
spearman_test <- cor.test(bl$RPKG, bl$ColonLengthCM, method = "spearman")
correlation <- spearman_test$estimate
p_value <- spearman_test$p.value

# Plot
p <- ggplot(bl, aes(x = RPKG, y = ColonLengthCM)) + 
  geom_smooth(method = "lm", color = "black")  + 
  geom_jitter(width = 0.001, height = 0.05, aes(color = Group)) + 
  xlab("Baseline preTA level (RPKG)") + 
  ylab("Colon length (cm)") + 
  scale_color_manual(values = hilo_palette) + 
  theme_pubr() + 
  theme(legend.position = "none") + 
  annotate("text", x=0.05, y=9.5, label = sprintf("Spearman's rho = %.2f", correlation)) +
annotate("text", x=0.05, y=9.3, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) +
  ylim(6, 9.5) 

# Save
fn <- paste0(figdir, "preTAvsColonLength.pdf")
ggsave(plot = p, filename = fn, height = 3, width = 3)
```

## Gnoto preTA Baseline
```{r}
# Read in preTA data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/KO1.csv")
preta <- fread(fn) %>% as.data.frame() %>%
  filter(KO %in% c("K17723", "K17722")) %>%
  select(-V1) %>% select(-KO) %>%
  melt()
colnames(preta) <- c("Sample", "RPKG")
preta <- preta %>% group_by(Sample) %>% 
  dplyr::summarise(RPKG = sum(RPKG)) %>%
  as.data.frame()
preta$Time <- case_when(grepl("_Post", preta$Sample) ~ "Day 15",
                        TRUE ~ "Day 0")
preta$MouseID <- gsub(".*_", "", preta$Sample) %>% as.numeric()
preta$Donor <- case_when(preta$MouseID <= 8 ~ "Patient 35",
                         TRUE ~ "Patient 32")
preta$Group <- case_when(as.numeric(preta$MouseID) %in% c(1, 2, 3, 4, 9, 10, 16) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!

# Filter
mice_remove = "8"
preta <- preta %>% filter(!(MouseID %in% mice_remove))
preta <- preta %>% filter(Time == "Day 0")
preta1 <- preta

# Read in data from Experiment 2
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/KO2.csv")
preta <- fread(fn) %>% as.data.frame() %>%
  filter(KO %in% c("K17723", "K17722")) %>%
  select(-V1) %>% select(-KO) %>%
  melt()
colnames(preta) <- c("Sample", "RPKG")
preta <- preta %>% group_by(Sample) %>% 
  dplyr::summarise(RPKG = sum(RPKG)) %>%
  as.data.frame()
preta$Time <- "Day 0"
preta$MouseID <- gsub(".*_", "", preta$Sample) %>% as.numeric()
preta$Donor <- case_when(preta$MouseID <= 8 ~ "Patient 7",
                         TRUE ~ "Patient 39")
preta$Group <- case_when(as.numeric(preta$MouseID) %in% c(1, 2, 3, 4, 5, 7) ~ "Low",
                     TRUE ~ "High") ## Grouping by measured preTA levels!!

# Filter
mice_remove = ""
preta <- preta %>% filter(!(MouseID %in% mice_remove))
preta <- preta %>% filter(Time == "Day 0")
preta2 <- preta

# Merge
preta <- rbind(preta1, preta2)
preta$Donor <- gsub("Patient ", "", preta$Donor)
preta$Group <- factor(preta$Group, levels = c("Low", "High"))

# Plot
set.seed(6)
p <- ggplot(preta, aes(x = Group, y = RPKG)) + 
  geom_boxplot(aes(fill = Group), alpha = 0.5, outlier.size = -1) + 
  ylab("Baseline preTA (RPKG)") + 
  scale_fill_manual(values = hilo_palette) + 
  xlab("preTA level") + 
  geom_jitter(height = 0.001, width = 0.05, aes(shape = Donor), size = 1.5) +
  theme_pubr() + 
  stat_compare_means(method = "wilcox", label.x = 1, label.y = 0.51, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "right") + 
  ylim(-0.005, 0.52) +
  guides(fill = "none")

# Save
fn <- paste0(figdir, "preTABaseline.pdf")
ggsave(plot = p, filename = fn, height = 3, width = 2.5)
```

## Gnoto preTA vs time - Exp 1
```{r}
fn <- "/mnt/tank/labmainshare/qb3share/ktrepka/GOStudy/GnotoPreTAHiLo/Tables/GeneTables/KO.csv"
preta <- fread(fn) %>% as.data.frame() %>%
  filter(KO %in% c("K17723", "K17722")) %>% # Note - K17722 only is NOT DETECTABLE pre-treatment, but becomes detectable post-treatment in many mice
  select(-V1) %>% select(-KO) %>%
  melt()
colnames(preta) <- c("Sample", "RPKG")
preta <- preta %>% group_by(Sample) %>% 
  dplyr::summarise(RPKG = sum(RPKG)) %>%
  as.data.frame()
preta$Time <- case_when(grepl("_Post", preta$Sample) ~ "Day 15",
                        TRUE ~ "Day 0")
preta$MouseID <- gsub(".*_", "", preta$Sample) %>% as.numeric()

# Filter
mice_remove = "8"
preta <- preta %>% filter(!(MouseID %in% mice_remove))


# Plot
time_palette = c("Day 0" = "grey", "Day 15" = "red")
mice_test <- preta %>% group_by(MouseID) %>% 
  dplyr::summarise(n = n()) %>% 
  filter(n == 2) %>% 
  select(MouseID) %>%
  pull()
p <- ggplot(preta, aes(x = Time, y = RPKG)) + 
  geom_boxplot(aes(fill = Time), alpha = 0.5) +
  xlab("Days of CAP") +
  ylab("preTA (RPKG)") + 
  geom_jitter(aes(color = Time), width = 0.05, height = 0.001) + 
  stat_compare_means(label.x = 1.2, label.y = 0.12, method = "wilcox.test", method.args = list(alternative = "less"), paired = TRUE, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE, data = preta %>% filter(MouseID %in% mice_test)) +
  scale_color_manual(values = time_palette) + 
  scale_fill_manual(values = time_palette) + 
  theme_pubr() + 
  theme(legend.position = "none") 

# Save
fn <- paste0(figdir, "preTAvsTime.pdf")
ggsave(plot = p, filename = fn, height = 3, width = 2.5)
```

## Gnoto preTA+ species vs time Exp1
```{r}
# Read in preTA data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Humanization/TaxaSpecies1.csv")
tax <- fread(fn)
tax[,2:17] <- 100*tax[,2:17]/colSums(tax[,2:17]) # make sure columns sum to 100% relative abundance
tax <- tax %>% as.data.frame() %>%
  filter(grepl("Anaerostipes|Escherichia|Eubacterium|Citrobacter", Taxa)) %>% 
  select(-Taxa) %>%
  melt()
colnames(tax) <- c("Sample", "RelativeAbundance")
tax <- tax %>% group_by(Sample) %>% 
  dplyr::summarise(RelativeAbundance = sum(RelativeAbundance)) %>%
  as.data.frame()
tax$Time <- case_when(grepl("_Post", tax$Sample) ~ "Day 15",
                        TRUE ~ "Day 0")
tax$MouseID <- gsub(".*_", "", tax$Sample) %>% as.numeric()
tax$Group <- case_when(tax$MouseID <= 8 ~ "Low",
                         TRUE ~ "High")

# Filter
tax <- tax %>% filter(!(MouseID %in% mice_remove))

# Plot
time_palette = c("Day 0" = "grey", "Day 15" = "red")
mice_test <- tax %>% group_by(MouseID) %>% 
  dplyr::summarise(n = n()) %>% 
  filter(n == 2) %>% 
  select(MouseID) %>%
  pull()
p <- ggplot(tax, aes(x = Time, y = RelativeAbundance)) + 
  geom_boxplot(aes(fill = Time), alpha = 0.5, outlier.size = -1) +
  xlab("Days of CAP") +
  ylab("preTA+ species (%)") + 
  geom_jitter(aes(color = Time), width = 0.05, height = 0.001) + 
  stat_compare_means(label.x = 1.2, label.y = 0.12, method = "wilcox.test", method.args = list(alternative = "less"), paired = TRUE, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE, data = tax %>% filter(MouseID %in% mice_test)) +
  scale_color_manual(values = time_palette) + 
  scale_fill_manual(values = time_palette) + 
  theme_pubr() + 
  theme(legend.position = "none") 
```

## SPF Rescue Weight
```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/SPFCAPRescueExperiment2.xlsx")
w <- read_excel(fn, sheet = "Weights") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w$Group <- case_when(as.numeric(w$MouseID) <= 12 ~ "ΔpreTA",
                     as.numeric(w$MouseID) <= 24 ~ "preTA-wt",
                     TRUE  ~ "preTA++")
w$Group <- factor(w$Group, levels = c("ΔpreTA", "preTA-wt", "preTA++"))

# Two-way anova testing
w_model <- w %>% filter(!is.na(Weight))
res.aov2 <- aov(Weight ~ Group*Day, data = w_model)
p_int <- summary(res.aov2)[[1]][["Pr(>F)"]][[3]] # Interaction p value
p_value <- summary(res.aov2)[[1]][["Pr(>F)"]][[1]] # Group (i.e. colonization) p value

# Each group individually
w_model <- w %>% filter(!is.na(Weight)) %>% filter(Group != "preTA-wt") %>% filter(Day <= 8)
res.aov2 <- aov(Weight ~ Group*Day, data = w_model)

w_model <- w %>% filter(!is.na(Weight)) %>% filter(Group != "preTA++") %>% filter(Day <= 8)
res.aov2 <- aov(Weight ~ Group*Day, data = w_model)

# Plot
w_plot <- w %>% filter(Day <= 8)
w_plot$Day <- w_plot$Day # Baseline is already 0
p <- ggplot(w_plot, aes(x = Day, y = 100*Weight, group = Group, color = Group, fill = Group)) + 
  stat_summary(fun="mean", geom="line") + 
  stat_summary(fun="mean", geom="point") + 
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.2) + 
  xlab("CAP duration (days)") + 
  ylab("Weight vs baseline (%)") + 
  geom_jitter(width = 0.1, alpha = 0.2) + 
  theme_pubr() + 
  scale_color_manual(values = preta_palette, labels = c("ΔpreTA" = expression(Delta*italic(preTA)), "preTA-wt" = expression(italic(wt)), "preTA++" = expression(italic(preTA)^"++"))) + 
  scale_fill_manual(values = preta_palette, labels = c("ΔpreTA" = expression(Delta*italic(preTA)), "preTA-wt" = expression(italic(wt)), "preTA++" = expression(italic(preTA)^"++"))) + 
  theme(legend.title = element_blank()) 

# Save
fn <- paste0(figdir, "Weights.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)

fn <- paste0(figdir, "WeightsNoLegend.pdf")
ggsave(filename = fn, plot = p + theme(legend.position = "none"), width = 3, height = 3)
```

## SPF Rescue Endpoint weight
```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/SPFCAPRescueExperiment2.xlsx")
w <- read_excel(fn, sheet = "Weights") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w$Group <- case_when(as.numeric(w$MouseID) <= 12 ~ "ΔpreTA",
                     as.numeric(w$MouseID) <= 24 ~ "preTA-wt",
                     TRUE  ~ "preTA++")
w$Group <- factor(w$Group, levels = c("ΔpreTA", "preTA-wt", "preTA++"))

# Plot
w <- w %>% filter(Day == 8)
w$Day <- "Day 8"
p <- ggplot(w, aes(x = Group, y = Weight*100)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  ylab("Weight vs baseline (%)") + 
  scale_color_manual(values = preta_palette) + 
  scale_fill_manual(values = preta_palette) + 
  facet_wrap(~Day) + 
  stat_compare_means(method = "wilcox", label.x = 1.2, label.y = 106, aes(label = paste0("italic(p) ==", after_stat(p.format))), comparisons = list(c("ΔpreTA", "preTA-wt")), parse = TRUE) + 
  stat_compare_means(method = "wilcox", label.x = 2, label.y = 110, aes(label = paste0("italic(p) ==", after_stat(p.format))), comparisons = list(c("ΔpreTA", "preTA++")), parse = TRUE) + 
  theme(legend.position = "none") + 
  ylim(c(83, 113)) + 
  xlab("") + 
  scale_x_discrete(labels = c("ΔpreTA" = expression(Delta*italic(preTA)), "preTA-wt" = expression(italic(wt)), "preTA++" = expression(italic(preTA)^"++"))) +
  theme(axis.text.x = element_text(size = 11)) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) 

# Save
fn <- paste0(figdir, "EndpointWeight.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)
```

## SPF Rescue Survival

```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/SPFCAPRescueExperiment2.xlsx")
w <- read_excel(fn, sheet = "Survival") %>% as.data.frame() %>% dplyr::rename(Day = Time, Event = Status)
w$Group <- factor(w$Strain, levels = c("KO", "WT", "OE"))

# object generation
surv_obj <- with(w, Surv(Day, Event))
fit <- survfit(Surv(Day, Event) ~ Group, data = w)
surv_df <- fortify(fit)

# Analysis
log_rank_test <- survdiff(Surv(Day, Event) ~ Group, data = w)
p_value <- 1 - pchisq(log_rank_test$chisq, df = 2) # Adjust the degrees of freedom if needed. df = number of groups - 1

# Calculate individual p values
log_rank_test <- survdiff(Surv(Day, Event) ~ Group, data = w %>% filter(Strain != "OE"))
p_value_KOWT <- 1 - pchisq(log_rank_test$chisq, df = 1) # Adjust the degrees of freedom if needed. df = number of groups - 1

log_rank_test <- survdiff(Surv(Day, Event) ~ Group, data = w %>% filter(Strain != "WT"))
p_value_KOOE <- 1 - pchisq(log_rank_test$chisq, df = 1) # Adjust the degrees of freedom if needed. df = number of groups - 1

# Plot the survival curve
ggsurv <- ggsurvplot(fit, data = w, pval = FALSE, conf.int = FALSE,
           risk.table = FALSE)
p <- ggsurv$plot
p <- p + scale_color_manual(values = preta_surv_palette) +
  xlab("CAP duration (days)") + 
  ylab("Survival probability") + 
  theme_pubr() + 
  theme(legend.position = "none") + 
  annotate("text", x = 0.1, y = .15, label = paste("Delta*italic(preTA)~v~italic(preTA)^'++'*':'~italic(p) == ", formatC(signif(p_value_KOOE, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE, hjust = 0) +
  annotate("text", x = 0.1, y = .05, label = paste("Delta*italic(preTA)~v~italic(wt)*':'~italic(p) == ", formatC(signif(p_value_KOWT, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE, hjust = 0)

# Save
fn <- paste0(figdir, "Survival.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 2.75)
```

## SPF Rescue Colon length
```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/SPFCAPRescueExperiment2.xlsx")
w <- read_excel(fn, sheet = "OrganSize") %>% as.data.frame()
w$Group <- case_when(as.numeric(w$MouseID) <= 12 ~ "ΔpreTA",
                     as.numeric(w$MouseID) <= 24 ~ "preTA-wt",
                     TRUE  ~ "preTA++")
w$Group <- factor(w$Group, levels = c("ΔpreTA", "preTA-wt", "preTA++"))

# Plot
p <- ggplot(w, aes(x = Group, y = LILengthCM)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  xlab("") + 
  ylab("Colon length (cm)") + 
  scale_color_manual(values = preta_palette) + 
  scale_fill_manual(values = preta_palette) + 
  scale_x_discrete(labels = c("ΔpreTA" = expression(Delta*italic(preTA)), "preTA-wt" = expression(italic(wt)), "preTA++" = expression(italic(preTA)^"++"))) + 
  theme(legend.position = "none") +
  stat_compare_means(method = "wilcox", label.x = 1.2, label.y = 7.7, aes(label = paste0("italic(p) ==", after_stat(p.format))), comparisons = list(c("ΔpreTA", "preTA-wt")), parse = TRUE) + 
  stat_compare_means(method = "wilcox", label.x = 2, label.y = 8.1, aes(label = paste0("italic(p) ==", after_stat(p.format))), comparisons = list(c("ΔpreTA", "preTA++")), parse = TRUE) + 
  theme(legend.position = "none") + 
  ylim(c(5.3, 8.3)) +
  theme(axis.text.x = element_text(size = 11)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) 

# Save
fn <- paste0(figdir, "ColonLength.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)
```

# Figure S11 - preTA KO/OE experiment 1
```{r}
# Make directory
figdir <- paste0(dir, "FigureS11/")
dir.create(figdir)
```

## SPF CFU D5

```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/D5Stool.csv")
df <- read.csv(fn)

# Exclude samples with no stool or dried stool
df <- df %>% filter(!(Sample.number %in% c("6", "19", "23")))
df$Group <- case_when(as.numeric(df$Sample.number) <= 16 ~ "KO",
                      TRUE ~ "OE")

# Plot
p <- ggplot(df, aes(x = Group, y = CFU.g + 10)) + 
  geom_boxplot(aes(fill = Group), alpha = 0.5, outlier.size = -1) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  xlab("") + 
  ylab("Colonization (CFU/g)") + 
  stat_compare_means(method = "wilcox", label.x = 1, label.y = 9.5, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  scale_y_log10(limits = c(10, 10000000000)) + 
  scale_color_manual(values = preta_palette) + 
  scale_fill_manual(values = preta_palette) + 
  theme(legend.position = "none")

fn <- paste0(figdir, "SPF_CFU.pdf")
ggsave(filename = fn, plot = p, width = 1.5, height = 3)

```

## SPF Colonization
Baseline colonization was measured by cage. I also have data for D5 stool and Endpoint LI.

```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/SPFWTCAPRescueExperiment1.xlsx")
col <- read_excel(fn, sheet = "Colonization") %>% as.data.frame()
c <- col %>% filter(Day == "0")
c$Group <- case_when(grepl("1|2|3|4", c$MouseID) ~ "ΔpreTA",
                     TRUE ~ "preTA++")
c$Group <- factor(c$Group, levels = c("ΔpreTA", "preTA++"))

# Graph
p <- ggplot(c, aes(x = Group, y = CFUPerG)) + 
  geom_boxplot(aes(fill = Group), alpha = 0.5, outlier.size = -1) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  xlab("") + 
  ylab("Colonization (CFU/g)") + 
  stat_compare_means(method = "wilcox", label.x = 1, label.y = 9, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  scale_y_log10(limits = c(1,5e9)) + 
  scale_color_manual(values = preta_palette) + 
  scale_fill_manual(values = preta_palette) + 
  theme(legend.position = "none") + 
  scale_x_discrete(labels = c("ΔpreTA" = expression(Delta*italic(preTA)), "preTA++" = expression(italic(preTA)^"++"))) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Save
fn <- paste0(figdir, "SPF_colonization.pdf")
ggsave(filename = fn, plot = p, width = 1.5, height = 3)
```

## SPF Colonization levels pre/post gavage

```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/SPFWTCAPRescueExperiment1.xlsx")
col <- read_excel(fn, sheet = "Colonization") %>% as.data.frame()
c <- col %>% filter(Day %in% c("0", "-1"))
c$Group <- case_when(grepl("1|2|3|4", c$MouseID) ~ "KO",
                     TRUE ~ "OE")
c$Group <- factor(c$Group)
c$Day <- case_when(c$Day == "-1" ~ "Pre",
                   c$Day == "0" ~ "Post",
                   TRUE ~ c$Day)
c$Day <- factor(c$Day, levels = c("Pre", "Post"))

# Plot
p <- ggplot(c, aes(x = Day, y = CFUPerG+1, group = Group, color = Group)) + 
  geom_bar(position="dodge", stat = "summary", fill = "white") + 
  geom_jitter(position=position_jitterdodge(0.5, seed = 5)) +
  scale_y_log10() + 
  theme_pubr() + 
  xlab("") + 
  ylab("Colonization(CFU/g)") + 
  scale_color_manual(values = preta_palette)

# Save
fn <- paste0(figdir, "SPF_colonization_prepost.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## SPF preTA weight
```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/SPFWTCAPRescueExperiment1.xlsx")
w <- read_excel(fn, sheet = "TractableWeightsNormalized") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w$Group <- case_when(grepl("1_|2_|3_|4_", w$MouseID) ~ "ΔpreTA",
                     TRUE ~ "preTA++")
w$Group <- factor(w$Group, levels = c("ΔpreTA", "preTA++"))

# Mixed-effects model p value - group*time. The interaction term is significant (p = 0.0002)
w_model <- w %>% filter(!is.na(Weight))
m1 <- lme(Weight ~ Group*Day, random = ~ 1 | MouseID, data=w_model)
summary(m1)
p_value <- summary(m1)$tTable["GrouppreTA++:Day", 5]

# Two-way anova testing
res.aov2 <- aov(Weight ~ Group*Day, data = w_model)
p_int <- summary(res.aov2)[[1]][["Pr(>F)"]][[3]] # Interaction p value
p_value <- summary(res.aov2)[[1]][["Pr(>F)"]][[1]] # Group (i.e. colonization) p value

# Plot
w_plot <- w %>% filter(Day <= 9)
w_plot$Day <- w_plot$Day - 1 # Set baseline as 0
p <- ggplot(w_plot, aes(x = Day, y = Weight, group = Group, color = Group, fill = Group)) + 
  stat_summary(fun="mean", geom="line") + 
  stat_summary(fun="mean", geom="point") + 
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.2) + 
  xlab("CAP duration (days)") + 
  ylab("Weight vs baseline (%)") + 
  geom_jitter(width = 0.1, alpha = 0.2) + 
  theme_pubr() + 
  scale_color_manual(values = preta_palette, labels = c("ΔpreTA" = expression(Delta*italic(preTA)), "preTA++" = expression(italic(preTA)^"++"))) + 
  scale_fill_manual(values = preta_palette, labels = c("ΔpreTA" = expression(Delta*italic(preTA)), "preTA++" = expression(italic(preTA)^"++"))) + 
  annotate("text", x=1.5, y=87, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) +
  theme(legend.title = element_blank())

# Save
fn <- paste0(figdir, "SPF_Weights.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## SPF preTA survival
```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/SPFWTCAPRescueExperiment1.xlsx")
w <- read_excel(fn, sheet = "Survival") %>% as.data.frame() %>% dplyr::rename(Day = DayPostCAP, Event = Status)
w$Group <- case_when(grepl("1_|2_|3_|4_", w$MouseID) ~ "KO", TRUE ~ "OE")

# object generation
surv_obj <- with(w, Surv(Day, Event))
fit <- survfit(Surv(Day, Event) ~ Group, data = w)
surv_df <- fortify(fit)

# Analysis
log_rank_test <- survdiff(Surv(Day, Event) ~ Group, data = w)
p_value <- 1 - pchisq(log_rank_test$chisq, df = 1) # Adjust the degrees of freedom if needed

# Plot the survival curve
ggsurv <- ggsurvplot(fit, data = w, pval = FALSE, conf.int = FALSE,
           risk.table = FALSE)
p <- ggsurv$plot
p <- p + scale_color_manual(values = preta_surv_palette) +
  xlab("CAP duration (days)") + 
  ylab("Survival probability") + 
  theme_pubr() + 
  theme(legend.position = "none") + 
  annotate("text", x = 3, y = .1, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) 

# Save
fn <- paste0(figdir, "SPF_Survival.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## SPF endpoint weight
```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/SPFWTCAPRescueExperiment1.xlsx")
w <- read_excel(fn, sheet = "TractableWeightsNormalized") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w$Group <- case_when(grepl("1_|2_|3_|4_", w$MouseID) ~ "ΔpreTA",
                     TRUE ~ "preTA++")
w$Group <- factor(w$Group, levels = c("ΔpreTA", "preTA++"))

# Plot
w <- w %>% filter(Day <= 9)
w$Day <- w$Day - 1 # Set baseline as 0
w <- w %>% filter(Day == 8)
w$Day <- "Day 8"
p <- ggplot(w, aes(x = Group, y = Weight)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  ylab("Weight vs baseline (%)") + 
  scale_color_manual(values = preta_palette) + 
  scale_fill_manual(values = preta_palette) + 
  facet_wrap(~Day) + 
  stat_compare_means(method = "wilcox", label.x = 1.1, label.y = 107, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "none") + 
  ylim(c(82, 109)) + 
  xlab("") + 
  scale_x_discrete(labels = c("ΔpreTA" = expression(Delta*italic(preTA)), "preTA++" = expression(italic(preTA)^"++"))) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

fn <- paste0(figdir, "SPF_EndpointWeight.pdf")
ggsave(filename = fn, plot = p, width = 1.5, height = 3)
```

# Figure S12 - preTA qPCR
```{r}
# Make directory
figdir <- paste0(dir, "FigureS12/")
dir.create(figdir)
```

## Isolate qPCR
```{r}
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/qPCR/Cq_Isolate.csv")
t <- read.csv(fn) %>% as.data.frame()

# Plot
t$Strain <- factor(t$Strain, levels = c("WT", "OE", "KO"))
t_rep = data.frame(Strain = rep(t$Strain, 3), NormalizedExpression = rep(t$NormalizedExpression, 3)) # for wilcox testing
p <- ggplot(t, aes(x = Strain, y = NormalizedExpression, fill = Strain)) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = NormalizedExpression - NormalizedSE, ymax = NormalizedExpression + NormalizedSE), width = 0.1) +
  theme_minimal() +
  labs(
    x = "",
    y = expression(italic(preTA)~"relative expression"),
  ) + 
  theme_pubr() + 
  theme(legend.position = "none") + 
  scale_x_discrete(labels = c("WT" = expression(italic(wt)), "KO" = expression(Delta*italic(preTA)), "OE" = expression(italic(preTA)^"++"))) + 
  scale_fill_manual(values = c("WT" = "grey", "KO" = "orange", "OE" = "blue")) + 
  stat_compare_means(data = t_rep, method = "wilcox.test", aes(label = p.signif), 
                     comparisons = list(c("WT", "OE"), c("KO", "WT"))) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) + 
  ylim(0, 5.5)

# Save
fn <- paste0(figdir, "IsolatepreTA.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)
```

## Mouse qPCR
```{r}
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/qPCR/Cq_Mouse.csv")
t <- read.csv(fn) %>% as.data.frame()

# Plot
t$Strain <- factor(t$Strain, levels = c("WT", "OE", "KO"))
p <- ggplot(t, aes(x = Strain, y = Normalized)) +
  geom_bar(aes(y = NormalizedExpression, fill = Strain), stat = "identity") +
  geom_errorbar(aes(y = NormalizedExpression, ymin = NormalizedExpression - NormalizedSE, ymax = NormalizedExpression + NormalizedSE), width = 0.1) +
  theme_minimal() +
  labs(
    x = "",
    y = expression(italic(preTA)~"relative expression"),
  ) + 
  theme_pubr() + 
  theme(legend.position = "none") + 
  scale_x_discrete(labels = c("WT" = expression(italic(wt)), "KO" = expression(Delta*italic(preTA)), "OE" = expression(italic(preTA)^"++"))) + 
  scale_fill_manual(values = c("WT" = "grey", "KO" = "orange", "OE" = "blue")) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) + 
  stat_compare_means(data = t, method = "wilcox.test", aes(label = p.signif), 
                     comparisons = list(c("WT", "OE"), c("KO", "WT"))) + 
  ylim(0, 5.5)

# Save
fn <- paste0(figdir, "MousepreTA.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)
```

# Figure S13 - Sex rescue
```{r}
# Make directory
figdir <- paste0(dir, "FigureS13/")
dir.create(figdir)
```

## Weight
```{r}
# Load new data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/SPFCAPRescueExperiment2.xlsx")
w <- read_excel(fn, sheet = "Weights") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w$Group <- case_when(as.numeric(w$MouseID) <= 12 ~ "ΔpreTA",
                     as.numeric(w$MouseID) <= 24 ~ "preTA-wt",
                     TRUE  ~ "preTA++")
w$Sex <- case_when(as.numeric(w$MouseID) <= 6 ~ "Male",
                     as.numeric(w$MouseID) <= 12 ~ "Female",
                     as.numeric(w$MouseID) <= 18 ~ "Male",
                     as.numeric(w$MouseID) <= 24 ~ "Female",
                     as.numeric(w$MouseID) <= 30 ~ "Male",
                     TRUE  ~ "Female")
w$Group <- factor(w$Group, levels = c("ΔpreTA", "preTA-wt", "preTA++"))
w <- w %>% filter(Group != "preTA-wt")

# Two-way anova testing
w_model <- w %>% filter(!is.na(Weight)) %>% filter(Day <= 8)
res.aov2 <- aov(Weight ~ Group*Day, data = w_model %>% filter(Sex == "Female"))
p_int <- summary(res.aov2)[[1]][["Pr(>F)"]][[3]] # Interaction p value
p_value <- summary(res.aov2)[[1]][["Pr(>F)"]][[1]] # Group (i.e. colonization) p value

# p value dataframe
res.aov2 <- aov(Weight ~ Group*Day, data = w_model %>% filter(Sex == "Male"))
p_male <- summary(res.aov2)[[1]][["Pr(>F)"]][[1]] # Group (i.e. colonization) p value
res.aov2 <- aov(Weight ~ Group*Day, data = w_model %>% filter(Sex == "Female"))
p_female <- summary(res.aov2)[[1]][["Pr(>F)"]][[1]] # Group (i.e. colonization) p value
p_value_df <- data.frame(Sex = c("Male", "Female"), p_value = c(p_male, p_female))

# Plot
w_plot <- w %>% filter(Day <= 8)
w_plot$Day <- w_plot$Day # Baseline is already 0
p <- ggplot(w_plot, aes(x = Day, y = 100*Weight, group = Group, color = Group, fill = Group)) + 
  stat_summary(fun="mean", geom="line") + 
  stat_summary(fun="mean", geom="point") + 
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.2) + 
  xlab("CAP duration (days)") + 
  ylab("Weight vs baseline (%)") + 
  facet_wrap(~Sex) + 
  geom_jitter(width = 0.1, alpha = 0.2) + 
  theme_pubr() + 
  scale_color_manual(values = preta_palette, labels = c("ΔpreTA" = expression(Delta*italic(preTA)), "preTA-wt" = expression(italic(wt)), "preTA++" = expression(italic(preTA)^"++"))) + 
  scale_fill_manual(values = preta_palette, labels = c("ΔpreTA" = expression(Delta*italic(preTA)), "preTA-wt" = expression(italic(wt)), "preTA++" = expression(italic(preTA)^"++"))) + 
  geom_text(data = p_value_df, aes(x = 1.5, y = 87, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), format="e", digits=2))), inherit.aes = FALSE, parse = TRUE) + 
  theme(legend.title = element_blank()) 

# Save
fn <- paste0(figdir, "WeightsSex.pdf")
ggsave(filename = fn, plot = p, width = 6, height = 3)

fn <- paste0(figdir, "WeightsSexNoLegend.pdf")
ggsave(filename = fn, plot = p + theme(legend.position = "none"), width = 6, height = 3)
```

## Colon length
```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/SPFCAPRescueExperiment2.xlsx")
w <- read_excel(fn, sheet = "OrganSize") %>% as.data.frame()
w$Group <- case_when(as.numeric(w$MouseID) <= 12 ~ "ΔpreTA",
                     as.numeric(w$MouseID) <= 24 ~ "preTA-wt",
                     TRUE  ~ "preTA++")
w$Sex <- case_when(as.numeric(w$MouseID) <= 6 ~ "Male",
                     as.numeric(w$MouseID) <= 12 ~ "Female",
                     as.numeric(w$MouseID) <= 18 ~ "Male",
                     as.numeric(w$MouseID) <= 24 ~ "Female",
                     as.numeric(w$MouseID) <= 30 ~ "Male",
                     TRUE  ~ "Female")
w$Group <- factor(w$Group, levels = c("ΔpreTA", "preTA-wt", "preTA++"))
w <- w %>% filter(Group != "preTA-wt")

# Plot
p <- ggplot(w, aes(x = Group, y = LILengthCM)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  xlab("") + 
  facet_wrap(~Sex) + 
  ylab("Colon length (cm)") + 
  scale_color_manual(values = preta_palette) + 
  scale_fill_manual(values = preta_palette) + 
  scale_x_discrete(labels = c("ΔpreTA" = expression(Delta*italic(preTA)), "preTA-wt" = expression(italic(wt)), "preTA++" = expression(italic(preTA)^"++"))) + 
  theme(legend.position = "none") +
  stat_compare_means(method = "wilcox", label.x = 1.2, label.y = 7.7, aes(label = paste0("italic(p) ==", after_stat(p.format))), comparisons = list(c("ΔpreTA", "preTA-wt")), parse = TRUE) + 
  stat_compare_means(method = "wilcox", label.x = 2, label.y = 7.3, aes(label = paste0("italic(p) ==", after_stat(p.format))), comparisons = list(c("ΔpreTA", "preTA++")), parse = TRUE) + 
  theme(legend.position = "none") + 
  ylim(c(5.3, 7.6)) +
  theme(axis.text.x = element_text(size = 11)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) 

# Save
fn <- paste0(figdir, "ColonLengthSex.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)
```

## Survival male

```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/SPFCAPRescueExperiment2.xlsx")
w <- read_excel(fn, sheet = "Survival") %>% as.data.frame() %>% dplyr::rename(Day = Time, Event = Status)
w$Group <- factor(w$Strain, levels = c("KO", "WT", "OE"))
w <- w %>% filter(Sex == "M")
w <- w %>% filter(Group != "WT")

# object generation
surv_obj <- with(w, Surv(Day, Event))
fit <- survfit(Surv(Day, Event) ~ Group, data = w)
surv_df <- fortify(fit)

# Analysis
log_rank_test <- survdiff(Surv(Day, Event) ~ Group, data = w)
p_value <- 1 - pchisq(log_rank_test$chisq, df = 1) # Adjust the degrees of freedom if needed. df = number of groups - 1

# Plot the survival curve - one-sided test
ggsurv <- ggsurvplot(fit, data = w, pval = FALSE, conf.int = FALSE,
           risk.table = FALSE)
p <- ggsurv$plot
p <- p + scale_color_manual(values = preta_surv_palette) +
  xlab("CAP duration (days)") + 
  ylab("Survival probability") + 
  theme_pubr() + 
  theme(legend.position = "none") + 
  annotate("text", x = 0.1, y = .05, label = paste("italic(p) == ", formatC(signif(p_value/2, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE, hjust = 0)

# Save
fn <- paste0(figdir, "SurvivalMale.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 2.5)
```

## Survival female
```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/SPFCAPRescueExperiment2.xlsx")
w <- read_excel(fn, sheet = "Survival") %>% as.data.frame() %>% dplyr::rename(Day = Time, Event = Status)
w$Group <- factor(w$Strain, levels = c("KO", "WT", "OE"))
w <- w %>% filter(Sex == "F")
w <- w %>% filter(Group != "WT")

# object generation
surv_obj <- with(w, Surv(Day, Event))
fit <- survfit(Surv(Day, Event) ~ Group, data = w)
surv_df <- fortify(fit)

# Analysis
log_rank_test <- survdiff(Surv(Day, Event) ~ Group, data = w)
p_value <- 1 - pchisq(log_rank_test$chisq, df = 1) # Adjust the degrees of freedom if needed. df = number of groups - 1

# Plot the survival curve - one-sided test
ggsurv <- ggsurvplot(fit, data = w, pval = FALSE, conf.int = FALSE,
           risk.table = FALSE)
p <- ggsurv$plot
p <- p + scale_color_manual(values = preta_surv_palette) +
  xlab("CAP duration (days)") + 
  ylab("Survival probability") + 
  theme_pubr() + 
  theme(legend.position = "none") + 
  annotate("text", x = 0.1, y = .05, label = paste("italic(p) == ", formatC(signif(p_value/2, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE, hjust = 0)

# Save
fn <- paste0(figdir, "SurvivalFemale.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 2.5)
```

## Endpoint weight
```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/SPF_Rescue/SPFCAPRescueExperiment2.xlsx")
w <- read_excel(fn, sheet = "Weights") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w$Group <- case_when(as.numeric(w$MouseID) <= 12 ~ "ΔpreTA",
                     as.numeric(w$MouseID) <= 24 ~ "preTA-wt",
                     TRUE  ~ "preTA++")
w$Group <- factor(w$Group, levels = c("ΔpreTA", "preTA-wt", "preTA++"))
w$Sex <- case_when(as.numeric(w$MouseID) <= 6 ~ "Male",
                     as.numeric(w$MouseID) <= 12 ~ "Female",
                     as.numeric(w$MouseID) <= 18 ~ "Male",
                     as.numeric(w$MouseID) <= 24 ~ "Female",
                     as.numeric(w$MouseID) <= 30 ~ "Male",
                     TRUE  ~ "Female")
w <- w %>% filter(Group != "preTA-wt")

# Plot
w <- w %>% filter(Day == 8)
w$Day <- "Day 8"
p <- ggplot(w, aes(x = Group, y = Weight*100)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  ylab("Weight vs baseline (%)") + 
  scale_color_manual(values = preta_palette) + 
  scale_fill_manual(values = preta_palette) + 
  stat_compare_means(method = "wilcox", label.x = 1.2, label.y = 106, aes(label = paste0("italic(p) ==", after_stat(p.format))), comparisons = list(c("ΔpreTA", "preTA-wt")), parse = TRUE) + 
  stat_compare_means(method = "wilcox", label.x = 2, label.y = 103, aes(label = paste0("italic(p) ==", after_stat(p.format))), comparisons = list(c("ΔpreTA", "preTA++")), parse = TRUE) + 
  theme(legend.position = "none") + 
  ylim(c(83, 105)) + 
  xlab("") + 
  facet_wrap(~Sex) + 
  scale_x_discrete(labels = c("ΔpreTA" = expression(Delta*italic(preTA)), "preTA-wt" = expression(italic(wt)), "preTA++" = expression(italic(preTA)^"++"))) +
  theme(axis.text.x = element_text(size = 11)) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) 

# Save
fn <- paste0(figdir, "EndpointWeightSex.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)
```

# Figure S14 - GF model
This data is all from Peter S toxicity experiment. 6-8 week old male C57BL/6 mice were treated with 1500 mg/kg once daily. 
```{r}
# Make directory
figdir <- paste0(dir, "FigureS14/")
dir.create(figdir)
```

## GF colonization
```{r}
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Rescue/Gnoto_CAP_preTA_PeterS_Compiled.xlsx")
col <- read_excel(fn, sheet = "Metadata") %>% as.data.frame()

# Update factor levels to include the new group names with formatting
col$Group <- case_when(col$Group == "Mock" ~ "Mock",
                       col$Group == "KO" ~ "ΔpreTA",
                       TRUE ~ "preTA++")
levels(col$Group) <- c("Mock", "ΔpreTA", "preTA++")

# Plot with updated labels
p <- ggplot(col, aes(x = Group, y = CFUPerG+1)) + 
  geom_boxplot(aes(fill = Group), alpha = 0.5, outlier.size = -1) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  scale_y_log10(limits = c(1, 1e10)) + 
  theme_pubr() + 
  xlab("") + 
  ylab("Colonization (CFU/g)") + 
  theme(legend.position = "none") +
  stat_compare_means(method = "wilcox", comparisons = list(c("ΔpreTA","preTA++")), aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  scale_color_manual(values = gf_preta_palette3) +
  annotate("text", x=1, y = 5, label = "n.d.") + 
  scale_fill_manual(values = gf_preta_palette3) + 
  theme(axis.text.x = element_text(angle = 30, hjust = 1, vjust = 1)) + 
  scale_x_discrete(labels = c("Mock" = "Mock", "ΔpreTA" = expression(Delta*italic(preTA)), "preTA++" = expression(italic(preTA)^"++")))

# Save
fn <- paste0(figdir, "GF_colonization.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3, device = cairo_pdf(family = "Arial"), dpi = 320)
```

## GF weight loss
```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Rescue/Gnoto_CAP_preTA_PeterS_Compiled.xlsx")
w <- read_excel(fn, sheet = "WeightsNorm") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w$Group <- case_when(as.numeric(w$MouseID) <= 12 ~ "Mock + ΔpreTA",
                     TRUE ~ "preTA++")

# Mixed-effects model p value - group*time. The interaction term is significant
w_model <- w %>% filter(!is.na(Weight))
m1 <- lme(Weight ~ Group*Day, random = ~ 1 | MouseID, data=w_model)
summary(m1)
p_value <- summary(m1)$tTable["GrouppreTA++:Day", 5]

# Two-way anova testing
res.aov2 <- aov(Weight ~ Group*Day, data = w_model)
p_int <- summary(res.aov2)[[1]][["Pr(>F)"]][[3]] # Interaction p value
p_value <- summary(res.aov2)[[1]][["Pr(>F)"]][[1]] # Group (i.e. colonization) p value

# Plot
p <- ggplot(w, aes(x = Day, y = Weight, group = Group, color = Group, fill = Group)) + 
  stat_summary(fun="mean", geom="line") + 
  stat_summary(fun="mean", geom="point") + 
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.2) + 
  xlab("CAP duration (days)") + 
  ylab("Weight vs baseline (%)") + 
  geom_jitter(width = 0.1, alpha = 0.2) + 
  #geom_hline(yintercept = 85, linetype = "dotted", color = "red") + 
  theme_pubr() + 
  scale_color_manual(values = gf_preta_palette, labels = c("Mock + ΔpreTA" = expression(Mock~+~Delta*italic(preTA)), "preTA++" = expression(italic(preTA)^"++"))) + 
  scale_fill_manual(values = gf_preta_palette, labels = c("Mock + ΔpreTA" = expression(Mock~+~Delta*italic(preTA)), "preTA++" = expression(italic(preTA)^"++"))) + 
  annotate("text", x=1, y=80, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) +
  theme(legend.title = element_blank()) 

# Save
fn <- paste0(figdir, "GF_Weights.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3, device = cairo_pdf(family = "Arial"), dpi = 320)
```

## GF endpoint weight
```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Rescue/Gnoto_CAP_preTA_PeterS_Compiled.xlsx")
w <- read_excel(fn, sheet = "WeightsNorm") %>% as.data.frame()
w <- w %>% reshape2::melt(id.vars = "Day") %>% dplyr::rename(MouseID = variable, Weight = value)
w$Group <- case_when(as.numeric(w$MouseID) <= 12 ~ "Mock + ΔpreTA",
                     TRUE ~ "preTA++")

# Plot
w <- w %>% filter(Day == 6)
w$Day <- "Day 6"
p <- ggplot(w, aes(x = Group, y = Weight)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  ylab("Weight vs baseline (%)") + 
  scale_color_manual(values = gf_preta_palette, labels = c("Mock + ΔpreTA" = expression(Mock~+~Delta*italic(preTA)), "preTA++" = expression(italic(preTA)^"++"))) + 
  scale_fill_manual(values = gf_preta_palette, labels = c("Mock + ΔpreTA" = expression(Mock~+~Delta*italic(preTA)), "preTA++" = expression(italic(preTA)^"++"))) + 
  facet_wrap(~Day) + 
  stat_compare_means(method = "anova", label.x = 1.2, label.y = 99, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  theme(legend.position = "none") + 
  ylim(c(70, 105)) + 
  xlab("") + 
  theme(axis.text.x = element_text(angle = 30, hjust = 1, vjust = 1)) + 
  scale_x_discrete(labels = c("Mock + ΔpreTA" = expression(Mock~+~Delta*italic(preTA)), "preTA++" = expression(italic(preTA)^"++")))

fn <- paste0(figdir, "Gnoto_EndpointWeight.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3, device = cairo_pdf(family = "Arial"), dpi = 320)
```

## GF survival
```{r}
# Load data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Rescue/Gnoto_CAP_preTA_PeterS_Compiled.xlsx")
w <- read_excel(fn, sheet = "Survival") %>% as.data.frame() %>% dplyr::rename(Day = Time)
w$Group <- w$Group2

# object generation
surv_obj <- with(w, Surv(Day, Event))
fit <- survfit(Surv(Day, Event) ~ Group, data = w)
surv_df <- fortify(fit)

# Analysis
log_rank_test <- survdiff(Surv(Day, Event) ~ Group, data = w)
p_value <- 1 - pchisq(log_rank_test$chisq, df = 1) # Adjust the degrees of freedom if needed

# Plot the survival curve
ggsurv <- ggsurvplot(fit, data = w, pval = FALSE, conf.int = FALSE,
           risk.table = FALSE)
p <- ggsurv$plot
p <- p + scale_color_manual(values = gf_preta_surv_palette) +
  xlab("CAP duration (days)") + 
  ylab("Survival probability") + 
  theme_pubr() + 
  theme(legend.position = "none") + 
  annotate("text", x = 12, y = 1, label = paste("italic(p) == ", formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE) 

# Save
fn <- paste0(figdir, "GF_Survival.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3, device = cairo_pdf(family = "Arial"), dpi = 320)
```

## GF colon length
```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Gnoto_Rescue/Gnoto_CAP_preTA_PeterS_Compiled.xlsx")
col <- read_excel(fn, sheet = "Metadata") %>% as.data.frame()
col$Group2 <- case_when(col$Group == "OE" ~ "preTA++",
                        TRUE ~ "Mock + ΔpreTA")
col$Group2 <- factor(col$Group2, levels = c("Mock + ΔpreTA", "preTA++"))

# Plot
p <- ggplot(col, aes(x = Group2, y = `Colon length`, Group = Group2)) + 
  geom_boxplot(outlier.size = -1, aes(fill = Group2), alpha = 0.5) + 
  geom_jitter(width = 0.1, aes(color = Group2)) +
  theme_pubr() + 
  xlab("") + 
  ylab("Colon length (cm)") +
  theme(legend.position = "none") +
  stat_compare_means(method = "wilcox", label.y = 10.5, label.x = 1.25, aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) + 
  scale_color_manual(values = gf_preta_palette, labels = c("Mock + ΔpreTA" = expression(Mock~+~Delta*italic(preTA)), "preTA++" = expression(italic(preTA)^"++"))) +
  scale_fill_manual(values = gf_preta_palette, labels = c("Mock + ΔpreTA" = expression(Mock~+~Delta*italic(preTA)), "preTA++" = expression(italic(preTA)^"++"))) +
  ylim(c(6, 11)) + 
  theme(axis.text.x = element_text(angle = 35, hjust = 1, vjust = 1)) + 
  scale_x_discrete(labels = c("Mock + ΔpreTA" = expression(Mock~+~Delta*italic(preTA)), "preTA++" = expression(italic(preTA)^"++")))
  
# Save
fn <- paste0(figdir, "GF_ColonLength.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3, device = cairo_pdf(family = "Arial"), dpi = 320)
```

# Figure 6 - GO patients
```{r}
# Make directory
figdir <- paste0(dir, "Figure6/")
dir.create(figdir)

# Load 16S data
fn_meta <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/Metadata170523_with16Salpha.csv")
meta <- read.csv(fn_meta) %>% as.data.frame() %>% mutate(Patient_ID = replace(Patient_ID, Patient_ID == "9b", "9"))
stat_in <- meta %>% select(Sample_ID, Patient_ID, Treatment_Cycle, Reads, NumSVs_NotRarefied, NumSVs_Rarefied, Shannon_NotRarefied, Shannon_Rarefied, Cohort, Metadata_Radiation) %>%
  as.data.frame()

# Load date data
fn = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/GO_metadata_16S_WithDates.xlsx")
dates <- read_excel(fn) %>% as.data.frame() %>% select(c(Sample_ID, Patient_ID, Treatment_Cycle, Batch_16S, Days_Since_Tx))
dates <- dates %>% filter(!(Patient_ID %in% c("9", "9c")))
dates$Patient_ID[dates$Patient_ID == "9b"] <- "9"

# Add toxicity data
pts_tox <- c("1",  "25", "2", "11", "14", "51", "54", "7", "20", "38", "22", "21", "33", "31", "19", "4", "23", "35", "45", "13", "18")
dates$Toxicity <- case_when(dates$Patient_ID %in% pts_tox ~ "Yes",
                             TRUE ~ "No")
```

## GO Toxicity vs preTA level
Note that the trend holds when looking at just GI, HFS, or dose reduction as well!

```{r}
# Load preta data
fn1 <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/KO_stratified_K17722.csv")
fn2 <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/KO_stratified_K17723.csv")
preta1 <- fread(fn1)
preta2 <- fread(fn2)
preta <- rbind(preta1, preta2)
preta$MetaphlanID <- preta$Sample_ID

# Metadata
preta <- preta %>% group_by(MetaphlanID) %>% dplyr::summarise(Abundance = sum(RPKG)) %>% as.data.frame()
preta$Treatment_Cycle <- case_when(grepl("Baseline", preta$MetaphlanID) ~ "Baseline",
                                   grepl("C1D1", preta$MetaphlanID) ~ "C1D1",
                                   grepl("C1D3", preta$MetaphlanID) ~ "C1D3",
                                   grepl("C1D7", preta$MetaphlanID) ~ "C1D7",
                                   grepl("C2D1", preta$MetaphlanID) ~ "C2D1",
                                   grepl("C3D1", preta$MetaphlanID) ~ "C3D1",
                                   grepl("Final", preta$MetaphlanID) ~ "Final",
                                   TRUE ~ "Other")
preta$Patient_ID <- sub(".*Pt", "", preta$MetaphlanID) 
preta$Patient_ID <- sub("_.*", "", preta$Patient_ID) 
preta$Sample_ID <- paste0("GO-Pt", preta$Patient_ID, "-", preta$Treatment_Cycle)

# Remove duplicates
dups = preta %>% 
  group_by(Sample_ID) %>% 
  dplyr::summarise(n = n()) %>% 
  as.data.frame() %>% 
  filter(n >= 2) %>%
  select(Sample_ID) %>% 
  pull()
preta <- preta %>% filter(!(MetaphlanID %in% c("GO_Pt40_C1D7_S82", "GO_Pt1_C1D7_S56", "GO_Pt12_C2D1_S7")))

# Compare expression in Tox vs No Tox patients. 
stomatitis <- c("1", "25") # "stomatitis" 
hfs <- c("25", "1", "2", "11", "14", "20", "21") # "hfs", "hand", "neuropathy", "erythrodys*" - ensure it's new during treatment
reduc <- c("7", "11", "25", "20", "38", "1") # "reduc", "decreas"
gi <- c("1", "7", "20", "21", "54", "51", "22", "33", "31") #  "diarrhea" or "constipation" - ensure it's new during treatment
nausea <- c("19", "33", "4", "54", "1") # "nausea" 
fatigue <- c("23", "4", "11", "35", "45", "54") # "fatigue"
neutropenia <- c("4") # "neutropenia"

# From validated clinical trial spreadsheet, "PCB". Note that the Patient 6 toxicities are all several months after the initial cycles of treatment, so we exclude these
hfs_validated <- c("1", "2", "13", "14", "18") # "Definite"
notox_validated <- c("15", "39") # no hfs

# Select patients
pts <- unique(c(stomatitis, hfs, reduc, gi, fatigue, nausea, neutropenia, hfs_validated)) 

# Metadata
fn = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/Metadata170523.csv")
met <- read.csv(fn) %>% as.data.frame()
metadata_full <- met %>% select(-c("Treatment_Cycle", "Patient_ID", "Sample_ID"))
preta$MetaphlanID <- gsub("_S.*", "", preta$MetaphlanID)
preta <- inner_join(preta, metadata_full, by = "MetaphlanID")
preta$Patient_ID[preta$Patient_ID == "9b"] <- "9"
preta$Dose <- case_when(preta$Patient_ID %in% pts ~ "Toxicity",
                        TRUE ~ "No documented toxicity")

# Check missing patients
met_add <- met
met_add$Patient_ID[met_add$Patient_ID == "9b"] <- "9"
missing_pts <- setdiff(unique(met_add$Patient_ID), unique(preta$Patient_ID))

# Average baseline and C1D1 preTA (or if only one available, use that one)
bms <- preta %>% filter(Treatment_Cycle %in% c("Baseline", "C1D1")) 
bms <- bms %>% group_by(Patient_ID, Dose, Metadata_Radiation) %>% dplyr::summarise(Abundance = mean(Abundance))

# Add missing patients - 11, 7, 53
pts_lost <- setdiff(unique(preta$Patient_ID), unique(bms$Patient_ID))
pt_add <- c(pts_lost, missing_pts)
bms_add <- met %>% filter(Patient_ID %in% pt_add) %>% 
  filter(Treatment_Cycle == "Baseline") %>% 
  select(all_of(c("Patient_ID", "Metadata_Radiation")))
bms_add$Abundance <- 0
bms <- rbind(bms, bms_add)

# Fix dosing info
bms$Dose <- case_when(bms$Patient_ID %in% pts ~ "Toxicity",
                        TRUE ~ "No toxicity")
bms$Group <- factor(bms$Dose, levels = c("No toxicity", "Toxicity"))

# Plot
offset = min(bms$Abundance[bms$Abundance > 0])/2
p_b <- ggplot(bms, aes(x = Group, y = Abundance + offset)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  xlab("") + 
  ylab("PreTA level (RPKG)") +
  scale_y_log10(limits = c(offset/2, 2)) + 
  scale_color_manual(values = tox_palette_2) + 
  scale_fill_manual(values = tox_palette_2) + 
  stat_compare_means(label.x = 1.2, label.y = log10(1.5), method = "wilcox.test", method.args = list(alternative = "greater"), aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) +
  theme(legend.position = "none") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) 

# Save
fn <- paste0(figdir, "GO_PtTox.pdf")
ggsave(filename = fn, plot = p_b, width = 2, height = 3)
```

## GO Toxicity Frequency
```{r}
# Compare expression in Tox vs No Tox patients. 
stomatitis <- c("1", "25") # "stomatitis" 
hfs <- c("25", "1", "2", "11", "14", "20", "21") # "hfs", "hand", "neuropathy", "erythrodys*" - ensure it's new during treatment
reduc <- c("7", "11", "25", "20", "38", "1") # "reduc", "decreas"
gi <- c("1", "7", "20", "21", "54", "51", "22", "33", "31") #  "diarrhea" or "constipation" - ensure it's new during treatment
nausea <- c("19", "33", "4", "54", "1") # "nausea" 
fatigue <- c("23", "4", "11", "35", "45", "54") # "fatigue"
neutropenia <- c("4") # "neutropenia"
hfs_validated <- c("1", "2", "13", "14", "18") # "Definite"

# Groupings
all = unique(c(stomatitis, hfs, reduc, gi, nausea, fatigue, neutropenia, hfs_validated)) 
all_hfs = unique(c(hfs, hfs_validated))

# Note that "dose reduction" includes interruptions; HFS = hand-foot syndrome
Toxicity = c("Any toxicity", "HFS", "Dose reduction", "GI toxicity", "Nausea", "Fatigue", "Neutropenia")
Number = c(length(all), length(all_hfs), length(reduc), length(gi), length(nausea), length(fatigue), length(neutropenia))
ToxDF = data.frame(Toxicity = Toxicity, Number = Number, Percent = round(Number/40*100))

# Plot
# Number of patients is out of 40
p <- ggplot(ToxDF, aes(x = reorder(Toxicity, Number), y = Number)) +
  geom_bar(stat = "identity", color = "black", fill = "#FF000080") +
  coord_flip() + # Flip coordinates to have 'Toxicity' on the y-axis
  geom_text(aes(label = paste0(Number, " (", Percent, "%)")), hjust = -0.1) + # Add text labels
  labs(x = "", y = "Incidence [N (%)]") +
  theme_pubr() + 
  theme(legend.position = "none") + 
  ylim(0, 40)

# Save
fn <- paste0(figdir, "GO_ToxBargraph.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## GO Beta vs preTA level 
```{r}
## Create beta diversity dataframe
# Load distance data
fn = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/DistanceFromBaselineEuclidean_FinalIncluded.csv")
b <- read.csv(fn)
b$Patient_ID[b$Patient_ID == "9b"] <- "9"
b <- b %>% select(Patient_ID, Treatment_Cycle, value)

# Load metadata
dates_b <- dates %>% select(Treatment_Cycle, Patient_ID, Days_Since_Tx)
b <- left_join(b, dates_b, by = c("Patient_ID", "Treatment_Cycle")) 

# Add toxicity data
b$Toxicity <- case_when(b$Patient_ID %in% pts_tox ~ "Yes",
                        TRUE ~ "No")

# Filter to cycle of interest
b <- b %>% filter(Treatment_Cycle == "C1D7")

# Load preta data
fn1 <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/KO_stratified_K17722.csv")
fn2 <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/KO_stratified_K17723.csv")
preta1 <- fread(fn1)
preta2 <- fread(fn2)
preta <- rbind(preta1, preta2)
preta$MetaphlanID <- preta$Sample_ID

# Add metadata
preta <- preta %>% group_by(MetaphlanID) %>% dplyr::summarise(Abundance = sum(RPKG)) %>% as.data.frame()
preta$Treatment_Cycle <- case_when(grepl("Baseline", preta$MetaphlanID) ~ "Baseline",
                                   grepl("C1D1", preta$MetaphlanID) ~ "C1D1",
                                   grepl("C1D3", preta$MetaphlanID) ~ "C1D3",
                                   grepl("C1D7", preta$MetaphlanID) ~ "C1D7",
                                   grepl("C2D1", preta$MetaphlanID) ~ "C2D1",
                                   grepl("C3D1", preta$MetaphlanID) ~ "C3D1",
                                   grepl("Final", preta$MetaphlanID) ~ "Final",
                                   TRUE ~ "Other")
preta$Patient_ID <- sub(".*Pt", "", preta$MetaphlanID) 
preta$Patient_ID <- sub("_.*", "", preta$Patient_ID) 
preta$Sample_ID <- paste0("GO-Pt", preta$Patient_ID, "-", preta$Treatment_Cycle)

# Remove duplicates
dups = preta %>% 
  group_by(Sample_ID) %>% 
  dplyr::summarise(n = n()) %>% 
  as.data.frame() %>% 
  filter(n >= 2) %>%
  select(Sample_ID) %>% 
  pull()
preta <- preta %>% filter(!(MetaphlanID %in% c("GO_Pt40_C1D7_S82", "GO_Pt1_C1D7_S56", "GO_Pt12_C2D1_S7")))

# Baseline only
preta_baseline <- preta %>% filter(Treatment_Cycle %in% c("C1D1")) %>%
  select(Patient_ID, Abundance) %>%
  group_by(Patient_ID) %>% 
  dplyr::summarise(Abundance = mean(Abundance)) %>%
  as.data.frame() 

# Add missing patients 
pt_add <- setdiff(unique(dates$Patient_ID), unique(preta_baseline$Patient_ID))
pt_add <- pt_add[!is.na(pt_add)]
preta_add <- data.frame(Patient_ID = pt_add, Abundance = rep(0, length(pt_add)))
preta_baseline <- rbind(preta_add, preta_baseline)

# Merge
b_plot <- left_join(b, preta_baseline, by = "Patient_ID")

# Stats
spearman_test <- cor.test(b_plot$Abundance, b_plot$value, method = "spearman")
correlation <- spearman_test$estimate
p_value <- spearman_test$p.value

# Plot
p <- ggplot(b_plot, aes(x = Abundance, y = value)) + 
  geom_point() +
  geom_smooth(method = "lm", level = 0.95) + 
  theme_pubr() +
  annotate("text", x = 0.6, y = 50, label = sprintf("Spearman's rho = %.2f, p = %.2f", correlation, p_value))

# Alternative plot
b_plot$BetaChange <- case_when(b_plot$value <= 30 ~ "Low",
                            TRUE ~ "High")
p <- ggplot(b_plot, aes(x = BetaChange, y = Abundance)) + 
  geom_point() + 
  theme_pubr() + 
  stat_compare_means(method = "wilcox.test") + 
  xlab("Beta diversity shift") + 
  ylab("preTA level (RPKG)")

# Alternative plot 2
epsilon = 0.01 # adding epsilon to median heightens significance
b_plot$Group <- case_when(b_plot$Abundance <= median(b_plot$Abundance) ~ "Low",
                            TRUE ~ "High")
p <- ggplot(b_plot, aes(x = Group, y = value)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  ylab(expression(beta~" shift (CLR-Euclidean)")) + 
  xlab("preTA level") +
  scale_color_manual(values = hilo_pt_palette) + 
  scale_fill_manual(values = hilo_pt_palette) + 
  stat_compare_means(method = "wilcox", label.x = 1.2, label.y = 58, method.args = list(alternative = "less"), aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) +
  theme(legend.position = "none") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  ylim(c(15, 60))
  
# Save
fn <- paste0(figdir, "GO_BetaPreTA.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)
```

## GO Alpha vs preTA level
```{r}
# Load preta data
fn1 <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/KO_stratified_K17722.csv")
fn2 <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/KO_stratified_K17723.csv")
preta1 <- fread(fn1)
preta2 <- fread(fn2)
preta <- rbind(preta1, preta2)
preta$MetaphlanID <- preta$Sample_ID

# Add metadata
preta <- preta %>% group_by(MetaphlanID) %>% dplyr::summarise(Abundance = sum(RPKG)) %>% as.data.frame()
preta$Treatment_Cycle <- case_when(grepl("Baseline", preta$MetaphlanID) ~ "Baseline",
                                   grepl("C1D1", preta$MetaphlanID) ~ "C1D1",
                                   grepl("C1D3", preta$MetaphlanID) ~ "C1D3",
                                   grepl("C1D7", preta$MetaphlanID) ~ "C1D7",
                                   grepl("C2D1", preta$MetaphlanID) ~ "C2D1",
                                   grepl("C3D1", preta$MetaphlanID) ~ "C3D1",
                                   grepl("Final", preta$MetaphlanID) ~ "Final",
                                   TRUE ~ "Other")
preta$Patient_ID <- sub(".*Pt", "", preta$MetaphlanID) 
preta$Patient_ID <- sub("_.*", "", preta$Patient_ID) 
preta$Sample_ID <- paste0("GO-Pt", preta$Patient_ID, "-", preta$Treatment_Cycle)

# Remove duplicates
dups = preta %>% 
  group_by(Sample_ID) %>% 
  dplyr::summarise(n = n()) %>% 
  as.data.frame() %>% 
  filter(n >= 2) %>%
  select(Sample_ID) %>% 
  pull()
preta <- preta %>% filter(!(MetaphlanID %in% c("GO_Pt40_C1D7_S82", "GO_Pt1_C1D7_S56", "GO_Pt12_C2D1_S7")))

# Baseline only
preta_baseline <- preta %>% filter(Treatment_Cycle %in% c("Baseline", "C1D1")) %>%
                                     select(Patient_ID, Abundance) %>% 
                                     group_by(Patient_ID) %>% 
                                     dplyr::summarise(Abundance = mean(Abundance)) %>%
                                     as.data.frame()

# Add missing patients 
pt_add <- setdiff(unique(dates$Patient_ID), unique(preta_baseline$Patient_ID))
pt_add <- pt_add[!is.na(pt_add)]
preta_add <- data.frame(Patient_ID = pt_add, Abundance = rep(0, length(pt_add)))
preta_baseline <- rbind(preta_add, preta_baseline)

# Add to alpha diversity data
stat <- stat_in %>% left_join(preta_baseline, by = "Patient_ID")

# Compute delta alpha relative to baseline
delta_pre <- stat %>% filter(Treatment_Cycle %in% c("C1D1")) %>% arrange(Patient_ID) 
delta_pre = delta_pre[!duplicated(delta_pre$Patient_ID),]
delta_post <- stat %>% filter(Treatment_Cycle == "C1D7")
pts = intersect(delta_pre$Patient_ID, delta_post$Patient_ID)
delta_pre <- delta_pre %>% filter(Patient_ID %in% pts); delta_post <- delta_post %>% filter(Patient_ID %in% pts)
delta <- inner_join(delta_pre, delta_post, by = "Patient_ID") # pre = .x, post = .y
delta$Delta <- delta$Shannon_NotRarefied.y - delta$Shannon_NotRarefied.x

# Stats
spearman_test <- cor.test(delta$Abundance.x, delta$Delta, method = "pearson")
correlation <- spearman_test$estimate
p_value <- spearman_test$p.value

# Plot
p <- ggplot(delta, aes(x = Abundance.x, y = Delta)) + 
  geom_point() + 
  geom_smooth(method = "lm", level = 0.95) + 
  theme_pubr() +
  annotate("text", x = 0.6, y = 1, label = sprintf("Spearman's rho = %.2f, p = %.2f", correlation, p_value))

# Alternative plot
delta$AlphaChange <- case_when(delta$Delta <= 0 ~ "Decreased",
                            TRUE ~ "Not decreased")
p <- ggplot(delta, aes(x = AlphaChange, y = Abundance.x)) + 
  geom_point() + 
  theme_pubr() + 
  stat_compare_means(method = "wilcox") + 
  xlab("Alpha diversity shift") + 
  ylab("preTA level (RPKG)")

# Alternative plot 2
delta$Group <- case_when(delta$Abundance.x <= mean(delta$Abundance.x) ~ "Low",
                            TRUE ~ "High")
p <- ggplot(delta, aes(x = Group, y = Delta)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  ylab(expression(alpha~"diversity shift (Shannon)")) + 
  xlab("preTA level") +
  scale_color_manual(values = hilo_pt_palette) + 
  scale_fill_manual(values = hilo_pt_palette) + 
  stat_compare_means(label.x = 1.2, label.y = 0.47, method = "wilcox.test", method.args = list(alternative = "greater"), aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) +
  theme(legend.position = "none") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  ylim(c(-0.5, 0.5))

# Save
fn <- paste0(figdir, "GO_AlphaPreTA.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)
```

# Figure S15 - sex preTA-tox interaction
```{r}
figdir <- paste0(dir, "FigureS15/")
dir.create(figdir)

# Load 16S data
fn_meta <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/Metadata170523_with16Salpha.csv")
meta <- read.csv(fn_meta) %>% as.data.frame() %>% mutate(Patient_ID = replace(Patient_ID, Patient_ID == "9b", "9"))
stat_in <- meta %>% select(Sample_ID, Patient_ID, Treatment_Cycle, Reads, NumSVs_NotRarefied, NumSVs_Rarefied, Shannon_NotRarefied, Shannon_Rarefied, Cohort, Metadata_Radiation) %>%
  as.data.frame()

# Load date data
fn = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/GO_metadata_16S_WithDates.xlsx")
dates <- read_excel(fn) %>% as.data.frame() %>% select(c(Sample_ID, Patient_ID, Treatment_Cycle, Batch_16S, Days_Since_Tx))
dates <- dates %>% filter(!(Patient_ID %in% c("9", "9c")))
dates$Patient_ID[dates$Patient_ID == "9b"] <- "9"

# Add toxicity data
pts_tox <- c("1",  "25", "2", "11", "14", "51", "54", "7", "20", "38", "22", "21", "33", "31", "19", "4", "23", "35", "45", "13", "18")
dates$Toxicity <- case_when(dates$Patient_ID %in% pts_tox ~ "Yes",
                             TRUE ~ "No")
```

## GO toxicity vs preTA level - Sex
```{r}
# Load preta data
fn1 <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/KO_stratified_K17722.csv")
fn2 <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/KO_stratified_K17723.csv")
preta1 <- fread(fn1)
preta2 <- fread(fn2)
preta <- rbind(preta1, preta2)
preta$MetaphlanID <- preta$Sample_ID

# Metadata
preta <- preta %>% group_by(MetaphlanID) %>% dplyr::summarise(Abundance = sum(RPKG)) %>% as.data.frame()
preta$Treatment_Cycle <- case_when(grepl("Baseline", preta$MetaphlanID) ~ "Baseline",
                                   grepl("C1D1", preta$MetaphlanID) ~ "C1D1",
                                   grepl("C1D3", preta$MetaphlanID) ~ "C1D3",
                                   grepl("C1D7", preta$MetaphlanID) ~ "C1D7",
                                   grepl("C2D1", preta$MetaphlanID) ~ "C2D1",
                                   grepl("C3D1", preta$MetaphlanID) ~ "C3D1",
                                   grepl("Final", preta$MetaphlanID) ~ "Final",
                                   TRUE ~ "Other")
preta$Patient_ID <- sub(".*Pt", "", preta$MetaphlanID) 
preta$Patient_ID <- sub("_.*", "", preta$Patient_ID) 
preta$Sample_ID <- paste0("GO-Pt", preta$Patient_ID, "-", preta$Treatment_Cycle)

# Remove duplicates
dups = preta %>% 
  group_by(Sample_ID) %>% 
  dplyr::summarise(n = n()) %>% 
  as.data.frame() %>% 
  filter(n >= 2) %>%
  select(Sample_ID) %>% 
  pull()
preta <- preta %>% filter(!(MetaphlanID %in% c("GO_Pt40_C1D7_S82", "GO_Pt1_C1D7_S56", "GO_Pt12_C2D1_S7")))

# Compare expression in Tox vs No Tox patients. 
stomatitis <- c("1", "25") # "stomatitis" 
hfs <- c("25", "1", "2", "11", "14", "20", "21") # "hfs", "hand", "neuropathy", "erythrodys*" - ensure it's new during treatment
reduc <- c("7", "11", "25", "20", "38", "1") # "reduc", "decreas"
gi <- c("1", "7", "20", "21", "54", "51", "22", "33", "31") #  "diarrhea" or "constipation" - ensure it's new during treatment
nausea <- c("19", "33", "4", "54", "1") # "nausea" 
fatigue <- c("23", "4", "11", "35", "45", "54") # "fatigue"
neutropenia <- c("4") # "neutropenia"

# From validated clinical trial spreadsheet, "PCB". Note that the Patient 6 toxicities are all several months after the initial cycles of treatment, so we exclude these
hfs_validated <- c("1", "2", "13", "14", "18") # "Definite"
notox_validated <- c("15", "39") # no hfs

# Select patients
pts <- unique(c(stomatitis, hfs, reduc, gi, fatigue, nausea, neutropenia, hfs_validated)) 

# Metadata
fn = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/Metadata170523.csv")
met <- read.csv(fn) %>% as.data.frame()
metadata_full <- met %>% select(-c("Treatment_Cycle", "Patient_ID", "Sample_ID"))
preta$MetaphlanID <- gsub("_S.*", "", preta$MetaphlanID)
preta <- inner_join(preta, metadata_full, by = "MetaphlanID")
preta$Patient_ID[preta$Patient_ID == "9b"] <- "9"
preta$Dose <- case_when(preta$Patient_ID %in% pts ~ "Toxicity",
                        TRUE ~ "No documented toxicity")

# Check missing patients
met_add <- met
met_add$Patient_ID[met_add$Patient_ID == "9b"] <- "9"
missing_pts <- setdiff(unique(met_add$Patient_ID), unique(preta$Patient_ID))

# Average baseline and C1D1 preTA (or if only one available, use that one)
bms <- preta %>% filter(Treatment_Cycle %in% c("Baseline", "C1D1")) 
bms <- bms %>% group_by(Patient_ID, Dose, Metadata_Radiation, Metadata_Sex) %>% dplyr::summarise(Abundance = mean(Abundance))

# Add missing patients - 11, 7, 53
pts_lost <- setdiff(unique(preta$Patient_ID), unique(bms$Patient_ID))
pt_add <- c(pts_lost, missing_pts)
bms_add <- met %>% filter(Patient_ID %in% pt_add) %>% 
  filter(Treatment_Cycle == "Baseline") %>% 
  select(all_of(c("Patient_ID", "Metadata_Radiation", "Metadata_Sex")))
bms_add$Abundance <- 0
bms <- rbind(bms, bms_add)

# Fix dosing info
offset = min(bms$Abundance[bms$Abundance > 0])/2
both <- bms
both$Metadata_Sex = "Combined"
bms <- rbind(bms, both)

bms$Dose <- case_when(bms$Patient_ID %in% pts ~ "Toxicity",
                        TRUE ~ "No toxicity")

bms$Group <- factor(bms$Dose, levels = c("No toxicity", "Toxicity"))

p_b <- ggplot(bms, aes(x = Group, y = Abundance + offset)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  theme_pubr() + 
  xlab("") + 
  facet_wrap(~Metadata_Sex) + 
  ylab("PreTA level (RPKG)") +
  scale_y_log10(limits = c(offset/2, 2)) + 
  scale_color_manual(values = tox_palette_2) + 
  scale_fill_manual(values = tox_palette_2) + 
  stat_compare_means(label.x = 1.2, label.y = log10(1.5), method = "wilcox.test", aes(label = paste0("italic(p) ==", after_stat(sprintf('%.3f', p / 2)))), parse = TRUE) +
  theme(legend.position = "none") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) 

# Save
fn <- paste0(figdir, "GO_PtTox_MF.pdf")
ggsave(filename = fn, plot = p_b, width = 3.5, height = 3)
```

## GO Beta vs preTA level - Sex
```{r}
sex <- meta %>% select(Patient_ID, Metadata_Sex) %>% distinct()
## Create beta diversity dataframe
# Load distance data
fn = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/DistanceFromBaselineEuclidean_FinalIncluded.csv")
b <- read.csv(fn)
b$Patient_ID[b$Patient_ID == "9b"] <- "9"
b <- b %>% select(Patient_ID, Treatment_Cycle, value)

# Load metadata
dates_b <- dates %>% select(Treatment_Cycle, Patient_ID, Days_Since_Tx)
b <- left_join(b, dates_b, by = c("Patient_ID", "Treatment_Cycle")) 

# Add toxicity data
b$Toxicity <- case_when(b$Patient_ID %in% pts_tox ~ "Yes",
                        TRUE ~ "No")

# Filter to cycle of interest
b <- b %>% filter(Treatment_Cycle == "C1D7")

# Load preta data
fn1 <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/KO_stratified_K17722.csv")
fn2 <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/KO_stratified_K17723.csv")
preta1 <- fread(fn1)
preta2 <- fread(fn2)
preta <- rbind(preta1, preta2)
preta$MetaphlanID <- preta$Sample_ID

# Add metadata
preta <- preta %>% group_by(MetaphlanID) %>% dplyr::summarise(Abundance = sum(RPKG)) %>% as.data.frame()
preta$Treatment_Cycle <- case_when(grepl("Baseline", preta$MetaphlanID) ~ "Baseline",
                                   grepl("C1D1", preta$MetaphlanID) ~ "C1D1",
                                   grepl("C1D3", preta$MetaphlanID) ~ "C1D3",
                                   grepl("C1D7", preta$MetaphlanID) ~ "C1D7",
                                   grepl("C2D1", preta$MetaphlanID) ~ "C2D1",
                                   grepl("C3D1", preta$MetaphlanID) ~ "C3D1",
                                   grepl("Final", preta$MetaphlanID) ~ "Final",
                                   TRUE ~ "Other")
preta$Patient_ID <- sub(".*Pt", "", preta$MetaphlanID) 
preta$Patient_ID <- sub("_.*", "", preta$Patient_ID) 
preta$Sample_ID <- paste0("GO-Pt", preta$Patient_ID, "-", preta$Treatment_Cycle)

# Remove duplicates
dups = preta %>% 
  group_by(Sample_ID) %>% 
  dplyr::summarise(n = n()) %>% 
  as.data.frame() %>% 
  filter(n >= 2) %>%
  select(Sample_ID) %>% 
  pull()
preta <- preta %>% filter(!(MetaphlanID %in% c("GO_Pt40_C1D7_S82", "GO_Pt1_C1D7_S56", "GO_Pt12_C2D1_S7")))

# Baseline only
preta_baseline <- preta %>% filter(Treatment_Cycle %in% c("C1D1")) %>%
  select(Patient_ID, Abundance) %>%
  group_by(Patient_ID) %>% 
  dplyr::summarise(Abundance = mean(Abundance)) %>%
  as.data.frame() 

# Add missing patients 
pt_add <- setdiff(unique(dates$Patient_ID), unique(preta_baseline$Patient_ID))
pt_add <- pt_add[!is.na(pt_add)]
preta_add <- data.frame(Patient_ID = pt_add, Abundance = rep(0, length(pt_add)))
preta_baseline <- rbind(preta_add, preta_baseline)

# Merge
b_plot <- left_join(b, preta_baseline, by = "Patient_ID")

# Stats
spearman_test <- cor.test(b_plot$Abundance, b_plot$value, method = "spearman")
correlation <- spearman_test$estimate
p_value <- spearman_test$p.value

# Plot
p <- ggplot(b_plot, aes(x = Abundance, y = value)) + 
  geom_point() +
  geom_smooth(method = "lm", level = 0.95) + 
  theme_pubr() +
  annotate("text", x = 0.6, y = 50, label = sprintf("Spearman's rho = %.2f, p = %.2f", correlation, p_value))

# Alternative plot
b_plot$BetaChange <- case_when(b_plot$value <= 30 ~ "Low",
                            TRUE ~ "High")
p <- ggplot(b_plot, aes(x = BetaChange, y = Abundance)) + 
  geom_point() + 
  theme_pubr() + 
  stat_compare_means(method = "wilcox.test") + 
  xlab("Beta diversity shift") + 
  ylab("preTA level (RPKG)")

# Alternative plot 2
epsilon = 0.01 # adding epsilon to median heightens significance
b_plot$Group <- case_when(b_plot$Abundance <= median(b_plot$Abundance) ~ "Low",
                            TRUE ~ "High")
b_plot <- left_join(b_plot, sex, by = "Patient_ID")
sex_df <- b_plot; sex_df$Metadata_Sex <- "Combined"
b_plot <- rbind(b_plot, sex_df)
p <- ggplot(b_plot, aes(x = Group, y = value)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  theme_pubr() + 
  ylab(expression(beta~" shift (CLR-Euclidean)")) + 
  xlab("preTA level") +
  facet_wrap(~Metadata_Sex) + 
  scale_color_manual(values = hilo_pt_palette) + 
  scale_fill_manual(values = hilo_pt_palette) + 
  stat_compare_means(method = "wilcox", label.x = 1.2, label.y = 58, aes(label = paste0("italic(p) ==", after_stat(sprintf('%.3f', p / 2)))), parse = TRUE) +
  theme(legend.position = "none") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  ylim(c(15, 60))
  
# Save
fn <- paste0(figdir, "GO_BetaPreTA_Sex.pdf")
ggsave(filename = fn, plot = p, width = 3.5, height = 3)

pt_beta <- b_plot %>% select(Patient_ID) %>% pull() %>% unique()
```

## GO Alpha vs preTA level - Sex
```{r}
sex <- meta %>% select(Patient_ID, Metadata_Sex) %>% distinct()

# Load preta data
fn1 <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/KO_stratified_K17722.csv")
fn2 <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/KO_stratified_K17723.csv")
preta1 <- fread(fn1)
preta2 <- fread(fn2)
preta <- rbind(preta1, preta2)
preta$MetaphlanID <- preta$Sample_ID

# Add metadata
preta <- preta %>% group_by(MetaphlanID) %>% dplyr::summarise(Abundance = sum(RPKG)) %>% as.data.frame()
preta$Treatment_Cycle <- case_when(grepl("Baseline", preta$MetaphlanID) ~ "Baseline",
                                   grepl("C1D1", preta$MetaphlanID) ~ "C1D1",
                                   grepl("C1D3", preta$MetaphlanID) ~ "C1D3",
                                   grepl("C1D7", preta$MetaphlanID) ~ "C1D7",
                                   grepl("C2D1", preta$MetaphlanID) ~ "C2D1",
                                   grepl("C3D1", preta$MetaphlanID) ~ "C3D1",
                                   grepl("Final", preta$MetaphlanID) ~ "Final",
                                   TRUE ~ "Other")
preta$Patient_ID <- sub(".*Pt", "", preta$MetaphlanID) 
preta$Patient_ID <- sub("_.*", "", preta$Patient_ID) 
preta$Sample_ID <- paste0("GO-Pt", preta$Patient_ID, "-", preta$Treatment_Cycle)

# Remove duplicates
dups = preta %>% 
  group_by(Sample_ID) %>% 
  dplyr::summarise(n = n()) %>% 
  as.data.frame() %>% 
  filter(n >= 2) %>%
  select(Sample_ID) %>% 
  pull()
preta <- preta %>% filter(!(MetaphlanID %in% c("GO_Pt40_C1D7_S82", "GO_Pt1_C1D7_S56", "GO_Pt12_C2D1_S7")))

# Baseline only
preta_baseline <- preta %>% filter(Treatment_Cycle %in% c("Baseline", "C1D1")) %>%
                                     select(Patient_ID, Abundance) %>% 
                                     group_by(Patient_ID) %>% 
                                     dplyr::summarise(Abundance = mean(Abundance)) %>%
                                     as.data.frame()

# Add missing patients 
pt_add <- setdiff(unique(dates$Patient_ID), unique(preta_baseline$Patient_ID))
pt_add <- pt_add[!is.na(pt_add)]
preta_add <- data.frame(Patient_ID = pt_add, Abundance = rep(0, length(pt_add)))
preta_baseline <- rbind(preta_add, preta_baseline)

# Add to alpha diversity data
stat <- stat_in %>% left_join(preta_baseline, by = "Patient_ID")

# Compute delta alpha relative to baseline
delta_pre <- stat %>% filter(Treatment_Cycle %in% c("C1D1")) %>% arrange(Patient_ID) 
delta_pre = delta_pre[!duplicated(delta_pre$Patient_ID),]
delta_post <- stat %>% filter(Treatment_Cycle == "C1D7")
pts = intersect(delta_pre$Patient_ID, delta_post$Patient_ID)
delta_pre <- delta_pre %>% filter(Patient_ID %in% pts); delta_post <- delta_post %>% filter(Patient_ID %in% pts)
delta <- inner_join(delta_pre, delta_post, by = "Patient_ID") # pre = .x, post = .y
delta$Delta <- delta$Shannon_NotRarefied.y - delta$Shannon_NotRarefied.x

# Stats
spearman_test <- cor.test(delta$Abundance.x, delta$Delta, method = "pearson")
correlation <- spearman_test$estimate
p_value <- spearman_test$p.value

# Plot
p <- ggplot(delta, aes(x = Abundance.x, y = Delta)) + 
  geom_point() + 
  geom_smooth(method = "lm", level = 0.95) + 
  theme_pubr() +
  annotate("text", x = 0.6, y = 1, label = sprintf("Spearman's rho = %.2f, p = %.2f", correlation, p_value))

# Alternative plot
delta$AlphaChange <- case_when(delta$Delta <= 0 ~ "Decreased",
                            TRUE ~ "Not decreased")
p <- ggplot(delta, aes(x = AlphaChange, y = Abundance.x)) + 
  geom_point() + 
  theme_pubr() + 
  stat_compare_means(method = "wilcox") + 
  xlab("Alpha diversity shift") + 
  ylab("preTA level (RPKG)")

# Alternative plot 2
delta$Group <- case_when(delta$Abundance.x <= mean(delta$Abundance.x) ~ "Low",
                            TRUE ~ "High")
delta <- left_join(delta, sex, by = "Patient_ID")
sex_df <- delta; sex_df$Metadata_Sex <- "Combined"
delta <- rbind(delta, sex_df)
p <- ggplot(delta, aes(x = Group, y = Delta)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  theme_pubr() + 
  facet_wrap(~Metadata_Sex) + 
  ylab(expression(alpha~"diversity shift (Shannon)")) + 
  xlab("preTA level") +
  scale_color_manual(values = hilo_pt_palette) + 
  scale_fill_manual(values = hilo_pt_palette) + 
  stat_compare_means(label.x = 1.2, label.y = 0.45, method = "wilcox.test", aes(label = paste0("italic(p) ==", after_stat(sprintf('%.3f', p / 2)))), parse = TRUE) + 
  theme(legend.position = "none") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  ylim(c(-0.5, 0.5))

# Save
fn <- paste0(figdir, "GO_AlphaPreTA_Sex.pdf")
ggsave(filename = fn, plot = p, width = 3.5, height = 3)

pt_alpha <- delta %>% select(Patient_ID) %>% pull() %>% unique()
```

## preTA vs time by sex
```{r}
# Load date data
fn = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/GO_metadata_16S_WithDates.xlsx")
dates <- read_excel(fn) %>% as.data.frame() %>% select(c(Sample_ID, Patient_ID, Treatment_Cycle, Batch_16S, Days_Since_Tx, Sex))
dates <- dates %>% filter(!(Patient_ID %in% c("9", "9c")))
dates$Patient_ID[dates$Patient_ID == "9b"] <- "9"

hits <- c("K17723", "K17722")
directions <- rep("Enriched", 2)

fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/genefamilies_unstratified_nonzero0.csv")
d <- fread(fn) %>% filter(KO %in% hits) %>% as.data.frame() %>% select(-V1)

# Melt
a <- d %>% reshape2::melt(by = "GeneFamily")
colnames(a) <- c("GeneFamily", "Sample_ID", "Abundance")
a$Sample_ID <- gsub("_", "-", a$Sample_ID)
a <- a %>% group_by(Sample_ID) %>% 
  dplyr::summarise(Abundance = sum(Abundance)) %>%
  as.data.frame()

# Merge with metadata, direction
a <- left_join(a, dates, by = "Sample_ID") 
a_plot <- a
a_plot$Days_Since_Tx <- as.numeric(a_plot$Days_Since_Tx)
a_plot$Days_Since_Tx <- case_when(a_plot$Days_Since_Tx < 0 ~ 0, TRUE ~ a_plot$Days_Since_Tx)
a_plot <- a_plot %>% filter(Days_Since_Tx <= 9) 
offset = min(a_plot$Abundance[a_plot$Abundance > 0])

# Don't bias towards patients that submitted more samples! Take the average of each patient's samples at each timepoint
a_plot$Group <- case_when(a_plot$Days_Since_Tx <= 1 ~ "0-1",
                          a_plot$Days_Since_Tx <= 5 ~ "2-5",
                          TRUE ~ "6-9")
a_plot$Group <- factor(a_plot$Group)
a_plot <- a_plot %>% group_by(Patient_ID, Group, Sex) %>%
  dplyr::summarise(Abundance = mean(Abundance)) %>%
  as.data.frame()
pt_paired <- a_plot %>% group_by(Patient_ID) %>%
  dplyr::summarise(n = n()) %>% 
  filter(n == 3) %>% 
  select(Patient_ID) %>%
  pull()
sex_all <- a_plot
sex_all$Sex <- "Combined"
a_plot <- rbind(a_plot, sex_all)
p <- ggplot(a_plot, aes(x = Group, y = Abundance + offset)) + 
  geom_boxplot(color = "black", alpha = 0.5, aes(group = Group, fill = Group), outlier.size = -1) +
  theme_pubr() + 
  scale_fill_manual(values = c("0-1" = "lightgrey", "2-5" = "#4E4EFF", "6-9" = "#1414FF")) + 
  ylab("preTA abundance (RPKG)") + 
  xlab("Days post-CAP initiation") + 
  theme(legend.position = "none") +
  scale_y_log10(limits = c(0.005, 6)) + 
  facet_wrap(~Sex) + 
  stat_compare_means(method = "t.test", paired = TRUE, method.args = list(alternative = "less"), comparisons = list(c("0-1", "2-5"), c("0-1", "6-9")), aes(label = paste0("p = ", after_stat(p.format))), data = dplyr::filter(a_plot, Patient_ID %in% pt_paired))

# Save
fn <- paste0(figdir, "SexPreTA.pdf") 
ggsave(filename = fn, plot = p, width = 4.5, height = 3)
```

## Baseline sex permanova
```{r}
rerun_permanova = TRUE
filter = TRUE

# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/TaxaSummaryASVs.csv")
d <- fread(fn) %>% as.data.frame()
rownames(d) <- d$V1; d <- d %>% select(-V1) 

# CLR transform
geometric_means <- apply(d, 1, function(row) exp(mean(log(row[row > 0]))))
pseudo_count = min(d[d > 0])/2
clr_data <- log((d + pseudo_count) / geometric_means)

# Omit NA values
clr_data <- na.omit(clr_data)

# Remove zero-variance columns
clr_data_t <- t(clr_data)
clr_data_t <- clr_data_t[ , which(apply(clr_data_t, 2, var) != 0)]
d_t <- t(d)
d_t <- d_t[ , which(apply(d_t, 2, var) != 0)]

# Concatenate metadata
meta_c <- meta %>% filter(Sample_ID %in% rownames(clr_data_t) & Sample_ID %in% rownames(d_t)) # only keep samples that we have sequencing data for
clr_data_t <- clr_data_t[meta_c$Sample_ID,] # and vice versa for metadata matching
d_t <- d_t[meta_c$Sample_ID,]

# PERMANOVA testing 
meta_perm <- meta_c 
if (filter){
  meta_row <- meta_c[meta_c$Sample_ID == "GO-Pt6-C1D1",]
  meta_row$Treatment_Cycle <- "Baseline"  # Establish Patient 6 baseline (only patient that has C1D1, but no BL)
  meta_perm <- rbind(meta_perm, meta_row)
  meta_perm <- meta_perm %>% filter(Treatment_Cycle %in% c("Baseline"))
}
row_order <- meta_perm %>% select(Sample_ID) %>% pull(); clr_data_ordered <-clr_data_t[row_order,]; d_ordered <- d_t[row_order,] # arrange
if(sum(meta_perm$Sample_ID == rownames(clr_data_ordered)) + sum(meta_perm$Sample_ID == rownames(d_ordered)) != 2*length(meta_perm$Sample_ID)){ # check that they match
    print("PERMANVA order wrong!")
}
if (rerun_permanova){
  dist_clr <- suppressWarnings(vegdist(clr_data_ordered, method="euclidean", na.rm = TRUE, binary = TRUE)) # CLR-Euclidean
  dist_bray <- vegdist(d_ordered, method="bray", na.rm = TRUE) # Bray
  set.seed(42)
  res_bray <- adonis2(dist_bray ~ Metadata_Sex, data = meta_perm, permutations=1000, by = "margin")
  res_clr <- adonis2(dist_clr ~ Metadata_Sex, data = meta_perm, permutations=1000, by = "margin")
}
```

# Figure S16 - efficacy
```{r}
figdir <- paste0(dir, "FigureS16/")
dir.create(figdir)

# Load date data
fn = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/GO_metadata_16S_WithDates.xlsx")
dates <- read_excel(fn) %>% as.data.frame() %>% select(c(Sample_ID, Patient_ID, Treatment_Cycle, Batch_16S, Days_Since_Tx))
dates <- dates %>% filter(!(Patient_ID %in% c("9", "9c")))
dates$Patient_ID[dates$Patient_ID == "9b"] <- "9"
```

## Bargraph of incidence
```{r}
Toxicity = c("Progression", "No progression")
Number = c(12, 28)
ToxDF = data.frame(Toxicity = Toxicity, Number = Number, Percent = round(Number/40*100))

# Plot
# Number of patients is out of 40
p <- ggplot(ToxDF, aes(x = reorder(Toxicity, Number), y = Number)) +
  geom_bar(stat = "identity", color = "black", fill = "black") +
  coord_flip() + # Flip coordinates to have 'Toxicity' on the y-axis
  geom_text(aes(label = paste0(Number, " (", Percent, "%)")), hjust = -0.1) + # Add text labels
  labs(x = "", y = "Incidence [N (%)]") +
  theme_pubr() + 
  theme(legend.position = "none") + 
  ylim(0, 40)

# Save
fn <- paste0(figdir, "GO_ProgBargraph.pdf")
ggsave(filename = fn, plot = p, width = 4, height = 2)
```

## Efficacy vs preTA
```{r}
# Load date data
fn = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_16S/GO_metadata_16S_WithDates.xlsx")
dates <- read_excel(fn) %>% as.data.frame() %>% select(c(Sample_ID, Patient_ID, Treatment_Cycle, Batch_16S, Days_Since_Tx))
dates <- dates %>% filter(!(Patient_ID %in% c("9", "9c")))
dates$Patient_ID[dates$Patient_ID == "9b"] <- "9"

# Add toxicity data
pts_progress <- c("20", "30", "47", "14", "6", "17", # "progress*" in REDCAP
                  "2", "4", "7", "8", "13", "15", "44") # GO_Samples_TLab_DS spreadsheet for discontinued (clinical decline) or deceased
dates$Progression <- case_when(dates$Patient_ID %in% pts_progress ~ "Yes",
                            TRUE ~ "No")

# Load preta data
fn1 <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/KO_stratified_K17722.csv")
fn2 <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/KO_stratified_K17723.csv")
preta1 <- data.table::fread(fn1)
preta2 <- data.table::fread(fn2)
preta <- rbind(preta1, preta2)
preta$MetaphlanID <- preta$Sample_ID

# Metadata
preta <- preta %>% group_by(MetaphlanID) %>% dplyr::summarise(Abundance = sum(RPKG)) %>% as.data.frame()
preta$Treatment_Cycle <- case_when(grepl("Baseline", preta$MetaphlanID) ~ "Baseline",
                                   grepl("C1D1", preta$MetaphlanID) ~ "C1D1",
                                   grepl("C1D3", preta$MetaphlanID) ~ "C1D3",
                                   grepl("C1D7", preta$MetaphlanID) ~ "C1D7",
                                   grepl("C2D1", preta$MetaphlanID) ~ "C2D1",
                                   grepl("C3D1", preta$MetaphlanID) ~ "C3D1",
                                   grepl("Final", preta$MetaphlanID) ~ "Final",
                                   TRUE ~ "Other")
preta$Patient_ID <- sub(".*Pt", "", preta$MetaphlanID) 
preta$Patient_ID <- sub("_.*", "", preta$Patient_ID) 
preta$Sample_ID <- paste0("GO-Pt", preta$Patient_ID, "-", preta$Treatment_Cycle)

# Remove duplicates
dups = preta %>% 
  group_by(Sample_ID) %>% 
  dplyr::summarise(n = n()) %>% 
  as.data.frame() %>% 
  filter(n >= 2) %>%
  select(Sample_ID) %>% 
  pull()
preta <- preta %>% filter(!(MetaphlanID %in% c("GO_Pt40_C1D7_S82", "GO_Pt1_C1D7_S56", "GO_Pt12_C2D1_S7")))

# Metadata
fn = paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/Metadata170523.csv")
met <- read.csv(fn) %>% as.data.frame()
metadata_full <- met %>% select(-c("Treatment_Cycle", "Patient_ID", "Sample_ID"))
preta$MetaphlanID <- gsub("_S.*", "", preta$MetaphlanID)
preta <- inner_join(preta, metadata_full, by = "MetaphlanID")
preta$Patient_ID[preta$Patient_ID == "9b"] <- "9"
preta$Dose <- case_when(preta$Patient_ID %in% pts_progress ~ "Progression",
                        TRUE ~ "No progression")

# Check missing patients
met_add <- met
met_add$Patient_ID[met_add$Patient_ID == "9b"] <- "9"
missing_pts <- setdiff(unique(met_add$Patient_ID), unique(preta$Patient_ID))

# Average baseline and C1D1 preTA (or if only one available, use that one)
bms <- preta %>% filter(Treatment_Cycle %in% c("Baseline", "C1D1")) 
bms <- bms %>% group_by(Patient_ID, Dose, Metadata_Radiation) %>% dplyr::summarise(Abundance = mean(Abundance))

# Add missing patients - 11, 7, 53
pts_lost <- setdiff(unique(preta$Patient_ID), unique(bms$Patient_ID))
pt_add <- c(pts_lost, missing_pts)
bms_add <- met %>% filter(Patient_ID %in% pt_add) %>% 
  filter(Treatment_Cycle == "Baseline") %>% 
  select(all_of(c("Patient_ID", "Metadata_Radiation")))
bms_add$Abundance <- 0
bms <- rbind(bms, bms_add)

# Fix dosing info
bms$Dose <- case_when(bms$Patient_ID %in% pts_progress ~ "Progression",
                      TRUE ~ "No progression")
bms$Group <- factor(bms$Dose, levels = c("No progression", "Progression"))

# Plot
offset = min(bms$Abundance[bms$Abundance > 0])/2
p <- ggplot(bms, aes(x = Group, y = Abundance + offset)) + 
  geom_boxplot(aes(fill = Group), outlier.size = -1, alpha = 0.5) + 
  geom_jitter(width = 0.05, aes(color = Group)) + 
  theme_pubr() + 
  xlab("") + 
  ylab("PreTA level (RPKG)") +
  scale_y_log10(limits = c(offset/2, 2)) + 
  scale_color_manual(values = c("Progression" = "red", "No progression" = "grey")) + 
  scale_fill_manual(values = c("Progression" = "red", "No progression" = "grey")) + 
  stat_compare_means(label.x = 1.2, label.y = log10(1.5), method = "wilcox.test", aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE) +
  theme(legend.position = "none") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) 

# Save
fn <- paste0(figdir, "GO_PtEfficacy.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)
```

## Genes PERMANOVA progression
```{r}
rerun_permanova = TRUE
filter = TRUE # TRUE -> only do Baseline 
pts_progress <- c("20", "30", "47", "14", "6", "17", # "progress*" in REDCAP
                  "2", "4", "7", "8", "13", "15", "44") # GO_Samples_TLab_DS spreadsheet for discontinued (clinical decline) or deceased

# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/genefamilies_unstratified_nonzero0.csv")
d <- fread(fn) %>% as.data.frame()
rownames(d) <- d$KO; d <- d %>% select(-KO) %>% select(-V1)

# CLR transform
geometric_means <- apply(d, 1, function(row) exp(mean(log(row[row > 0]))))
pseudo_count = min(d[d > 0])/2
clr_data <- log((d + pseudo_count) / geometric_means)

# Omit NA values
clr_data <- na.omit(clr_data)

# Remove zero-variance columns
clr_data_t <- t(clr_data)
clr_data_t <- clr_data_t[ , which(apply(clr_data_t, 2, var) != 0)]
d_t <- t(d)
d_t <- d_t[ , which(apply(d_t, 2, var) != 0)]

# Load metadata
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/GO_MGS/Metadata170523.csv")
meta <- read.csv(fn) %>% as.data.frame()
meta$Sample_ID <- gsub("-", "_", meta$Sample_ID)
meta <- meta %>% filter(Sample_ID %in% rownames(clr_data_t) & Sample_ID %in% rownames(d_t)) # only keep samples that we have sequencing data for
meta$Progression <- case_when(meta$Patient_ID %in% pts_progress ~ "Yes", TRUE ~ "No")
clr_data_t <- clr_data_t[meta$Sample_ID,] # and vice versa for metadata matching
d_t <- d_t[meta$Sample_ID,]

# PERMANOVA testing 
meta_perm <- meta 
if (filter){
  meta_row <- meta[meta$Sample_ID == "GO_Pt6_C1D1",]
  meta_row$Treatment_Cycle <- "Baseline"  # Establish Patient 6 baseline (only patient that has C1D1, but no BL)
  meta_perm <- rbind(meta_perm, meta_row)
  meta_perm <- meta_perm %>% filter(Treatment_Cycle %in% c("Baseline"))
}
row_order <- meta_perm %>% select(Sample_ID) %>% pull(); clr_data_ordered <-clr_data_t[row_order,]; d_ordered <- d_t[row_order,] # arrange
if(sum(meta_perm$Sample_ID == rownames(clr_data_ordered)) + sum(meta_perm$Sample_ID == rownames(d_ordered)) != 2*length(meta_perm$Sample_ID)){ # check that they match
    print("PERMANVA order wrong!")
}
if (rerun_permanova){
  dist_clr <- suppressWarnings(vegdist(clr_data_ordered, method="euclidean", na.rm = TRUE, binary = TRUE)) # CLR-Euclidean
  dist_bray <- vegdist(d_ordered, method="bray", na.rm = TRUE) # Bray
  set.seed(42)
  res_bray <- adonis2(dist_bray ~ Progression, data = meta_perm, permutations=1000, by = "margin")
  res_clr <- adonis2(dist_clr ~ Progression, data = meta_perm, permutations=1000, by = "margin")
}
```

# Figure S17 - prediction
```{r}
figdir <- paste0(dir, "FigureS17/")
dir.create(figdir)
```

## GO prediction - preta
```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Predictions/GO_Tox.csv")
data_lr <- read.csv(fn) %>% as.data.frame()
data_lr$Outcome <- case_when(data_lr$Dose == "Toxicity" ~ 1,
                             TRUE ~ 0) # set "Tox" to 1
offset <- min(data_lr$Abundance[data_lr$Abundance > 0]) 

# Modeling
logit_m = glm(formula = Outcome ~ log10(Abundance + offset), data = data_lr, family='binomial')
logit_P = predict(logit_m, newdata = data_lr, type = 'response')
roc_score=roc(response = data_lr$Outcome, predictor = logit_P) #AUC score

# Remove duplicates and sort for proper step formation
auc <- roc_score$auc
aic <- logit_m$aic
roc_data <- data.frame("Sensitivity" = roc_score$sensitivities, "Specificity" = roc_score$specificities)
roc_data <- roc_data[!duplicated(roc_data$Specificity, fromLast = TRUE),]
roc_data <- roc_data[order(roc_data$Specificity, decreasing = TRUE),]

# Plot
p <- ggplot(roc_data, aes(y = Sensitivity, x = 1-Specificity)) + 
  geom_step(direction = "vh") + 
  xlab("FPR") + 
  ylab("TPR") + 
  theme_pubr() + 
  annotate("text", label = paste0("AUC = ", round(100*auc)/100), x = 0.25, y = 1) 

# Save
fn <- paste0(figdir, "ROC_GO.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## GO dose reduction - preta
```{r}
# Read in data
reduc <- c("7", "11", "25", "20", "38", "1") # "reduc", "decreas"

fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Predictions/GO_Tox.csv")
data_lr <- read.csv(fn) %>% as.data.frame()
data_lr$Outcome <- case_when(data_lr$Patient_ID %in% reduc ~ 1,
                             TRUE ~ 0) # set "Tox" to 1
offset <- min(data_lr$Abundance[data_lr$Abundance > 0]) 

# Modeling
logit_m = glm(formula = Outcome ~ log10(Abundance + offset), data = data_lr, family='binomial')
logit_P = predict(logit_m, newdata = data_lr, type = 'response')
roc_score=roc(response = data_lr$Outcome, predictor = logit_P) #AUC score

# Remove duplicates and sort for proper step formation
auc <- roc_score$auc
aic <- logit_m$aic
roc_data <- data.frame("Sensitivity" = roc_score$sensitivities, "Specificity" = roc_score$specificities)
roc_data <- roc_data[!duplicated(roc_data$Specificity, fromLast = TRUE),]
roc_data <- roc_data[order(roc_data$Specificity, decreasing = TRUE),]

# Plot
p <- ggplot(roc_data, aes(y = Sensitivity, x = 1-Specificity)) + 
  geom_step(direction = "vh") + 
  xlab("FPR") + 
  ylab("TPR") + 
  theme_pubr() + 
  annotate("text", label = paste0("AUC = ", round(100*auc)/100), x = 0.25, y = 1) 

# Save
fn <- paste0(figdir, "ROC_GO_Dose.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```


## Mouse prediction - preta
```{r}
# Read in data
fn <- paste0(prefix, "ktrepka/GOStudy/1_FIGURES/Compiled_Data/Predictions/Mouse_Tox.csv")
data_lr <- read.csv(fn) %>% as.data.frame()
data_lr$Dose <- case_when(data_lr$EndpointWeightPercent <= 95 ~ "Toxicity", TRUE ~ "No toxicity") # 5% weight loss = "Toxicity"
data_lr$Outcome <- case_when(data_lr$Dose == "Toxicity" ~ 1,
                             TRUE ~ 0) # set "Tox" to 1
offset <- min(data_lr$Abundance[data_lr$Abundance > 0]) 

# Modeling
logit_m = glm(formula = Outcome ~ log10(Abundance + offset), data = data_lr, family='binomial')
logit_P = predict(logit_m, newdata = data_lr, type = 'response')
roc_score=roc(response = data_lr$Outcome, predictor = logit_P) #AUC score

# Remove duplicates and sort for proper step formation
auc <- roc_score$auc
aic <- logit_m$aic
roc_data <- data.frame("Sensitivity" = roc_score$sensitivities, "Specificity" = roc_score$specificities)
roc_data <- roc_data[!duplicated(roc_data$Specificity, fromLast = TRUE),]
roc_data <- roc_data[order(roc_data$Specificity, decreasing = TRUE),]

# Plot
p <- ggplot(roc_data, aes(y = Sensitivity, x = 1-Specificity)) + 
  geom_step(direction = "vh") + 
  xlab("FPR") + 
  ylab("TPR") + 
  theme_pubr() + 
  annotate("text", label = paste0("AUC = ", round(100*auc)/100), x = 0.2, y = 1) 

# Save
fn <- paste0(figdir, "ROC_Mouse.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```